{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pynoma import RegionSearch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "rs = RegionSearch(3, 4, 1002741, 1002771)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "rs = RegionSearch(3, 4, 987090, 987118)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ac': 1,\n",
       " 'ac_hemi': 0,\n",
       " 'ac_hom': 0,\n",
       " 'an': 143122,\n",
       " 'af': 6.987046016685066e-06,\n",
       " 'filters': [],\n",
       " 'populations': [{'id': 'AFR',\n",
       "   'ac': 1,\n",
       "   'an': 42014,\n",
       "   'ac_hemi': 0,\n",
       "   'ac_hom': 0},\n",
       "  {'id': 'AMI', 'ac': 0, 'an': 900, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'AMR', 'ac': 0, 'an': 13644, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'ASJ', 'ac': 0, 'an': 3324, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'EAS', 'ac': 0, 'an': 3128, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'FIN', 'ac': 0, 'an': 10412, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'NFE', 'ac': 0, 'an': 64504, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'OTH', 'ac': 0, 'an': 2148, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'SAS', 'ac': 0, 'an': 3048, 'ac_hemi': 0, 'ac_hom': 0}]}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal, clinical = rs.get_data(standard=False)\n",
    "normal['genome'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>flags</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>hgvs</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>lof</th>\n",
       "      <th>lof_filter</th>\n",
       "      <th>lof_flags</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>exome</th>\n",
       "      <th>genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val2Ala</td>\n",
       "      <td>c.4T&gt;C</td>\n",
       "      <td>p.Val2Ala</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987090</td>\n",
       "      <td>rs918033021</td>\n",
       "      <td>4-987090-T-C</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Leu4_Pro6dup</td>\n",
       "      <td>c.11_19dup</td>\n",
       "      <td>p.Leu4_Pro6dup</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987090</td>\n",
       "      <td>rs928648528</td>\n",
       "      <td>4-987090-T-TCCCCTGCGC</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 3, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Pro3His</td>\n",
       "      <td>c.8C&gt;A</td>\n",
       "      <td>p.Pro3His</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987092</td>\n",
       "      <td>rs1333424959</td>\n",
       "      <td>4-987092-C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 5, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Leu4Leu</td>\n",
       "      <td>c.10C&gt;T</td>\n",
       "      <td>p.Leu4Leu</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987094</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987094-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 4, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Leu4Pro</td>\n",
       "      <td>c.11T&gt;C</td>\n",
       "      <td>p.Leu4Pro</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987095</td>\n",
       "      <td>rs180984980</td>\n",
       "      <td>4-987095-T-C</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 87, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Arg5Pro</td>\n",
       "      <td>c.14G&gt;C</td>\n",
       "      <td>p.Arg5Pro</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987098</td>\n",
       "      <td>rs1352458557</td>\n",
       "      <td>4-987098-G-C</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala5Asp</td>\n",
       "      <td>c.13C&gt;A</td>\n",
       "      <td>p.Ala5Asp</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987099</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987099-C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala7Val</td>\n",
       "      <td>c.19C&gt;T</td>\n",
       "      <td>p.Ala7Val</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987105</td>\n",
       "      <td>rs1240723907</td>\n",
       "      <td>4-987105-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala8Val</td>\n",
       "      <td>c.23C&gt;T</td>\n",
       "      <td>p.Ala8Val</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987107</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987107-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Pro8Gln</td>\n",
       "      <td>c.22C&gt;A</td>\n",
       "      <td>p.Pro8Gln</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987108</td>\n",
       "      <td>rs11248061</td>\n",
       "      <td>4-987108-C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 56875, 'ac_hemi': 0, 'ac_hom': 11517, '...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Leu13_Ala15dup</td>\n",
       "      <td>c.36_44dup</td>\n",
       "      <td>p.Leu13_Ala15dup</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987114</td>\n",
       "      <td>rs1284049920</td>\n",
       "      <td>4-987114-G-GCTGGCGCTC</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 58, 'ac_hemi': 0, 'ac_hom': 1, 'an': 14...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Cys10Tyr</td>\n",
       "      <td>c.28G&gt;A</td>\n",
       "      <td>p.Cys10Tyr</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987114</td>\n",
       "      <td>rs549738466</td>\n",
       "      <td>4-987114-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 5, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Leu11Val</td>\n",
       "      <td>c.31C&gt;G</td>\n",
       "      <td>p.Leu11Val</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987115</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987115-C-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ser16_Ala19del</td>\n",
       "      <td>c.46_57del</td>\n",
       "      <td>p.Ser16_Ala19del</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987118</td>\n",
       "      <td>rs398123260</td>\n",
       "      <td>4-987118-GCGCTCCTGGCCT-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 7, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ser16_Ala19dup</td>\n",
       "      <td>c.46_57dup</td>\n",
       "      <td>p.Ser16_Ala19dup</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987118</td>\n",
       "      <td>rs938656749</td>\n",
       "      <td>4-987118-G-GCGCTCCTGGCCT</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           consequence flags          gene_id gene_symbol              hgvs  \\\n",
       "0     missense_variant    []  ENSG00000127415        IDUA         p.Val2Ala   \n",
       "1    inframe_insertion    []  ENSG00000127415        IDUA    p.Leu4_Pro6dup   \n",
       "2     missense_variant    []  ENSG00000127415        IDUA         p.Pro3His   \n",
       "3   synonymous_variant    []  ENSG00000127415        IDUA         p.Leu4Leu   \n",
       "4     missense_variant    []  ENSG00000127415        IDUA         p.Leu4Pro   \n",
       "5     missense_variant    []  ENSG00000127415        IDUA         p.Arg5Pro   \n",
       "6     missense_variant    []  ENSG00000127415        IDUA         p.Ala5Asp   \n",
       "7     missense_variant    []  ENSG00000127415        IDUA         p.Ala7Val   \n",
       "8     missense_variant    []  ENSG00000127415        IDUA         p.Ala8Val   \n",
       "9     missense_variant    []  ENSG00000127415        IDUA         p.Pro8Gln   \n",
       "10   inframe_insertion    []  ENSG00000127415        IDUA  p.Leu13_Ala15dup   \n",
       "11    missense_variant    []  ENSG00000127415        IDUA        p.Cys10Tyr   \n",
       "12    missense_variant    []  ENSG00000127415        IDUA        p.Leu11Val   \n",
       "13    inframe_deletion    []  ENSG00000127415        IDUA  p.Ser16_Ala19del   \n",
       "14   inframe_insertion    []  ENSG00000127415        IDUA  p.Ser16_Ala19dup   \n",
       "\n",
       "         hgvsc             hgvsp   lof lof_filter lof_flags     pos  \\\n",
       "0       c.4T>C         p.Val2Ala  None       None      None  987090   \n",
       "1   c.11_19dup    p.Leu4_Pro6dup  None       None      None  987090   \n",
       "2       c.8C>A         p.Pro3His  None       None      None  987092   \n",
       "3      c.10C>T         p.Leu4Leu  None       None      None  987094   \n",
       "4      c.11T>C         p.Leu4Pro  None       None      None  987095   \n",
       "5      c.14G>C         p.Arg5Pro  None       None      None  987098   \n",
       "6      c.13C>A         p.Ala5Asp  None       None      None  987099   \n",
       "7      c.19C>T         p.Ala7Val  None       None      None  987105   \n",
       "8      c.23C>T         p.Ala8Val  None       None      None  987107   \n",
       "9      c.22C>A         p.Pro8Gln  None       None      None  987108   \n",
       "10  c.36_44dup  p.Leu13_Ala15dup  None       None      None  987114   \n",
       "11     c.28G>A        p.Cys10Tyr  None       None      None  987114   \n",
       "12     c.31C>G        p.Leu11Val  None       None      None  987115   \n",
       "13  c.46_57del  p.Ser16_Ala19del  None       None      None  987118   \n",
       "14  c.46_57dup  p.Ser16_Ala19dup  None       None      None  987118   \n",
       "\n",
       "            rsid                variant_id exome  \\\n",
       "0    rs918033021              4-987090-T-C  None   \n",
       "1    rs928648528     4-987090-T-TCCCCTGCGC  None   \n",
       "2   rs1333424959              4-987092-C-A  None   \n",
       "3           None              4-987094-C-T  None   \n",
       "4    rs180984980              4-987095-T-C  None   \n",
       "5   rs1352458557              4-987098-G-C  None   \n",
       "6           None              4-987099-C-A  None   \n",
       "7   rs1240723907              4-987105-C-T  None   \n",
       "8           None              4-987107-C-T  None   \n",
       "9     rs11248061              4-987108-C-A  None   \n",
       "10  rs1284049920     4-987114-G-GCTGGCGCTC  None   \n",
       "11   rs549738466              4-987114-G-A  None   \n",
       "12          None              4-987115-C-G  None   \n",
       "13   rs398123260  4-987118-GCGCTCCTGGCCT-G  None   \n",
       "14   rs938656749  4-987118-G-GCGCTCCTGGCCT  None   \n",
       "\n",
       "                                               genome  \n",
       "0   {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "1   {'ac': 3, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "2   {'ac': 5, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "3   {'ac': 4, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "4   {'ac': 87, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...  \n",
       "5   {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "6   {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "7   {'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "8   {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "9   {'ac': 56875, 'ac_hemi': 0, 'ac_hom': 11517, '...  \n",
       "10  {'ac': 58, 'ac_hemi': 0, 'ac_hom': 1, 'an': 14...  \n",
       "11  {'ac': 5, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "12  {'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "13  {'ac': 7, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "14  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Trp402Ter</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>[]</td>\n",
       "      <td>120</td>\n",
       "      <td>142464</td>\n",
       "      <td>8.423181e-04</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-1002750-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Val</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>141870</td>\n",
       "      <td>1.409741e-05</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-1002751-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143004</td>\n",
       "      <td>1.398562e-05</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-1002756-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val405Gly</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>141746</td>\n",
       "      <td>7.054873e-06</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-1002759-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ser406Trp</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>142602</td>\n",
       "      <td>1.402505e-05</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4-1002766-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala408Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>143012</td>\n",
       "      <td>2.097726e-05</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4-1002767-G-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Arg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2750</td>\n",
       "      <td>142238</td>\n",
       "      <td>1.933379e-02</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4-1002768-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Glu</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143002</td>\n",
       "      <td>1.398582e-05</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4-1002770-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Thr410Pro</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>140858</td>\n",
       "      <td>3.549674e-05</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4-1002770-AC-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val411SerfsTer29</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>140860</td>\n",
       "      <td>7.099247e-06</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variant ID  Gene         Consequence          Annotation Flags  \\\n",
       "0   4-1002747-G-A  IDUA         p.Trp402Ter         stop_gained    []   \n",
       "1   4-1002750-C-T  IDUA         p.Ala403Val    missense_variant    []   \n",
       "2   4-1002751-C-G  IDUA         p.Ala403Ala  synonymous_variant    []   \n",
       "3   4-1002756-T-G  IDUA         p.Val405Gly    missense_variant    []   \n",
       "4   4-1002759-C-G  IDUA         p.Ser406Trp    missense_variant    []   \n",
       "5   4-1002766-C-T  IDUA         p.Ala408Ala  synonymous_variant    []   \n",
       "6   4-1002767-G-C  IDUA         p.Gly409Arg    missense_variant    []   \n",
       "7   4-1002768-G-A  IDUA         p.Gly409Glu    missense_variant    []   \n",
       "8   4-1002770-A-C  IDUA         p.Thr410Pro    missense_variant    []   \n",
       "9  4-1002770-AC-A  IDUA  p.Val411SerfsTer29  frameshift_variant    []   \n",
       "\n",
       "   Allele Count  Allele Number Allele Frequency  Number of Homozygotes  \n",
       "0           120         142464     8.423181e-04                      0  \n",
       "1             2         141870     1.409741e-05                      0  \n",
       "2             2         143004     1.398562e-05                      0  \n",
       "3             1         141746     7.054873e-06                      0  \n",
       "4             2         142602     1.402505e-05                      0  \n",
       "5             3         143012     2.097726e-05                      0  \n",
       "6          2750         142238     1.933379e-02                     67  \n",
       "7             2         143002     1.398582e-05                      0  \n",
       "8             5         140858     3.549674e-05                      0  \n",
       "9             1         140860     7.099247e-06                      0  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal, clinical = rs.get_data()\n",
    "normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>African</th>\n",
       "      <th>Amish</th>\n",
       "      <th>Latino</th>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <th>East Asian</th>\n",
       "      <th>European (Finnish)</th>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <th>Other</th>\n",
       "      <th>South Asian</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Trp402Ter</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>[]</td>\n",
       "      <td>120</td>\n",
       "      <td>142464</td>\n",
       "      <td>8.423181e-04</td>\n",
       "      <td>0</td>\n",
       "      <td>3.122898e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.205882e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.906413e-04</td>\n",
       "      <td>1.552843e-03</td>\n",
       "      <td>4.668534e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-1002750-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Val</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>141870</td>\n",
       "      <td>1.409741e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>4.811856e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-1002751-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143004</td>\n",
       "      <td>1.398562e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.103181e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-1002756-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val405Gly</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>141746</td>\n",
       "      <td>7.054873e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.561378e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-1002759-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ser406Trp</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>142602</td>\n",
       "      <td>1.402505e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.108873e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4-1002766-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala408Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>143012</td>\n",
       "      <td>2.097726e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.656072e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4-1002767-G-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Arg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2750</td>\n",
       "      <td>142238</td>\n",
       "      <td>1.933379e-02</td>\n",
       "      <td>67</td>\n",
       "      <td>5.884051e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.006465e-02</td>\n",
       "      <td>9.052505e-03</td>\n",
       "      <td>1.856594e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.666791e-04</td>\n",
       "      <td>2.192164e-02</td>\n",
       "      <td>6.609385e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4-1002768-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Glu</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143002</td>\n",
       "      <td>1.398582e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.104144e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4-1002770-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Thr410Pro</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>140858</td>\n",
       "      <td>3.549674e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>1.217552e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4-1002770-AC-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val411SerfsTer29</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>140860</td>\n",
       "      <td>7.099247e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>2.435104e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variant ID  Gene         Consequence          Annotation Flags  \\\n",
       "0   4-1002747-G-A  IDUA         p.Trp402Ter         stop_gained    []   \n",
       "1   4-1002750-C-T  IDUA         p.Ala403Val    missense_variant    []   \n",
       "2   4-1002751-C-G  IDUA         p.Ala403Ala  synonymous_variant    []   \n",
       "3   4-1002756-T-G  IDUA         p.Val405Gly    missense_variant    []   \n",
       "4   4-1002759-C-G  IDUA         p.Ser406Trp    missense_variant    []   \n",
       "5   4-1002766-C-T  IDUA         p.Ala408Ala  synonymous_variant    []   \n",
       "6   4-1002767-G-C  IDUA         p.Gly409Arg    missense_variant    []   \n",
       "7   4-1002768-G-A  IDUA         p.Gly409Glu    missense_variant    []   \n",
       "8   4-1002770-A-C  IDUA         p.Thr410Pro    missense_variant    []   \n",
       "9  4-1002770-AC-A  IDUA  p.Val411SerfsTer29  frameshift_variant    []   \n",
       "\n",
       "   Allele Count  Allele Number Allele Frequency  Number of Homozygotes  \\\n",
       "0           120         142464     8.423181e-04                      0   \n",
       "1             2         141870     1.409741e-05                      0   \n",
       "2             2         143004     1.398562e-05                      0   \n",
       "3             1         141746     7.054873e-06                      0   \n",
       "4             2         142602     1.402505e-05                      0   \n",
       "5             3         143012     2.097726e-05                      0   \n",
       "6          2750         142238     1.933379e-02                     67   \n",
       "7             2         143002     1.398582e-05                      0   \n",
       "8             5         140858     3.549674e-05                      0   \n",
       "9             1         140860     7.099247e-06                      0   \n",
       "\n",
       "        African         Amish        Latino Ashkenazi Jewish    East Asian  \\\n",
       "0  3.122898e-04  0.000000e+00  2.205882e-04     0.000000e+00  0.000000e+00   \n",
       "1  4.811856e-05  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "2  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "3  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "4  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "5  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "6  5.884051e-02  0.000000e+00  1.006465e-02     9.052505e-03  1.856594e-02   \n",
       "7  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "8  1.217552e-04  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "9  2.435104e-05  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "\n",
       "  European (Finnish) European (non-Finnish)         Other   South Asian  \n",
       "0       2.906413e-04           1.552843e-03  4.668534e-04  0.000000e+00  \n",
       "1       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "2       0.000000e+00           3.103181e-05  0.000000e+00  0.000000e+00  \n",
       "3       0.000000e+00           1.561378e-05  0.000000e+00  0.000000e+00  \n",
       "4       0.000000e+00           3.108873e-05  0.000000e+00  0.000000e+00  \n",
       "5       0.000000e+00           4.656072e-05  0.000000e+00  0.000000e+00  \n",
       "6       0.000000e+00           4.666791e-04  2.192164e-02  6.609385e-04  \n",
       "7       0.000000e+00           3.104144e-05  0.000000e+00  0.000000e+00  \n",
       "8       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "9       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal, clinical = rs.get_data(additional_population_info=True)\n",
    "normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>clinvar_variation_id</th>\n",
       "      <th>gold_stars</th>\n",
       "      <th>major_consequence</th>\n",
       "      <th>pos</th>\n",
       "      <th>variant_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[Pathogenic]</td>\n",
       "      <td>11908</td>\n",
       "      <td>2</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1002747</td>\n",
       "      <td>4-1002747-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[Pathogenic]</td>\n",
       "      <td>555230</td>\n",
       "      <td>1</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1002748</td>\n",
       "      <td>4-1002748-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[Pathogenic]</td>\n",
       "      <td>552095</td>\n",
       "      <td>2</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1002752</td>\n",
       "      <td>4-1002752-G-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[Likely_benign]</td>\n",
       "      <td>724507</td>\n",
       "      <td>1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1002766</td>\n",
       "      <td>4-1002766-C-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[Conflicting_interpretations_of_pathogenicity]</td>\n",
       "      <td>92627</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1002767</td>\n",
       "      <td>4-1002767-G-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>[Uncertain_significance]</td>\n",
       "      <td>557616</td>\n",
       "      <td>1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1002768</td>\n",
       "      <td>4-1002768-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>[Uncertain_significance]</td>\n",
       "      <td>593572</td>\n",
       "      <td>1</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>1002771</td>\n",
       "      <td>4-1002771-CCGTCCTGGACAGCAACCACACGGTGGG-C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            clinical_significance clinvar_variation_id  \\\n",
       "0                                    [Pathogenic]                11908   \n",
       "1                                    [Pathogenic]               555230   \n",
       "2                                    [Pathogenic]               552095   \n",
       "3                                 [Likely_benign]               724507   \n",
       "4  [Conflicting_interpretations_of_pathogenicity]                92627   \n",
       "5                        [Uncertain_significance]               557616   \n",
       "6                        [Uncertain_significance]               593572   \n",
       "\n",
       "   gold_stars   major_consequence      pos  \\\n",
       "0           2         stop_gained  1002747   \n",
       "1           1         stop_gained  1002748   \n",
       "2           2         stop_gained  1002752   \n",
       "3           1  synonymous_variant  1002766   \n",
       "4           1    missense_variant  1002767   \n",
       "5           1    missense_variant  1002768   \n",
       "6           1    inframe_deletion  1002771   \n",
       "\n",
       "                                 variant_id  \n",
       "0                             4-1002747-G-A  \n",
       "1                             4-1002748-G-A  \n",
       "2                             4-1002752-G-T  \n",
       "3                             4-1002766-C-T  \n",
       "4                             4-1002767-G-C  \n",
       "5                             4-1002768-G-A  \n",
       "6  4-1002771-CCGTCCTGGACAGCAACCACACGGTGGG-C  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>African</th>\n",
       "      <th>Amish</th>\n",
       "      <th>Latino</th>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <th>East Asian</th>\n",
       "      <th>European (Finnish)</th>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <th>Other</th>\n",
       "      <th>South Asian</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Trp402Ter</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>[]</td>\n",
       "      <td>120</td>\n",
       "      <td>142464</td>\n",
       "      <td>8.423181e-04</td>\n",
       "      <td>0</td>\n",
       "      <td>3.122898e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.205882e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.906413e-04</td>\n",
       "      <td>1.552843e-03</td>\n",
       "      <td>4.668534e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-1002750-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Val</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>141870</td>\n",
       "      <td>1.409741e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>4.811856e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-1002751-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143004</td>\n",
       "      <td>1.398562e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.103181e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-1002756-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val405Gly</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>141746</td>\n",
       "      <td>7.054873e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.561378e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-1002759-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ser406Trp</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>142602</td>\n",
       "      <td>1.402505e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.108873e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4-1002766-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala408Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>143012</td>\n",
       "      <td>2.097726e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.656072e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4-1002767-G-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Arg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2750</td>\n",
       "      <td>142238</td>\n",
       "      <td>1.933379e-02</td>\n",
       "      <td>67</td>\n",
       "      <td>5.884051e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.006465e-02</td>\n",
       "      <td>9.052505e-03</td>\n",
       "      <td>1.856594e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.666791e-04</td>\n",
       "      <td>2.192164e-02</td>\n",
       "      <td>6.609385e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4-1002768-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Glu</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>143002</td>\n",
       "      <td>1.398582e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.104144e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4-1002770-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Thr410Pro</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>140858</td>\n",
       "      <td>3.549674e-05</td>\n",
       "      <td>0</td>\n",
       "      <td>1.217552e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4-1002770-AC-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val411SerfsTer29</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>140860</td>\n",
       "      <td>7.099247e-06</td>\n",
       "      <td>0</td>\n",
       "      <td>2.435104e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variant ID  Gene         Consequence          Annotation Flags  \\\n",
       "0   4-1002747-G-A  IDUA         p.Trp402Ter         stop_gained    []   \n",
       "1   4-1002750-C-T  IDUA         p.Ala403Val    missense_variant    []   \n",
       "2   4-1002751-C-G  IDUA         p.Ala403Ala  synonymous_variant    []   \n",
       "3   4-1002756-T-G  IDUA         p.Val405Gly    missense_variant    []   \n",
       "4   4-1002759-C-G  IDUA         p.Ser406Trp    missense_variant    []   \n",
       "5   4-1002766-C-T  IDUA         p.Ala408Ala  synonymous_variant    []   \n",
       "6   4-1002767-G-C  IDUA         p.Gly409Arg    missense_variant    []   \n",
       "7   4-1002768-G-A  IDUA         p.Gly409Glu    missense_variant    []   \n",
       "8   4-1002770-A-C  IDUA         p.Thr410Pro    missense_variant    []   \n",
       "9  4-1002770-AC-A  IDUA  p.Val411SerfsTer29  frameshift_variant    []   \n",
       "\n",
       "   Allele Count  Allele Number Allele Frequency  Number of Homozygotes  \\\n",
       "0           120         142464     8.423181e-04                      0   \n",
       "1             2         141870     1.409741e-05                      0   \n",
       "2             2         143004     1.398562e-05                      0   \n",
       "3             1         141746     7.054873e-06                      0   \n",
       "4             2         142602     1.402505e-05                      0   \n",
       "5             3         143012     2.097726e-05                      0   \n",
       "6          2750         142238     1.933379e-02                     67   \n",
       "7             2         143002     1.398582e-05                      0   \n",
       "8             5         140858     3.549674e-05                      0   \n",
       "9             1         140860     7.099247e-06                      0   \n",
       "\n",
       "        African         Amish        Latino Ashkenazi Jewish    East Asian  \\\n",
       "0  3.122898e-04  0.000000e+00  2.205882e-04     0.000000e+00  0.000000e+00   \n",
       "1  4.811856e-05  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "2  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "3  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "4  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "5  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "6  5.884051e-02  0.000000e+00  1.006465e-02     9.052505e-03  1.856594e-02   \n",
       "7  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "8  1.217552e-04  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "9  2.435104e-05  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "\n",
       "  European (Finnish) European (non-Finnish)         Other   South Asian  \n",
       "0       2.906413e-04           1.552843e-03  4.668534e-04  0.000000e+00  \n",
       "1       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "2       0.000000e+00           3.103181e-05  0.000000e+00  0.000000e+00  \n",
       "3       0.000000e+00           1.561378e-05  0.000000e+00  0.000000e+00  \n",
       "4       0.000000e+00           3.108873e-05  0.000000e+00  0.000000e+00  \n",
       "5       0.000000e+00           4.656072e-05  0.000000e+00  0.000000e+00  \n",
       "6       0.000000e+00           4.666791e-04  2.192164e-02  6.609385e-04  \n",
       "7       0.000000e+00           3.104144e-05  0.000000e+00  0.000000e+00  \n",
       "8       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "9       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynoma import RegionSearch\n",
    "rs = RegionSearch(3, 4, 1002741, 1002771)\n",
    "normal, clinical = rs.get_data(additional_population_info=True)\n",
    "normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "rs = RegionSearch(3, 4, 987000, 987818)\n",
    "normal, clinical = rs.get_data(standard=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>flags</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>hgvs</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>lof</th>\n",
       "      <th>lof_filter</th>\n",
       "      <th>lof_flags</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>exome</th>\n",
       "      <th>genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-85A&gt;G</td>\n",
       "      <td>c.-85A&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987000</td>\n",
       "      <td>rs1461025242</td>\n",
       "      <td>4-987000-A-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 140...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-83A&gt;G</td>\n",
       "      <td>c.-83A&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987002</td>\n",
       "      <td>rs1332916785</td>\n",
       "      <td>4-987002-A-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 140...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-82T&gt;G</td>\n",
       "      <td>c.-82T&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987003</td>\n",
       "      <td>rs1412324874</td>\n",
       "      <td>4-987003-T-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 139...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-81G&gt;A</td>\n",
       "      <td>c.-81G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987004</td>\n",
       "      <td>rs1386499538</td>\n",
       "      <td>4-987004-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 142...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-80G&gt;T</td>\n",
       "      <td>c.-80G&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987005</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987005-G-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000145217</td>\n",
       "      <td>SLC26A1</td>\n",
       "      <td>c.*1042G&gt;A</td>\n",
       "      <td>c.*1042G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987791</td>\n",
       "      <td>rs1230498270</td>\n",
       "      <td>4-987791-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.159-8G&gt;A</td>\n",
       "      <td>c.159-8G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987801</td>\n",
       "      <td>rs886180069</td>\n",
       "      <td>4-987801-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 23, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala53Val</td>\n",
       "      <td>c.157C&gt;T</td>\n",
       "      <td>p.Ala53Val</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987809</td>\n",
       "      <td>rs758190824</td>\n",
       "      <td>4-987809-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Pro54His</td>\n",
       "      <td>c.160C&gt;A</td>\n",
       "      <td>p.Pro54His</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987812</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987812-C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Pro55Leu</td>\n",
       "      <td>c.164C&gt;T</td>\n",
       "      <td>p.Pro55Leu</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987814</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987814-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 11, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>238 rows  15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               consequence flags          gene_id gene_symbol        hgvs  \\\n",
       "0      5_prime_UTR_variant    []  ENSG00000127415        IDUA    c.-85A>G   \n",
       "1      5_prime_UTR_variant    []  ENSG00000127415        IDUA    c.-83A>G   \n",
       "2      5_prime_UTR_variant    []  ENSG00000127415        IDUA    c.-82T>G   \n",
       "3      5_prime_UTR_variant    []  ENSG00000127415        IDUA    c.-81G>A   \n",
       "4      5_prime_UTR_variant    []  ENSG00000127415        IDUA    c.-80G>T   \n",
       "..                     ...   ...              ...         ...         ...   \n",
       "233    3_prime_UTR_variant    []  ENSG00000145217     SLC26A1  c.*1042G>A   \n",
       "234  splice_region_variant    []  ENSG00000127415        IDUA  c.159-8G>A   \n",
       "235       missense_variant    []  ENSG00000127415        IDUA  p.Ala53Val   \n",
       "236       missense_variant    []  ENSG00000127415        IDUA  p.Pro54His   \n",
       "237       missense_variant    []  ENSG00000127415        IDUA  p.Pro55Leu   \n",
       "\n",
       "          hgvsc       hgvsp   lof lof_filter lof_flags     pos          rsid  \\\n",
       "0      c.-85A>G        None  None       None      None  987000  rs1461025242   \n",
       "1      c.-83A>G        None  None       None      None  987002  rs1332916785   \n",
       "2      c.-82T>G        None  None       None      None  987003  rs1412324874   \n",
       "3      c.-81G>A        None  None       None      None  987004  rs1386499538   \n",
       "4      c.-80G>T        None  None       None      None  987005          None   \n",
       "..          ...         ...   ...        ...       ...     ...           ...   \n",
       "233  c.*1042G>A        None  None       None      None  987791  rs1230498270   \n",
       "234  c.159-8G>A        None  None       None      None  987801   rs886180069   \n",
       "235    c.157C>T  p.Ala53Val  None       None      None  987809   rs758190824   \n",
       "236    c.160C>A  p.Pro54His  None       None      None  987812          None   \n",
       "237    c.164C>T  p.Pro55Leu  None       None      None  987814          None   \n",
       "\n",
       "       variant_id exome                                             genome  \n",
       "0    4-987000-A-G  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 140...  \n",
       "1    4-987002-A-G  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 140...  \n",
       "2    4-987003-T-G  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 139...  \n",
       "3    4-987004-G-A  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 142...  \n",
       "4    4-987005-G-T  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "..            ...   ...                                                ...  \n",
       "233  4-987791-C-T  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "234  4-987801-G-A  None  {'ac': 23, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...  \n",
       "235  4-987809-C-T  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "236  4-987812-C-A  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "237  4-987814-C-T  None  {'ac': 11, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...  \n",
       "\n",
       "[238 rows x 15 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['lof_flag']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "47"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count = 0\n",
    "for f in normal['flags']:\n",
    "    count +=1\n",
    "    if f:\n",
    "        print(f)\n",
    "        break\n",
    "        \n",
    "count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>flags</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>hgvs</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>lof</th>\n",
       "      <th>lof_filter</th>\n",
       "      <th>lof_flags</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>exome</th>\n",
       "      <th>genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala20Val</td>\n",
       "      <td>c.59C&gt;T</td>\n",
       "      <td>p.Ala20Val</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987143</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987143-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>[lof_flag]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Pro22ArgfsTer86</td>\n",
       "      <td>c.60del</td>\n",
       "      <td>p.Pro22ArgfsTer86</td>\n",
       "      <td>HC</td>\n",
       "      <td>None</td>\n",
       "      <td>PHYLOCSF_WEAK</td>\n",
       "      <td>987143</td>\n",
       "      <td>rs751289519</td>\n",
       "      <td>4-987143-CG-C</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Arg20His</td>\n",
       "      <td>c.58G&gt;A</td>\n",
       "      <td>p.Arg20His</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987144</td>\n",
       "      <td>rs10902762</td>\n",
       "      <td>4-987144-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 56995, 'ac_hemi': 0, 'ac_hom': 11572, '...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Pro21Thr</td>\n",
       "      <td>c.61C&gt;A</td>\n",
       "      <td>p.Pro21Thr</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987145</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987145-C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 6, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Pro22Ala</td>\n",
       "      <td>c.64C&gt;G</td>\n",
       "      <td>p.Pro22Ala</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987148</td>\n",
       "      <td>rs1001972534</td>\n",
       "      <td>4-987148-C-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 6, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000145217</td>\n",
       "      <td>SLC26A1</td>\n",
       "      <td>c.*1042G&gt;A</td>\n",
       "      <td>c.*1042G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987791</td>\n",
       "      <td>rs1230498270</td>\n",
       "      <td>4-987791-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.159-8G&gt;A</td>\n",
       "      <td>c.159-8G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987801</td>\n",
       "      <td>rs886180069</td>\n",
       "      <td>4-987801-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 23, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala53Val</td>\n",
       "      <td>c.157C&gt;T</td>\n",
       "      <td>p.Ala53Val</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987809</td>\n",
       "      <td>rs758190824</td>\n",
       "      <td>4-987809-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Pro54His</td>\n",
       "      <td>c.160C&gt;A</td>\n",
       "      <td>p.Pro54His</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987812</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987812-C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Pro55Leu</td>\n",
       "      <td>c.164C&gt;T</td>\n",
       "      <td>p.Pro55Leu</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987814</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987814-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 11, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193 rows  15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               consequence       flags          gene_id gene_symbol  \\\n",
       "45        missense_variant          []  ENSG00000127415        IDUA   \n",
       "46      frameshift_variant  [lof_flag]  ENSG00000127415        IDUA   \n",
       "47        missense_variant          []  ENSG00000127415        IDUA   \n",
       "48        missense_variant          []  ENSG00000127415        IDUA   \n",
       "49        missense_variant          []  ENSG00000127415        IDUA   \n",
       "..                     ...         ...              ...         ...   \n",
       "233    3_prime_UTR_variant          []  ENSG00000145217     SLC26A1   \n",
       "234  splice_region_variant          []  ENSG00000127415        IDUA   \n",
       "235       missense_variant          []  ENSG00000127415        IDUA   \n",
       "236       missense_variant          []  ENSG00000127415        IDUA   \n",
       "237       missense_variant          []  ENSG00000127415        IDUA   \n",
       "\n",
       "                  hgvs       hgvsc              hgvsp   lof lof_filter  \\\n",
       "45          p.Ala20Val     c.59C>T         p.Ala20Val  None       None   \n",
       "46   p.Pro22ArgfsTer86     c.60del  p.Pro22ArgfsTer86    HC       None   \n",
       "47          p.Arg20His     c.58G>A         p.Arg20His  None       None   \n",
       "48          p.Pro21Thr     c.61C>A         p.Pro21Thr  None       None   \n",
       "49          p.Pro22Ala     c.64C>G         p.Pro22Ala  None       None   \n",
       "..                 ...         ...                ...   ...        ...   \n",
       "233         c.*1042G>A  c.*1042G>A               None  None       None   \n",
       "234         c.159-8G>A  c.159-8G>A               None  None       None   \n",
       "235         p.Ala53Val    c.157C>T         p.Ala53Val  None       None   \n",
       "236         p.Pro54His    c.160C>A         p.Pro54His  None       None   \n",
       "237         p.Pro55Leu    c.164C>T         p.Pro55Leu  None       None   \n",
       "\n",
       "         lof_flags     pos          rsid     variant_id exome  \\\n",
       "45            None  987143          None   4-987143-C-T  None   \n",
       "46   PHYLOCSF_WEAK  987143   rs751289519  4-987143-CG-C  None   \n",
       "47            None  987144    rs10902762   4-987144-G-A  None   \n",
       "48            None  987145          None   4-987145-C-A  None   \n",
       "49            None  987148  rs1001972534   4-987148-C-G  None   \n",
       "..             ...     ...           ...            ...   ...   \n",
       "233           None  987791  rs1230498270   4-987791-C-T  None   \n",
       "234           None  987801   rs886180069   4-987801-G-A  None   \n",
       "235           None  987809   rs758190824   4-987809-C-T  None   \n",
       "236           None  987812          None   4-987812-C-A  None   \n",
       "237           None  987814          None   4-987814-C-T  None   \n",
       "\n",
       "                                                genome  \n",
       "45   {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "46   {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "47   {'ac': 56995, 'ac_hemi': 0, 'ac_hom': 11572, '...  \n",
       "48   {'ac': 6, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "49   {'ac': 6, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "..                                                 ...  \n",
       "233  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "234  {'ac': 23, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...  \n",
       "235  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "236  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "237  {'ac': 11, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...  \n",
       "\n",
       "[193 rows x 15 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal.iloc[45:, :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "rs = RegionSearch(3, 4, 987000, 987818)\n",
    "normal, clinical = rs.get_data(standard=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ac': 1,\n",
       " 'ac_hemi': 0,\n",
       " 'ac_hom': 0,\n",
       " 'an': 143154,\n",
       " 'af': 6.985484163907401e-06,\n",
       " 'filters': [],\n",
       " 'populations': [{'id': 'AFR',\n",
       "   'ac': 0,\n",
       "   'an': 42026,\n",
       "   'ac_hemi': 0,\n",
       "   'ac_hom': 0},\n",
       "  {'id': 'AMI', 'ac': 0, 'an': 900, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'AMR', 'ac': 0, 'an': 13644, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'ASJ', 'ac': 0, 'an': 3324, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'EAS', 'ac': 0, 'an': 3130, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'FIN', 'ac': 0, 'an': 10410, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'NFE', 'ac': 1, 'an': 64518, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'OTH', 'ac': 0, 'an': 2152, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "  {'id': 'SAS', 'ac': 0, 'an': 3050, 'ac_hemi': 0, 'ac_hom': 0}]}"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal.iloc[46:47, :]['genome'][46]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>flags</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>hgvs</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>lof</th>\n",
       "      <th>lof_filter</th>\n",
       "      <th>lof_flags</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>exome</th>\n",
       "      <th>genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-85A&gt;G</td>\n",
       "      <td>c.-85A&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987000</td>\n",
       "      <td>rs1461025242</td>\n",
       "      <td>4-987000-A-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 140...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-83A&gt;G</td>\n",
       "      <td>c.-83A&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987002</td>\n",
       "      <td>rs1332916785</td>\n",
       "      <td>4-987002-A-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 140...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-82T&gt;G</td>\n",
       "      <td>c.-82T&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987003</td>\n",
       "      <td>rs1412324874</td>\n",
       "      <td>4-987003-T-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 139...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-81G&gt;A</td>\n",
       "      <td>c.-81G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987004</td>\n",
       "      <td>rs1386499538</td>\n",
       "      <td>4-987004-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 142...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-80G&gt;T</td>\n",
       "      <td>c.-80G&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987005</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987005-G-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000145217</td>\n",
       "      <td>SLC26A1</td>\n",
       "      <td>c.*1042G&gt;A</td>\n",
       "      <td>c.*1042G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987791</td>\n",
       "      <td>rs1230498270</td>\n",
       "      <td>4-987791-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>splice_region_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.159-8G&gt;A</td>\n",
       "      <td>c.159-8G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987801</td>\n",
       "      <td>rs886180069</td>\n",
       "      <td>4-987801-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 23, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala53Val</td>\n",
       "      <td>c.157C&gt;T</td>\n",
       "      <td>p.Ala53Val</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987809</td>\n",
       "      <td>rs758190824</td>\n",
       "      <td>4-987809-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Pro54His</td>\n",
       "      <td>c.160C&gt;A</td>\n",
       "      <td>p.Pro54His</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987812</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987812-C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Pro55Leu</td>\n",
       "      <td>c.164C&gt;T</td>\n",
       "      <td>p.Pro55Leu</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987814</td>\n",
       "      <td>None</td>\n",
       "      <td>4-987814-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 11, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>238 rows  15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               consequence flags          gene_id gene_symbol        hgvs  \\\n",
       "0      5_prime_UTR_variant    []  ENSG00000127415        IDUA    c.-85A>G   \n",
       "1      5_prime_UTR_variant    []  ENSG00000127415        IDUA    c.-83A>G   \n",
       "2      5_prime_UTR_variant    []  ENSG00000127415        IDUA    c.-82T>G   \n",
       "3      5_prime_UTR_variant    []  ENSG00000127415        IDUA    c.-81G>A   \n",
       "4      5_prime_UTR_variant    []  ENSG00000127415        IDUA    c.-80G>T   \n",
       "..                     ...   ...              ...         ...         ...   \n",
       "233    3_prime_UTR_variant    []  ENSG00000145217     SLC26A1  c.*1042G>A   \n",
       "234  splice_region_variant    []  ENSG00000127415        IDUA  c.159-8G>A   \n",
       "235       missense_variant    []  ENSG00000127415        IDUA  p.Ala53Val   \n",
       "236       missense_variant    []  ENSG00000127415        IDUA  p.Pro54His   \n",
       "237       missense_variant    []  ENSG00000127415        IDUA  p.Pro55Leu   \n",
       "\n",
       "          hgvsc       hgvsp   lof lof_filter lof_flags     pos          rsid  \\\n",
       "0      c.-85A>G        None  None       None      None  987000  rs1461025242   \n",
       "1      c.-83A>G        None  None       None      None  987002  rs1332916785   \n",
       "2      c.-82T>G        None  None       None      None  987003  rs1412324874   \n",
       "3      c.-81G>A        None  None       None      None  987004  rs1386499538   \n",
       "4      c.-80G>T        None  None       None      None  987005          None   \n",
       "..          ...         ...   ...        ...       ...     ...           ...   \n",
       "233  c.*1042G>A        None  None       None      None  987791  rs1230498270   \n",
       "234  c.159-8G>A        None  None       None      None  987801   rs886180069   \n",
       "235    c.157C>T  p.Ala53Val  None       None      None  987809   rs758190824   \n",
       "236    c.160C>A  p.Pro54His  None       None      None  987812          None   \n",
       "237    c.164C>T  p.Pro55Leu  None       None      None  987814          None   \n",
       "\n",
       "       variant_id exome                                             genome  \n",
       "0    4-987000-A-G  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 140...  \n",
       "1    4-987002-A-G  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 140...  \n",
       "2    4-987003-T-G  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 139...  \n",
       "3    4-987004-G-A  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 142...  \n",
       "4    4-987005-G-T  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "..            ...   ...                                                ...  \n",
       "233  4-987791-C-T  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "234  4-987801-G-A  None  {'ac': 23, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...  \n",
       "235  4-987809-C-T  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "236  4-987812-C-A  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 143...  \n",
       "237  4-987814-C-T  None  {'ac': 11, 'ac_hemi': 0, 'ac_hom': 0, 'an': 14...  \n",
       "\n",
       "[238 rows x 15 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
