{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from bs4 import BeautifulSoup\n",
    "import html5lib\n",
    "from bs4 import BeautifulSoup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "url = 'https://gnomad.broadinstitute.org/api/'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "header2 = {\n",
    "    \"Host\": \"gnomad.broadinstitute.org\",\n",
    "\"User-Agent\": \"Mozilla/5.0 (X11; Linux x86_64; rv:77.0) Gecko/20100101 Firefox/77.0\",\n",
    "\"Accept\": \"*/*\",\n",
    "\"Accept-Language\": \"en-US,en;q=0.5\",\n",
    "\"Accept-Encoding\": \"gzip, deflate, br\",\n",
    "\"Referer\": \"https://gnomad.broadinstitute.org/region/4-1002741-1002771?dataset=gnomad_r3\",\n",
    "\"content-type\": \"application/json\",\n",
    "\"Origin\": \"https://gnomad.broadinstitute.org\",\n",
    "\"Content-Length\": \"992\",\n",
    "\"Connection\": \"keep-alive\",\n",
    "\"Cookie\": \"_ga=GA1.2.1956078262.1593106327; _gid=GA1.2.2001994588.1593811914; _gat_gtag_UA_85987017_1=1\",\n",
    "\"TE\": \"Trailers\"\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "{\"Request Headers (500 B)\":{\"headers\":[{\"name\":\"Accept\",\"value\":\"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8\"},{\"name\":\"Accept-Encoding\",\"value\":\"gzip, deflate, br\"},{\"name\":\"Accept-Language\",\"value\":\"en-US,en;q=0.5\"},{\"name\":\"Cache-Control\",\"value\":\"no-cache\"},{\"name\":\"Connection\",\"value\":\"keep-alive\"},{\"name\":\"Cookie\",\"value\":\"_ga=GA1.2.1956078262.1593106327; _gid=GA1.2.2001994588.1593811914\"},{\"name\":\"Host\",\"value\":\"gnomad.broadinstitute.org\"},{\"name\":\"Pragma\",\"value\":\"no-cache\"},{\"name\":\"Upgrade-Insecure-Requests\",\"value\":\"1\"},{\"name\":\"User-Agent\",\"value\":\"Mozilla/5.0 (X11; Linux x86_64; rv:77.0) Gecko/20100101 Firefox/77.0\"}]}}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                      Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-45-6320a3903d53>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpost\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mheader2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstatus_code\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/site-packages/requests/api.py\u001b[0m in \u001b[0;36mpost\u001b[0;34m(url, data, json, **kwargs)\u001b[0m\n\u001b[1;32m    117\u001b[0m     \"\"\"\n\u001b[1;32m    118\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 119\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'post'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mjson\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mjson\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    120\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    121\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/site-packages/requests/api.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     59\u001b[0m     \u001b[0;31m# cases, and look like a memory leak in others.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     60\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0msessions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSession\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 61\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     62\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    528\u001b[0m         }\n\u001b[1;32m    529\u001b[0m         \u001b[0msend_kwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msettings\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 530\u001b[0;31m         \u001b[0mresp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprep\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0msend_kwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    531\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    532\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    641\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    642\u001b[0m         \u001b[0;31m# Send the request\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 643\u001b[0;31m         \u001b[0mr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0madapter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    644\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    645\u001b[0m         \u001b[0;31m# Total elapsed time of the request (approximately)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/site-packages/requests/adapters.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    437\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    438\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mchunked\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 439\u001b[0;31m                 resp = conn.urlopen(\n\u001b[0m\u001b[1;32m    440\u001b[0m                     \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    441\u001b[0m                     \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    668\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    669\u001b[0m             \u001b[0;31m# Make the request on the httplib connection object.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 670\u001b[0;31m             httplib_response = self._make_request(\n\u001b[0m\u001b[1;32m    671\u001b[0m                 \u001b[0mconn\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    672\u001b[0m                 \u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    424\u001b[0m                     \u001b[0;31m# Python 3 (including for exceptions like SystemExit).\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    425\u001b[0m                     \u001b[0;31m# Otherwise it looks like a bug in the code.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 426\u001b[0;31m                     \u001b[0msix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_from\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    427\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mSocketTimeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mBaseSSLError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mSocketError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    428\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_raise_timeout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout_value\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mread_timeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/site-packages/urllib3/packages/six.py\u001b[0m in \u001b[0;36mraise_from\u001b[0;34m(value, from_value)\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    419\u001b[0m                 \u001b[0;31m# Python 3\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    420\u001b[0m                 \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 421\u001b[0;31m                     \u001b[0mhttplib_response\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgetresponse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    422\u001b[0m                 \u001b[0;32mexcept\u001b[0m \u001b[0mBaseException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    423\u001b[0m                     \u001b[0;31m# Remove the TypeError from the exception chain in\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/http/client.py\u001b[0m in \u001b[0;36mgetresponse\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1330\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1331\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1332\u001b[0;31m                 \u001b[0mresponse\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbegin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1333\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mConnectionError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1334\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/http/client.py\u001b[0m in \u001b[0;36mbegin\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    301\u001b[0m         \u001b[0;31m# read until we get a non-100 response\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    302\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 303\u001b[0;31m             \u001b[0mversion\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstatus\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreason\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_read_status\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    304\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mstatus\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0mCONTINUE\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    305\u001b[0m                 \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/http/client.py\u001b[0m in \u001b[0;36m_read_status\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    262\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    263\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_read_status\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 264\u001b[0;31m         \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreadline\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_MAXLINE\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"iso-8859-1\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    265\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0m_MAXLINE\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    266\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mLineTooLong\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"status line\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/socket.py\u001b[0m in \u001b[0;36mreadinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    667\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    668\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 669\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    670\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    671\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_timeout_occurred\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/ssl.py\u001b[0m in \u001b[0;36mrecv_into\u001b[0;34m(self, buffer, nbytes, flags)\u001b[0m\n\u001b[1;32m   1239\u001b[0m                   \u001b[0;34m\"non-zero flags not allowed in calls to recv_into() on %s\"\u001b[0m \u001b[0;34m%\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1240\u001b[0m                   self.__class__)\n\u001b[0;32m-> 1241\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnbytes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1242\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1243\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbuffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnbytes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mflags\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.8/ssl.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self, len, buffer)\u001b[0m\n\u001b[1;32m   1097\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1098\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mbuffer\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1099\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sslobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1100\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1101\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sslobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "r = requests.post(url, headers=header2)\n",
    "r.status_code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "soup = BeautifulSoup(r.content, 'html.parser')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<html>\n",
      " <head>\n",
      "  <meta content=\"text/html;charset=utf-8\" http-equiv=\"content-type\"/>\n",
      "  <title>\n",
      "   408 Request Timeout\n",
      "  </title>\n",
      " </head>\n",
      " <body bgcolor=\"#ffffff\" text=\"#000000\">\n",
      "  <h1>\n",
      "   Error: Request Timeout\n",
      "  </h1>\n",
      "  <h2>\n",
      "   Your client has taken too long to issue its request.\n",
      "  </h2>\n",
      "  <h2>\n",
      "  </h2>\n",
      " </body>\n",
      "</html>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(soup.prettify())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'https://gnomad.broadinstitute.org/api/?query=query RegionCoverage($chrom: String!, $start: Int!, $stop: Int!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!, $includeExomeCoverage: Boolean!, $includeGenomeCoverage: Boolean!) {\\n  region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {\\n    exome_coverage(dataset: $datasetId) @include(if: $includeExomeCoverage) {\\n      pos\\n      mean\\n      median\\n      over_1\\n      over_5\\n      over_10\\n      over_15\\n      over_20\\n      over_25\\n      over_30\\n      over_50\\n      over_100\\n    }\\n    genome_coverage(dataset: $datasetId) @include(if: $includeGenomeCoverage) {\\n      pos\\n      mean\\n      median\\n      over_1\\n      over_5\\n      over_10\\n      over_15\\n      over_20\\n      over_25\\n      over_30\\n      over_50\\n      over_100\\n    }\\n  }\\n}\\n'"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import urllib.parse\n",
    "urllib.parse.unquote('https://gnomad.broadinstitute.org/api/?query=query%20RegionCoverage(%24chrom%3A%20String!%2C%20%24start%3A%20Int!%2C%20%24stop%3A%20Int!%2C%20%24datasetId%3A%20DatasetId!%2C%20%24referenceGenome%3A%20ReferenceGenomeId!%2C%20%24includeExomeCoverage%3A%20Boolean!%2C%20%24includeGenomeCoverage%3A%20Boolean!)%20%7B%0A%20%20region(chrom%3A%20%24chrom%2C%20start%3A%20%24start%2C%20stop%3A%20%24stop%2C%20reference_genome%3A%20%24referenceGenome)%20%7B%0A%20%20%20%20exome_coverage(dataset%3A%20%24datasetId)%20%40include(if%3A%20%24includeExomeCoverage)%20%7B%0A%20%20%20%20%20%20pos%0A%20%20%20%20%20%20mean%0A%20%20%20%20%20%20median%0A%20%20%20%20%20%20over_1%0A%20%20%20%20%20%20over_5%0A%20%20%20%20%20%20over_10%0A%20%20%20%20%20%20over_15%0A%20%20%20%20%20%20over_20%0A%20%20%20%20%20%20over_25%0A%20%20%20%20%20%20over_30%0A%20%20%20%20%20%20over_50%0A%20%20%20%20%20%20over_100%0A%20%20%20%20%7D%0A%20%20%20%20genome_coverage(dataset%3A%20%24datasetId)%20%40include(if%3A%20%24includeGenomeCoverage)%20%7B%0A%20%20%20%20%20%20pos%0A%20%20%20%20%20%20mean%0A%20%20%20%20%20%20median%0A%20%20%20%20%20%20over_1%0A%20%20%20%20%20%20over_5%0A%20%20%20%20%20%20over_10%0A%20%20%20%20%20%20over_15%0A%20%20%20%20%20%20over_20%0A%20%20%20%20%20%20over_25%0A%20%20%20%20%20%20over_30%0A%20%20%20%20%20%20over_50%0A%20%20%20%20%20%20over_100%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import json_normalize as json_normalize\n",
    "from tqdm import tqdm\n",
    "import pandas as pd\n",
    "import requests\n",
    "import argparse\n",
    "import json\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n    {\\n      gene(gene_name: \"TP53\") {\\n        variants(dataset: gnomad_r2_1) {\\n          gene_id\\n          gene_symbol\\n          chrom\\n          pos\\n          rsid\\n          ref\\n          alt\\n          consequence\\n          genome {\\n            genome_af:af\\n            genome_ac:ac\\n            genome_an:an\\n            genome_ac_hemi:ac_hemi\\n            genome_ac_hom:ac_hom\\n          }\\n          exome {\\n            exome_af:af\\n            exome_ac:ac\\n            exome_an:an\\n            exome_ac_hemi:ac_hemi\\n            exome_ac_hom:ac_hom\\n          }\\n          flags\\n          lof\\n          consequence_in_canonical_transcript\\n          gene_symbol\\n          hgvsc\\n          lof_filter\\n          lof_flags\\n          hgvsc\\n          hgvsp\\n          reference_genome\\n          variant_id: variantId\\n        }\\n      }\\n    }\\n    '"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"\"\"\n",
    "    {\n",
    "      %s(%s: \"%s\") {\n",
    "        variants(dataset: %s) {\n",
    "          gene_id\n",
    "          gene_symbol\n",
    "          chrom\n",
    "          pos\n",
    "          rsid\n",
    "          ref\n",
    "          alt\n",
    "          consequence\n",
    "          genome {\n",
    "            genome_af:af\n",
    "            genome_ac:ac\n",
    "            genome_an:an\n",
    "            genome_ac_hemi:ac_hemi\n",
    "            genome_ac_hom:ac_hom\n",
    "          }\n",
    "          exome {\n",
    "            exome_af:af\n",
    "            exome_ac:ac\n",
    "            exome_an:an\n",
    "            exome_ac_hemi:ac_hemi\n",
    "            exome_ac_hom:ac_hom\n",
    "          }\n",
    "          flags\n",
    "          lof\n",
    "          consequence_in_canonical_transcript\n",
    "          gene_symbol\n",
    "          hgvsc\n",
    "          lof_filter\n",
    "          lof_flags\n",
    "          hgvsc\n",
    "          hgvsp\n",
    "          reference_genome\n",
    "          variant_id: variantId\n",
    "        }\n",
    "      }\n",
    "    }\n",
    "    \"\"\"\n",
    "\n",
    "query = query % (\"gene\", \"gene_name\", \"TP53\", \"gnomad_r2_1\")\n",
    "query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n  query FetchRegion($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!) {\\n    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {\\n      reference_genome\\n      chrom\\n      start\\n      stop\\n      genes {\\n        gene_id\\n        symbol\\n        start\\n        stop\\n        exons {\\n          feature_type\\n          start\\n          stop\\n        }\\n      }\\n    }\\n  }\\n'"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"\\n  query FetchRegion($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!) {\\n    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {\\n      reference_genome\\n      chrom\\n      start\\n      stop\\n      genes {\\n        gene_id\\n        symbol\\n        start\\n        stop\\n        exons {\\n          feature_type\\n          start\\n          stop\\n        }\\n      }\\n    }\\n  }\\n\"\n",
    "query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'{\"chrom\":\"4\",\"start\":1002741,\"stop\":1002771,\"referenceGenome\":\"GRCh38\",\"datasetId\":\"gnomad_r3\"}'"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variables = \"\"\"{\"chrom\":\"4\",\"start\":1002741,\"stop\":1002771,\"referenceGenome\":\"GRCh38\",\"datasetId\":\"gnomad_r3\"}\"\"\"\n",
    "variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>symbol</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>exons</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>986997</td>\n",
       "      <td>1004506</td>\n",
       "      <td>[{'feature_type': 'CDS', 'start': 987085, 'sto...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           gene_id symbol   start     stop  \\\n",
       "0  ENSG00000127415   IDUA  986997  1004506   \n",
       "\n",
       "                                               exons  \n",
       "0  [{'feature_type': 'CDS', 'start': 987085, 'sto...  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "end_point = \"https://gnomad.broadinstitute.org/api/\"\n",
    "response = requests.post(end_point, data={'query': query, 'variables': variables}, timeout=None)\n",
    "#data = json_normalize(response.json()[\"data\"][\"region\"][\"reference_genome\"])\n",
    "#data.columns = data.columns.map(lambda x: x.split(\".\")[-1])\n",
    "data = json_normalize(response.json()[\"data\"][\"region\"][\"genes\"])\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': {'region': {'reference_genome': 'GRCh38',\n",
       "   'chrom': '4',\n",
       "   'start': 1002741,\n",
       "   'stop': 1002771,\n",
       "   'genes': [{'gene_id': 'ENSG00000127415',\n",
       "     'symbol': 'IDUA',\n",
       "     'start': 986997,\n",
       "     'stop': 1004506,\n",
       "     'exons': [{'feature_type': 'CDS', 'start': 987085, 'stop': 987242},\n",
       "      {'feature_type': 'CDS', 'start': 987809, 'stop': 987949},\n",
       "      {'feature_type': 'CDS', 'start': 991825, 'stop': 991940},\n",
       "      {'feature_type': 'CDS', 'start': 992139, 'stop': 992230},\n",
       "      {'feature_type': 'CDS', 'start': 1000612, 'stop': 1000697},\n",
       "      {'feature_type': 'CDS', 'start': 1000882, 'stop': 1000989},\n",
       "      {'feature_type': 'CDS', 'start': 1001468, 'stop': 1001563},\n",
       "      {'feature_type': 'CDS', 'start': 1001679, 'stop': 1001881},\n",
       "      {'feature_type': 'CDS', 'start': 1001982, 'stop': 1002161},\n",
       "      {'feature_type': 'CDS', 'start': 1002269, 'stop': 1002485},\n",
       "      {'feature_type': 'CDS', 'start': 1002732, 'stop': 1002944},\n",
       "      {'feature_type': 'CDS', 'start': 1003036, 'stop': 1003157},\n",
       "      {'feature_type': 'CDS', 'start': 1003345, 'stop': 1003470},\n",
       "      {'feature_type': 'CDS', 'start': 1003549, 'stop': 1003625},\n",
       "      {'feature_type': 'CDS', 'start': 1004012, 'stop': 1004112},\n",
       "      {'feature_type': 'CDS', 'start': 1004260, 'stop': 1004390},\n",
       "      {'feature_type': 'UTR', 'start': 986997, 'stop': 987084},\n",
       "      {'feature_type': 'UTR', 'start': 991872, 'stop': 992138},\n",
       "      {'feature_type': 'UTR', 'start': 999710, 'stop': 1000064},\n",
       "      {'feature_type': 'UTR', 'start': 1000612, 'stop': 1000697},\n",
       "      {'feature_type': 'UTR', 'start': 1000882, 'stop': 1000892},\n",
       "      {'feature_type': 'UTR', 'start': 1004391, 'stop': 1004506},\n",
       "      {'feature_type': 'exon', 'start': 987044, 'stop': 987242},\n",
       "      {'feature_type': 'exon', 'start': 987809, 'stop': 987949},\n",
       "      {'feature_type': 'exon', 'start': 989231, 'stop': 989623},\n",
       "      {'feature_type': 'exon', 'start': 1000612, 'stop': 1000697},\n",
       "      {'feature_type': 'exon', 'start': 1000882, 'stop': 1001179},\n",
       "      {'feature_type': 'exon', 'start': 1001468, 'stop': 1001563},\n",
       "      {'feature_type': 'exon', 'start': 1001679, 'stop': 1002485},\n",
       "      {'feature_type': 'exon', 'start': 1002732, 'stop': 1002944},\n",
       "      {'feature_type': 'exon', 'start': 1003036, 'stop': 1003470},\n",
       "      {'feature_type': 'exon', 'start': 1003549, 'stop': 1003724},\n",
       "      {'feature_type': 'exon', 'start': 1004012, 'stop': 1004112},\n",
       "      {'feature_type': 'exon', 'start': 1004260, 'stop': 1004505}]}]}}}"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "response.json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'query VariantInRegion($chrom: String!, $start: Int!, $stop: Int!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {\\n  region(start: $start, stop: $stop, chrom: $chrom, reference_genome: $referenceGenome) {\\n    clinvar_variants {\\n      clinical_significance\\n      clinvar_variation_id\\n      gold_stars\\n      major_consequence\\n      pos\\n      variant_id\\n    }\\n    variants(dataset: $datasetId) {\\n      consequence\\n      flags\\n      gene_id\\n      gene_symbol\\n      hgvs\\n      hgvsc\\n      hgvsp\\n      lof\\n      lof_filter\\n      lof_flags\\n      pos\\n      rsid\\n      variant_id: variantId\\n      exome {\\n        ac\\n        ac_hemi\\n        ac_hom\\n        an\\n        af\\n        filters\\n        populations {\\n          id\\n          ac\\n          an\\n          ac_hemi\\n          ac_hom\\n        }\\n      }\\n      genome {\\n        ac\\n        ac_hemi\\n        ac_hom\\n        an\\n        af\\n        filters\\n        populations {\\n          id\\n          ac\\n          an\\n          ac_hemi\\n          ac_hom\\n        }\\n      }\\n    }\\n  }\\n}'"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = \"\"\"query VariantInRegion($chrom: String!, $start: Int!, $stop: Int!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {\n",
    "  region(start: $start, stop: $stop, chrom: $chrom, reference_genome: $referenceGenome) {\n",
    "    clinvar_variants {\n",
    "      clinical_significance\n",
    "      clinvar_variation_id\n",
    "      gold_stars\n",
    "      major_consequence\n",
    "      pos\n",
    "      variant_id\n",
    "    }\n",
    "    variants(dataset: $datasetId) {\n",
    "      consequence\n",
    "      flags\n",
    "      gene_id\n",
    "      gene_symbol\n",
    "      hgvs\n",
    "      hgvsc\n",
    "      hgvsp\n",
    "      lof\n",
    "      lof_filter\n",
    "      lof_flags\n",
    "      pos\n",
    "      rsid\n",
    "      variant_id: variantId\n",
    "      exome {\n",
    "        ac\n",
    "        ac_hemi\n",
    "        ac_hom\n",
    "        an\n",
    "        af\n",
    "        filters\n",
    "        populations {\n",
    "          id\n",
    "          ac\n",
    "          an\n",
    "          ac_hemi\n",
    "          ac_hom\n",
    "        }\n",
    "      }\n",
    "      genome {\n",
    "        ac\n",
    "        ac_hemi\n",
    "        ac_hom\n",
    "        an\n",
    "        af\n",
    "        filters\n",
    "        populations {\n",
    "          id\n",
    "          ac\n",
    "          an\n",
    "          ac_hemi\n",
    "          ac_hom\n",
    "        }\n",
    "      }\n",
    "    }\n",
    "  }\n",
    "}\"\"\"\n",
    "query"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n    {\\n    \"chrom\":\"4\",\\n    \"datasetId\":\"gnomad_r3\",\\n    \"referenceGenome\":\"GRCh38\",\\n    \"start\":1002741,\\n    \"stop\":1002771}\\n'"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variables = \"\"\"\n",
    "    {\n",
    "    \"chrom\":\"4\",\n",
    "    \"datasetId\":\"gnomad_r3\",\n",
    "    \"referenceGenome\":\"GRCh38\",\n",
    "    \"start\":1002741,\n",
    "    \"stop\":1002771}\n",
    "\"\"\"\n",
    "variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': {'region': {'clinvar_variants': [{'clinical_significance': ['Pathogenic'],\n",
       "     'clinvar_variation_id': '11908',\n",
       "     'gold_stars': 2,\n",
       "     'major_consequence': 'stop_gained',\n",
       "     'pos': 1002747,\n",
       "     'variant_id': '4-1002747-G-A'},\n",
       "    {'clinical_significance': ['Pathogenic'],\n",
       "     'clinvar_variation_id': '555230',\n",
       "     'gold_stars': 1,\n",
       "     'major_consequence': 'stop_gained',\n",
       "     'pos': 1002748,\n",
       "     'variant_id': '4-1002748-G-A'},\n",
       "    {'clinical_significance': ['Pathogenic'],\n",
       "     'clinvar_variation_id': '552095',\n",
       "     'gold_stars': 2,\n",
       "     'major_consequence': 'stop_gained',\n",
       "     'pos': 1002752,\n",
       "     'variant_id': '4-1002752-G-T'},\n",
       "    {'clinical_significance': ['Likely_benign'],\n",
       "     'clinvar_variation_id': '724507',\n",
       "     'gold_stars': 1,\n",
       "     'major_consequence': 'synonymous_variant',\n",
       "     'pos': 1002766,\n",
       "     'variant_id': '4-1002766-C-T'},\n",
       "    {'clinical_significance': ['Conflicting_interpretations_of_pathogenicity'],\n",
       "     'clinvar_variation_id': '92627',\n",
       "     'gold_stars': 1,\n",
       "     'major_consequence': 'missense_variant',\n",
       "     'pos': 1002767,\n",
       "     'variant_id': '4-1002767-G-C'},\n",
       "    {'clinical_significance': ['Uncertain_significance'],\n",
       "     'clinvar_variation_id': '557616',\n",
       "     'gold_stars': 1,\n",
       "     'major_consequence': 'missense_variant',\n",
       "     'pos': 1002768,\n",
       "     'variant_id': '4-1002768-G-A'},\n",
       "    {'clinical_significance': ['Uncertain_significance'],\n",
       "     'clinvar_variation_id': '593572',\n",
       "     'gold_stars': 1,\n",
       "     'major_consequence': 'inframe_deletion',\n",
       "     'pos': 1002771,\n",
       "     'variant_id': '4-1002771-CCGTCCTGGACAGCAACCACACGGTGGG-C'}],\n",
       "   'variants': [{'consequence': 'stop_gained',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Trp402Ter',\n",
       "     'hgvsc': 'c.1205G>A',\n",
       "     'hgvsp': 'p.Trp402Ter',\n",
       "     'lof': 'HC',\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002747,\n",
       "     'rsid': 'rs121965019',\n",
       "     'variant_id': '4-1002747-G-A',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 120,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 142464,\n",
       "      'af': 0.0008423180592991914,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 13,\n",
       "        'an': 41628,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 898, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 3, 'an': 13600, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3316, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3124, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 3, 'an': 10322, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 100, 'an': 64398, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 1, 'an': 2142, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3036, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'missense_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Ala403Val',\n",
       "     'hgvsc': 'c.1208C>T',\n",
       "     'hgvsp': 'p.Ala403Val',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002750,\n",
       "     'rsid': 'rs1253435204',\n",
       "     'variant_id': '4-1002750-C-T',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 2,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 141870,\n",
       "      'af': 1.4097413124691619e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 2,\n",
       "        'an': 41564,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 888, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13556, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3308, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3104, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10206, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 64072, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2142, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3030, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'synonymous_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Ala403Ala',\n",
       "     'hgvsc': 'c.1209C>G',\n",
       "     'hgvsp': 'p.Ala403Ala',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002751,\n",
       "     'rsid': 'rs990733414',\n",
       "     'variant_id': '4-1002751-C-G',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 2,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 143004,\n",
       "      'af': 1.3985622779782384e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 41944,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 898, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13636, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3322, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3132, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10426, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 2, 'an': 64450, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2152, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3044, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'missense_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Val405Gly',\n",
       "     'hgvsc': 'c.1214T>G',\n",
       "     'hgvsp': 'p.Val405Gly',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002756,\n",
       "     'rsid': None,\n",
       "     'variant_id': '4-1002756-T-G',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 1,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 141746,\n",
       "      'af': 7.054872800643404e-06,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 41420,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 890, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13586, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3302, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3110, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10210, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 1, 'an': 64046, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2142, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3040, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'missense_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Ser406Trp',\n",
       "     'hgvsc': 'c.1217C>G',\n",
       "     'hgvsp': 'p.Ser406Trp',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002759,\n",
       "     'rsid': None,\n",
       "     'variant_id': '4-1002759-C-G',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 2,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 142602,\n",
       "      'af': 1.402504873704436e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 41804,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 890, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13624, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3320, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3120, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10320, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 2, 'an': 64332, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2152, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3040, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'synonymous_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Ala408Ala',\n",
       "     'hgvsc': 'c.1224C>T',\n",
       "     'hgvsp': 'p.Ala408Ala',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002766,\n",
       "     'rsid': None,\n",
       "     'variant_id': '4-1002766-C-T',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 3,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 143012,\n",
       "      'af': 2.097726064945599e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 41976,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 898, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13648, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3324, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3126, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10416, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 3, 'an': 64432, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2154, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3038, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'missense_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Gly409Arg',\n",
       "     'hgvsc': 'c.1225G>C',\n",
       "     'hgvsp': 'p.Gly409Arg',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002767,\n",
       "     'rsid': None,\n",
       "     'variant_id': '4-1002767-G-C',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 2750,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 67,\n",
       "      'an': 142238,\n",
       "      'af': 0.019333792657377072,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 2446,\n",
       "        'an': 41570,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 65},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 894, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 137, 'an': 13612, 'ac_hemi': 0, 'ac_hom': 2},\n",
       "       {'id': 'ASJ', 'ac': 30, 'an': 3314, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 58, 'an': 3124, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10270, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 30, 'an': 64284, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 47, 'an': 2144, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 2, 'an': 3026, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'missense_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Gly409Glu',\n",
       "     'hgvsc': 'c.1226G>A',\n",
       "     'hgvsp': 'p.Gly409Glu',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002768,\n",
       "     'rsid': 'rs1465083754',\n",
       "     'variant_id': '4-1002768-G-A',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 2,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 143002,\n",
       "      'af': 1.3985818380162515e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 41964,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 900, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13644, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3322, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3130, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10420, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 2, 'an': 64430, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2154, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3038, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'missense_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Thr410Pro',\n",
       "     'hgvsc': 'c.1228A>C',\n",
       "     'hgvsp': 'p.Thr410Pro',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002770,\n",
       "     'rsid': 'rs978350437',\n",
       "     'variant_id': '4-1002770-A-C',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 5,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 140858,\n",
       "      'af': 3.549674139913956e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 5,\n",
       "        'an': 41066,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 888, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13538, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3292, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3094, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10000, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 63854, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2114, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3012, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'frameshift_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Val411SerfsTer29',\n",
       "     'hgvsc': 'c.1230del',\n",
       "     'hgvsp': 'p.Val411SerfsTer29',\n",
       "     'lof': 'HC',\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002770,\n",
       "     'rsid': None,\n",
       "     'variant_id': '4-1002770-AC-A',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 1,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 140860,\n",
       "      'af': 7.099247479767145e-06,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 1,\n",
       "        'an': 41066,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 888, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13538, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3292, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3094, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10000, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 63856, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2114, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3012, 'ac_hemi': 0, 'ac_hom': 0}]}}]}}}"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "end_point = \"https://gnomad.broadinstitute.org/api/\"\n",
    "response = requests.post(end_point, data={'query': query, 'variables': variables}, timeout=None)\n",
    "response.json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "inRegionQuery = \"\"\"query VariantInRegion($chrom: String!, $start: Int!, $stop: Int!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {\n",
    "  region(start: $start, stop: $stop, chrom: $chrom, reference_genome: $referenceGenome) {\n",
    "    clinvar_variants {\n",
    "      clinical_significance\n",
    "      clinvar_variation_id\n",
    "      gold_stars\n",
    "      major_consequence\n",
    "      pos\n",
    "      variant_id\n",
    "    }\n",
    "    variants(dataset: $datasetId) {\n",
    "      consequence\n",
    "      flags\n",
    "      gene_id\n",
    "      gene_symbol\n",
    "      hgvs\n",
    "      hgvsc\n",
    "      hgvsp\n",
    "      lof\n",
    "      lof_filter\n",
    "      lof_flags\n",
    "      pos\n",
    "      rsid\n",
    "      variant_id: variantId\n",
    "      exome {\n",
    "        ac\n",
    "        ac_hemi\n",
    "        ac_hom\n",
    "        an\n",
    "        af\n",
    "        filters\n",
    "        populations {\n",
    "          id\n",
    "          ac\n",
    "          an\n",
    "          ac_hemi\n",
    "          ac_hom\n",
    "        }\n",
    "      }\n",
    "      genome {\n",
    "        ac\n",
    "        ac_hemi\n",
    "        ac_hom\n",
    "        an\n",
    "        af\n",
    "        filters\n",
    "        populations {\n",
    "          id\n",
    "          ac\n",
    "          an\n",
    "          ac_hemi\n",
    "          ac_hom\n",
    "        }\n",
    "      }\n",
    "    }\n",
    "  }\n",
    "}\"\"\"\n",
    "    \n",
    "regionCoverageQuery = \"\"\"query RegionCoverage($chrom: String!, $start: Int!, $stop: Int!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!, $includeExomeCoverage: Boolean!, $includeGenomeCoverage: Boolean!) {\n",
    "  region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {\n",
    "    exome_coverage(dataset: $datasetId) @include(if: $includeExomeCoverage) {\n",
    "      pos\n",
    "      mean\n",
    "      median\n",
    "      over_1\n",
    "      over_5\n",
    "      over_10\n",
    "      over_15\n",
    "      over_20\n",
    "      over_25\n",
    "      over_30\n",
    "      over_50\n",
    "      over_100\n",
    "    }\n",
    "    genome_coverage(dataset: $datasetId) @include(if: $includeGenomeCoverage) {\n",
    "      pos\n",
    "      mean\n",
    "      median\n",
    "      over_1\n",
    "      over_5\n",
    "      over_10\n",
    "      over_15\n",
    "      over_20\n",
    "      over_25\n",
    "      over_30\n",
    "      over_50\n",
    "      over_100\n",
    "    }\n",
    "  }\n",
    "}\"\"\"\n",
    "    \n",
    "fetchRegionQuery = \"\"\"query FetchRegion($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!) {\n",
    "    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {\n",
    "      reference_genome\n",
    "      chrom\n",
    "      start\n",
    "      stop\n",
    "      genes {\n",
    "        gene_id\n",
    "        symbol\n",
    "        start\n",
    "        stop\n",
    "        exons {\n",
    "          feature_type\n",
    "          start\n",
    "          stop\n",
    "        }\n",
    "      }\n",
    "    }\n",
    "  }\"\"\"\n",
    "\n",
    "class VariantsSearch:\n",
    "    \n",
    "    def __init__(self, chromossome, dataset, start, end):\n",
    "        self.inRegion = inRegionQuery\n",
    "        self.regionCoverage = regionCoverageQuery\n",
    "        self.fetchRegion = fetchRegionQuery\n",
    "        \n",
    "        self.chrom = chromossome\n",
    "        self.dataset = dataset\n",
    "        self.start = start\n",
    "        self.end = end\n",
    "        \n",
    "    \n",
    "    @classmethod\n",
    "    def requestGnomad(self, query, variables):\n",
    "        end_point = \"https://gnomad.broadinstitute.org/api/\"\n",
    "        response = requests.post(end_point, data={'query': query, 'variables': variables}, timeout=None)\n",
    "        return response.json()\n",
    "    \n",
    "    \n",
    "    def getInRegion(self):\n",
    "        \n",
    "        variables = \"\"\"\n",
    "            {\n",
    "                \"chrom\":\"%s\",\n",
    "                \"datasetId\":\"%s\",\n",
    "                \"referenceGenome\":\"GRCh38\",\n",
    "                \"start\":%s,\n",
    "                \"stop\":%s\n",
    "            }\"\"\" % (self.chrom, self.dataset, self.start, self.end)\n",
    "        \n",
    "        answer = self.requestGnomad(self.inRegion, variables)\n",
    "        return answer\n",
    "    \n",
    "    \n",
    "    def getRegionCoverage(self):\n",
    "        \n",
    "        variables = \"\"\"\n",
    "            {\n",
    "                \"chrom\":\"%s\",\n",
    "                \"datasetId\":\"%s\",\n",
    "                \"includeExomeCoverage\":false,\n",
    "                \"includeGenomeCoverage\":true,\n",
    "                \"referenceGenome\":\"GRCh38\",\n",
    "                \"start\":%s,\n",
    "                \"stop\":%s\n",
    "            }\"\"\" % (self.chrom, self.dataset, self.start, self.end)\n",
    "        \n",
    "        answer = self.requestGnomad(self.regionCoverage, variables)\n",
    "        return answer\n",
    "    \n",
    "    \n",
    "    def getFetchRegion(self):\n",
    "        \n",
    "        variables = \"\"\"\n",
    "            {\n",
    "                \"chrom\":\"%s\",\n",
    "                \"datasetId\":\"%s\",\n",
    "                \"referenceGenome\":\"GRCh38\",\n",
    "                \"start\":%s,\n",
    "                \"stop\":%s\n",
    "            }\"\"\" % (self.chrom, self.dataset, self.start, self.end)\n",
    "        \n",
    "        answer = self.requestGnomad(self.fetchRegion, variables)\n",
    "        return answer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': {'region': {'clinvar_variants': [{'clinical_significance': ['Pathogenic'],\n",
       "     'clinvar_variation_id': '11908',\n",
       "     'gold_stars': 2,\n",
       "     'major_consequence': 'stop_gained',\n",
       "     'pos': 1002747,\n",
       "     'variant_id': '4-1002747-G-A'},\n",
       "    {'clinical_significance': ['Pathogenic'],\n",
       "     'clinvar_variation_id': '555230',\n",
       "     'gold_stars': 1,\n",
       "     'major_consequence': 'stop_gained',\n",
       "     'pos': 1002748,\n",
       "     'variant_id': '4-1002748-G-A'},\n",
       "    {'clinical_significance': ['Pathogenic'],\n",
       "     'clinvar_variation_id': '552095',\n",
       "     'gold_stars': 2,\n",
       "     'major_consequence': 'stop_gained',\n",
       "     'pos': 1002752,\n",
       "     'variant_id': '4-1002752-G-T'},\n",
       "    {'clinical_significance': ['Likely_benign'],\n",
       "     'clinvar_variation_id': '724507',\n",
       "     'gold_stars': 1,\n",
       "     'major_consequence': 'synonymous_variant',\n",
       "     'pos': 1002766,\n",
       "     'variant_id': '4-1002766-C-T'},\n",
       "    {'clinical_significance': ['Conflicting_interpretations_of_pathogenicity'],\n",
       "     'clinvar_variation_id': '92627',\n",
       "     'gold_stars': 1,\n",
       "     'major_consequence': 'missense_variant',\n",
       "     'pos': 1002767,\n",
       "     'variant_id': '4-1002767-G-C'},\n",
       "    {'clinical_significance': ['Uncertain_significance'],\n",
       "     'clinvar_variation_id': '557616',\n",
       "     'gold_stars': 1,\n",
       "     'major_consequence': 'missense_variant',\n",
       "     'pos': 1002768,\n",
       "     'variant_id': '4-1002768-G-A'},\n",
       "    {'clinical_significance': ['Uncertain_significance'],\n",
       "     'clinvar_variation_id': '593572',\n",
       "     'gold_stars': 1,\n",
       "     'major_consequence': 'inframe_deletion',\n",
       "     'pos': 1002771,\n",
       "     'variant_id': '4-1002771-CCGTCCTGGACAGCAACCACACGGTGGG-C'}],\n",
       "   'variants': [{'consequence': 'stop_gained',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Trp402Ter',\n",
       "     'hgvsc': 'c.1205G>A',\n",
       "     'hgvsp': 'p.Trp402Ter',\n",
       "     'lof': 'HC',\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002747,\n",
       "     'rsid': 'rs121965019',\n",
       "     'variant_id': '4-1002747-G-A',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 120,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 142464,\n",
       "      'af': 0.0008423180592991914,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 13,\n",
       "        'an': 41628,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 898, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 3, 'an': 13600, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3316, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3124, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 3, 'an': 10322, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 100, 'an': 64398, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 1, 'an': 2142, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3036, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'missense_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Ala403Val',\n",
       "     'hgvsc': 'c.1208C>T',\n",
       "     'hgvsp': 'p.Ala403Val',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002750,\n",
       "     'rsid': 'rs1253435204',\n",
       "     'variant_id': '4-1002750-C-T',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 2,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 141870,\n",
       "      'af': 1.4097413124691619e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 2,\n",
       "        'an': 41564,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 888, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13556, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3308, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3104, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10206, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 64072, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2142, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3030, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'synonymous_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Ala403Ala',\n",
       "     'hgvsc': 'c.1209C>G',\n",
       "     'hgvsp': 'p.Ala403Ala',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002751,\n",
       "     'rsid': 'rs990733414',\n",
       "     'variant_id': '4-1002751-C-G',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 2,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 143004,\n",
       "      'af': 1.3985622779782384e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 41944,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 898, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13636, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3322, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3132, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10426, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 2, 'an': 64450, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2152, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3044, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'missense_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Val405Gly',\n",
       "     'hgvsc': 'c.1214T>G',\n",
       "     'hgvsp': 'p.Val405Gly',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002756,\n",
       "     'rsid': None,\n",
       "     'variant_id': '4-1002756-T-G',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 1,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 141746,\n",
       "      'af': 7.054872800643404e-06,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 41420,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 890, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13586, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3302, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3110, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10210, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 1, 'an': 64046, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2142, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3040, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'missense_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Ser406Trp',\n",
       "     'hgvsc': 'c.1217C>G',\n",
       "     'hgvsp': 'p.Ser406Trp',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002759,\n",
       "     'rsid': None,\n",
       "     'variant_id': '4-1002759-C-G',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 2,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 142602,\n",
       "      'af': 1.402504873704436e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 41804,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 890, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13624, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3320, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3120, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10320, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 2, 'an': 64332, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2152, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3040, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'synonymous_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Ala408Ala',\n",
       "     'hgvsc': 'c.1224C>T',\n",
       "     'hgvsp': 'p.Ala408Ala',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002766,\n",
       "     'rsid': None,\n",
       "     'variant_id': '4-1002766-C-T',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 3,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 143012,\n",
       "      'af': 2.097726064945599e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 41976,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 898, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13648, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3324, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3126, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10416, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 3, 'an': 64432, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2154, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3038, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'missense_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Gly409Arg',\n",
       "     'hgvsc': 'c.1225G>C',\n",
       "     'hgvsp': 'p.Gly409Arg',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002767,\n",
       "     'rsid': None,\n",
       "     'variant_id': '4-1002767-G-C',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 2750,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 67,\n",
       "      'an': 142238,\n",
       "      'af': 0.019333792657377072,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 2446,\n",
       "        'an': 41570,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 65},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 894, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 137, 'an': 13612, 'ac_hemi': 0, 'ac_hom': 2},\n",
       "       {'id': 'ASJ', 'ac': 30, 'an': 3314, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 58, 'an': 3124, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10270, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 30, 'an': 64284, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 47, 'an': 2144, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 2, 'an': 3026, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'missense_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Gly409Glu',\n",
       "     'hgvsc': 'c.1226G>A',\n",
       "     'hgvsp': 'p.Gly409Glu',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002768,\n",
       "     'rsid': 'rs1465083754',\n",
       "     'variant_id': '4-1002768-G-A',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 2,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 143002,\n",
       "      'af': 1.3985818380162515e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 0,\n",
       "        'an': 41964,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 900, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13644, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3322, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3130, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10420, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 2, 'an': 64430, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2154, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3038, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'missense_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Thr410Pro',\n",
       "     'hgvsc': 'c.1228A>C',\n",
       "     'hgvsp': 'p.Thr410Pro',\n",
       "     'lof': None,\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002770,\n",
       "     'rsid': 'rs978350437',\n",
       "     'variant_id': '4-1002770-A-C',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 5,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 140858,\n",
       "      'af': 3.549674139913956e-05,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 5,\n",
       "        'an': 41066,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 888, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13538, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3292, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3094, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10000, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 63854, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2114, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3012, 'ac_hemi': 0, 'ac_hom': 0}]}},\n",
       "    {'consequence': 'frameshift_variant',\n",
       "     'flags': [],\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Val411SerfsTer29',\n",
       "     'hgvsc': 'c.1230del',\n",
       "     'hgvsp': 'p.Val411SerfsTer29',\n",
       "     'lof': 'HC',\n",
       "     'lof_filter': None,\n",
       "     'lof_flags': None,\n",
       "     'pos': 1002770,\n",
       "     'rsid': None,\n",
       "     'variant_id': '4-1002770-AC-A',\n",
       "     'exome': None,\n",
       "     'genome': {'ac': 1,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0,\n",
       "      'an': 140860,\n",
       "      'af': 7.099247479767145e-06,\n",
       "      'filters': [],\n",
       "      'populations': [{'id': 'AFR',\n",
       "        'ac': 1,\n",
       "        'an': 41066,\n",
       "        'ac_hemi': 0,\n",
       "        'ac_hom': 0},\n",
       "       {'id': 'AMI', 'ac': 0, 'an': 888, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'AMR', 'ac': 0, 'an': 13538, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'ASJ', 'ac': 0, 'an': 3292, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'EAS', 'ac': 0, 'an': 3094, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'FIN', 'ac': 0, 'an': 10000, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'NFE', 'ac': 0, 'an': 63856, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'OTH', 'ac': 0, 'an': 2114, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "       {'id': 'SAS', 'ac': 0, 'an': 3012, 'ac_hemi': 0, 'ac_hom': 0}]}}]}}}"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vs = VariantsSearch(\"4\", \"gnomad_r3\", \"1002741\", \"1002771\")\n",
    "vs.getInRegion()\n",
    "#vs.getRegionCoverage()\n",
    "#vs.getFetchRegion()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#gene posio variante frequncia(allele frequency)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
