{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Region Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Source</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Trp402Ter</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>[]</td>\n",
       "      <td>130</td>\n",
       "      <td>151238</td>\n",
       "      <td>0.000860</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002747</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-1002750-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Val</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>150266</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002750</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-1002751-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>151840</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002751</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-1002756-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val405Gly</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>150082</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002756</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-1002759-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ser406Trp</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>151328</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002759</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4-1002766-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala408Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>151864</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002766</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4-1002767-G-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Arg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2757</td>\n",
       "      <td>150922</td>\n",
       "      <td>0.018268</td>\n",
       "      <td>68</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002767</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4-1002768-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Glu</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>151856</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4-1002770-AC-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val411SerfsTer29</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>148668</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4-1002770-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Thr410Pro</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>148666</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variant ID  Gene         Consequence          Annotation Flags  \\\n",
       "0   4-1002747-G-A  IDUA         p.Trp402Ter         stop_gained    []   \n",
       "1   4-1002750-C-T  IDUA         p.Ala403Val    missense_variant    []   \n",
       "2   4-1002751-C-G  IDUA         p.Ala403Ala  synonymous_variant    []   \n",
       "3   4-1002756-T-G  IDUA         p.Val405Gly    missense_variant    []   \n",
       "4   4-1002759-C-G  IDUA         p.Ser406Trp    missense_variant    []   \n",
       "5   4-1002766-C-T  IDUA         p.Ala408Ala  synonymous_variant    []   \n",
       "6   4-1002767-G-C  IDUA         p.Gly409Arg    missense_variant    []   \n",
       "7   4-1002768-G-A  IDUA         p.Gly409Glu    missense_variant    []   \n",
       "8  4-1002770-AC-A  IDUA  p.Val411SerfsTer29  frameshift_variant    []   \n",
       "9   4-1002770-A-C  IDUA         p.Thr410Pro    missense_variant    []   \n",
       "\n",
       "   Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0           130         151238          0.000860                      0   \n",
       "1             2         150266          0.000013                      0   \n",
       "2             2         151840          0.000013                      0   \n",
       "3             1         150082          0.000007                      0   \n",
       "4             3         151328          0.000020                      0   \n",
       "5             3         151864          0.000020                      0   \n",
       "6          2757         150922          0.018268                     68   \n",
       "7             2         151856          0.000013                      0   \n",
       "8             1         148668          0.000007                      0   \n",
       "9             5         148666          0.000034                      0   \n",
       "\n",
       "   Source Chromosome Location Reference Alternative  \n",
       "0  Genome          4  1002747         G           A  \n",
       "1  Genome          4  1002750         C           T  \n",
       "2  Genome          4  1002751         C           G  \n",
       "3  Genome          4  1002756         T           G  \n",
       "4  Genome          4  1002759         C           G  \n",
       "5  Genome          4  1002766         C           T  \n",
       "6  Genome          4  1002767         G           C  \n",
       "7  Genome          4  1002768         G           A  \n",
       "8  Genome          4  1002770        AC           A  \n",
       "9  Genome          4  1002770         A           C  "
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynoma import RegionSearch\n",
    "rs = RegionSearch(3, 4, 1002741, 1002771)\n",
    "df, df_clinical = rs.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_clinical"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Source</th>\n",
       "      <th>...</th>\n",
       "      <th>African</th>\n",
       "      <th>Amish</th>\n",
       "      <th>Latino</th>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <th>East Asian</th>\n",
       "      <th>European (Finnish)</th>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <th>Other</th>\n",
       "      <th>South Asian</th>\n",
       "      <th>Middle Eastern</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Trp402Ter</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>[]</td>\n",
       "      <td>130</td>\n",
       "      <td>151238</td>\n",
       "      <td>0.000860</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>3.168877e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.971868e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.865877e-04</td>\n",
       "      <td>1.621128e-03</td>\n",
       "      <td>4.803074e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-1002750-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Val</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>150266</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>4.895721e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-1002751-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>151840</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.944988e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-1002756-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val405Gly</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>150082</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.482756e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-1002759-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ser406Trp</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>151328</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.425693e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4-1002766-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala408Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>151864</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.418783e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4-1002767-G-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Arg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2757</td>\n",
       "      <td>150922</td>\n",
       "      <td>0.018268</td>\n",
       "      <td>68</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>5.894567e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.057680e-02</td>\n",
       "      <td>9.243212e-03</td>\n",
       "      <td>1.628089e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.872431e-04</td>\n",
       "      <td>1.294343e-02</td>\n",
       "      <td>1.048218e-03</td>\n",
       "      <td>9.740260e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4-1002768-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Glu</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>151856</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.945768e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4-1002770-AC-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val411SerfsTer29</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>148668</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>2.484596e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4-1002770-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Thr410Pro</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>148666</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>1.242298e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variant ID  Gene         Consequence          Annotation Flags  \\\n",
       "0   4-1002747-G-A  IDUA         p.Trp402Ter         stop_gained    []   \n",
       "1   4-1002750-C-T  IDUA         p.Ala403Val    missense_variant    []   \n",
       "2   4-1002751-C-G  IDUA         p.Ala403Ala  synonymous_variant    []   \n",
       "3   4-1002756-T-G  IDUA         p.Val405Gly    missense_variant    []   \n",
       "4   4-1002759-C-G  IDUA         p.Ser406Trp    missense_variant    []   \n",
       "5   4-1002766-C-T  IDUA         p.Ala408Ala  synonymous_variant    []   \n",
       "6   4-1002767-G-C  IDUA         p.Gly409Arg    missense_variant    []   \n",
       "7   4-1002768-G-A  IDUA         p.Gly409Glu    missense_variant    []   \n",
       "8  4-1002770-AC-A  IDUA  p.Val411SerfsTer29  frameshift_variant    []   \n",
       "9   4-1002770-A-C  IDUA         p.Thr410Pro    missense_variant    []   \n",
       "\n",
       "   Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0           130         151238          0.000860                      0   \n",
       "1             2         150266          0.000013                      0   \n",
       "2             2         151840          0.000013                      0   \n",
       "3             1         150082          0.000007                      0   \n",
       "4             3         151328          0.000020                      0   \n",
       "5             3         151864          0.000020                      0   \n",
       "6          2757         150922          0.018268                     68   \n",
       "7             2         151856          0.000013                      0   \n",
       "8             1         148668          0.000007                      0   \n",
       "9             5         148666          0.000034                      0   \n",
       "\n",
       "   Source  ...       African         Amish        Latino Ashkenazi Jewish  \\\n",
       "0  Genome  ...  3.168877e-04  0.000000e+00  1.971868e-04     0.000000e+00   \n",
       "1  Genome  ...  4.895721e-05  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "2  Genome  ...  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "3  Genome  ...  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "4  Genome  ...  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "5  Genome  ...  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "6  Genome  ...  5.894567e-02  0.000000e+00  1.057680e-02     9.243212e-03   \n",
       "7  Genome  ...  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "8  Genome  ...  2.484596e-05  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "9  Genome  ...  1.242298e-04  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "\n",
       "     East Asian European (Finnish) European (non-Finnish)         Other  \\\n",
       "0  0.000000e+00       2.865877e-04           1.621128e-03  4.803074e-04   \n",
       "1  0.000000e+00       0.000000e+00           0.000000e+00  0.000000e+00   \n",
       "2  0.000000e+00       0.000000e+00           2.944988e-05  0.000000e+00   \n",
       "3  0.000000e+00       0.000000e+00           1.482756e-05  0.000000e+00   \n",
       "4  0.000000e+00       0.000000e+00           4.425693e-05  0.000000e+00   \n",
       "5  0.000000e+00       0.000000e+00           4.418783e-05  0.000000e+00   \n",
       "6  1.628089e-02       0.000000e+00           4.872431e-04  1.294343e-02   \n",
       "7  0.000000e+00       0.000000e+00           2.945768e-05  0.000000e+00   \n",
       "8  0.000000e+00       0.000000e+00           0.000000e+00  0.000000e+00   \n",
       "9  0.000000e+00       0.000000e+00           0.000000e+00  0.000000e+00   \n",
       "\n",
       "    South Asian Middle Eastern  \n",
       "0  0.000000e+00   0.000000e+00  \n",
       "1  0.000000e+00   0.000000e+00  \n",
       "2  0.000000e+00   0.000000e+00  \n",
       "3  0.000000e+00   0.000000e+00  \n",
       "4  0.000000e+00   0.000000e+00  \n",
       "5  0.000000e+00   0.000000e+00  \n",
       "6  1.048218e-03   9.740260e-03  \n",
       "7  0.000000e+00   0.000000e+00  \n",
       "8  0.000000e+00   0.000000e+00  \n",
       "9  0.000000e+00   0.000000e+00  \n",
       "\n",
       "[10 rows x 24 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal, _ = rs.get_data(additional_population_info=True)\n",
    "normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Source</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "      <th>African</th>\n",
       "      <th>Amish</th>\n",
       "      <th>Latino</th>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <th>East Asian</th>\n",
       "      <th>European (Finnish)</th>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <th>Other</th>\n",
       "      <th>South Asian</th>\n",
       "      <th>Middle Eastern</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>130</td>\n",
       "      <td>151238</td>\n",
       "      <td>0.000860</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002747</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3.168877e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.971868e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.865877e-04</td>\n",
       "      <td>1.621128e-03</td>\n",
       "      <td>4.803074e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>150266</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002750</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4.895721e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>151840</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002751</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.944988e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>150082</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002756</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.482756e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>151328</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002759</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.425693e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3</td>\n",
       "      <td>151864</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002766</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.418783e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2757</td>\n",
       "      <td>150922</td>\n",
       "      <td>0.018268</td>\n",
       "      <td>68</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002767</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>5.894567e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.057680e-02</td>\n",
       "      <td>9.243212e-03</td>\n",
       "      <td>1.628089e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.872431e-04</td>\n",
       "      <td>1.294343e-02</td>\n",
       "      <td>1.048218e-03</td>\n",
       "      <td>9.740260e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>151856</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.945768e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>148668</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>2.484596e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5</td>\n",
       "      <td>148666</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1.242298e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0           130         151238          0.000860                      0   \n",
       "1             2         150266          0.000013                      0   \n",
       "2             2         151840          0.000013                      0   \n",
       "3             1         150082          0.000007                      0   \n",
       "4             3         151328          0.000020                      0   \n",
       "5             3         151864          0.000020                      0   \n",
       "6          2757         150922          0.018268                     68   \n",
       "7             2         151856          0.000013                      0   \n",
       "8             1         148668          0.000007                      0   \n",
       "9             5         148666          0.000034                      0   \n",
       "\n",
       "   Source Chromosome Location Reference Alternative       African  \\\n",
       "0  Genome          4  1002747         G           A  3.168877e-04   \n",
       "1  Genome          4  1002750         C           T  4.895721e-05   \n",
       "2  Genome          4  1002751         C           G  0.000000e+00   \n",
       "3  Genome          4  1002756         T           G  0.000000e+00   \n",
       "4  Genome          4  1002759         C           G  0.000000e+00   \n",
       "5  Genome          4  1002766         C           T  0.000000e+00   \n",
       "6  Genome          4  1002767         G           C  5.894567e-02   \n",
       "7  Genome          4  1002768         G           A  0.000000e+00   \n",
       "8  Genome          4  1002770        AC           A  2.484596e-05   \n",
       "9  Genome          4  1002770         A           C  1.242298e-04   \n",
       "\n",
       "          Amish        Latino Ashkenazi Jewish    East Asian  \\\n",
       "0  0.000000e+00  1.971868e-04     0.000000e+00  0.000000e+00   \n",
       "1  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "2  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "3  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "4  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "5  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "6  0.000000e+00  1.057680e-02     9.243212e-03  1.628089e-02   \n",
       "7  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "8  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "9  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "\n",
       "  European (Finnish) European (non-Finnish)         Other   South Asian  \\\n",
       "0       2.865877e-04           1.621128e-03  4.803074e-04  0.000000e+00   \n",
       "1       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "2       0.000000e+00           2.944988e-05  0.000000e+00  0.000000e+00   \n",
       "3       0.000000e+00           1.482756e-05  0.000000e+00  0.000000e+00   \n",
       "4       0.000000e+00           4.425693e-05  0.000000e+00  0.000000e+00   \n",
       "5       0.000000e+00           4.418783e-05  0.000000e+00  0.000000e+00   \n",
       "6       0.000000e+00           4.872431e-04  1.294343e-02  1.048218e-03   \n",
       "7       0.000000e+00           2.945768e-05  0.000000e+00  0.000000e+00   \n",
       "8       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "9       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "\n",
       "  Middle Eastern  \n",
       "0   0.000000e+00  \n",
       "1   0.000000e+00  \n",
       "2   0.000000e+00  \n",
       "3   0.000000e+00  \n",
       "4   0.000000e+00  \n",
       "5   0.000000e+00  \n",
       "6   9.740260e-03  \n",
       "7   0.000000e+00  \n",
       "8   0.000000e+00  \n",
       "9   0.000000e+00  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal.iloc[:, 5:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Number of Hemizygotes</th>\n",
       "      <th>Source</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>X-1002741-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>1</td>\n",
       "      <td>31378</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>X</td>\n",
       "      <td>1002741</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>X-1002742-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>2</td>\n",
       "      <td>31316</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>X</td>\n",
       "      <td>1002742</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>X-1002746-TATCA-T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>1</td>\n",
       "      <td>31376</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Genome</td>\n",
       "      <td>X</td>\n",
       "      <td>1002746</td>\n",
       "      <td>TATCA</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>X-1002765-TTATC-T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>9</td>\n",
       "      <td>31374</td>\n",
       "      <td>0.000287</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>Genome</td>\n",
       "      <td>X</td>\n",
       "      <td>1002765</td>\n",
       "      <td>TTATC</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>X-1002767-ATCTATCTT-A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>1</td>\n",
       "      <td>31348</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Genome</td>\n",
       "      <td>X</td>\n",
       "      <td>1002767</td>\n",
       "      <td>ATCTATCTT</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>X-1002771-ATCTT-A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>2</td>\n",
       "      <td>31338</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Genome</td>\n",
       "      <td>X</td>\n",
       "      <td>1002771</td>\n",
       "      <td>ATCTT</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Variant ID  Gene Consequence Annotation  Flags  Allele Count  \\\n",
       "0          X-1002741-C-T  None        None       None  [par]             1   \n",
       "1          X-1002742-G-A  None        None       None  [par]             2   \n",
       "2      X-1002746-TATCA-T  None        None       None  [par]             1   \n",
       "3      X-1002765-TTATC-T  None        None       None  [par]             9   \n",
       "4  X-1002767-ATCTATCTT-A  None        None       None  [par]             1   \n",
       "5      X-1002771-ATCTT-A  None        None       None  [par]             2   \n",
       "\n",
       "   Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0          31378          0.000032                      0   \n",
       "1          31316          0.000064                      0   \n",
       "2          31376          0.000032                      0   \n",
       "3          31374          0.000287                      0   \n",
       "4          31348          0.000032                      0   \n",
       "5          31338          0.000064                      0   \n",
       "\n",
       "   Number of Hemizygotes  Source Chromosome Location  Reference Alternative  \n",
       "0                      0  Genome          X  1002741          C           T  \n",
       "1                      0  Genome          X  1002742          G           A  \n",
       "2                      1  Genome          X  1002746      TATCA           T  \n",
       "3                      4  Genome          X  1002765      TTATC           T  \n",
       "4                      1  Genome          X  1002767  ATCTATCTT           A  \n",
       "5                      1  Genome          X  1002771      ATCTT           A  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rs = RegionSearch(2, 'X', 1002741, 1002771)\n",
    "df, df_clinical = rs.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_clinical"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>flags</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>hgvs</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>lof</th>\n",
       "      <th>lof_filter</th>\n",
       "      <th>lof_flags</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>exome</th>\n",
       "      <th>genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002741</td>\n",
       "      <td>rs782385877</td>\n",
       "      <td>X-1002741-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002742</td>\n",
       "      <td>rs181914603</td>\n",
       "      <td>X-1002742-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002746</td>\n",
       "      <td>None</td>\n",
       "      <td>X-1002746-TATCA-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002765</td>\n",
       "      <td>None</td>\n",
       "      <td>X-1002765-TTATC-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 9, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002767</td>\n",
       "      <td>None</td>\n",
       "      <td>X-1002767-ATCTATCTT-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002771</td>\n",
       "      <td>None</td>\n",
       "      <td>X-1002771-ATCTT-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  consequence  flags gene_id gene_symbol  hgvs hgvsc hgvsp   lof lof_filter  \\\n",
       "0        None  [par]    None        None  None  None  None  None       None   \n",
       "1        None  [par]    None        None  None  None  None  None       None   \n",
       "2        None  [par]    None        None  None  None  None  None       None   \n",
       "3        None  [par]    None        None  None  None  None  None       None   \n",
       "4        None  [par]    None        None  None  None  None  None       None   \n",
       "5        None  [par]    None        None  None  None  None  None       None   \n",
       "\n",
       "  lof_flags      pos         rsid             variant_id exome  \\\n",
       "0      None  1002741  rs782385877          X-1002741-C-T  None   \n",
       "1      None  1002742  rs181914603          X-1002742-G-A  None   \n",
       "2      None  1002746         None      X-1002746-TATCA-T  None   \n",
       "3      None  1002765         None      X-1002765-TTATC-T  None   \n",
       "4      None  1002767         None  X-1002767-ATCTATCTT-A  None   \n",
       "5      None  1002771         None      X-1002771-ATCTT-A  None   \n",
       "\n",
       "                                              genome  \n",
       "0  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "1  {'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "2  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "3  {'ac': 9, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "4  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "5  {'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df, _ = rs.get_data(standard=False)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Source</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-987000-A-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-85A&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>146532</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987000</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-987002-A-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-83A&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>147068</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987002</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-987003-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-82T&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>144336</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987003</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-987004-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-81G&gt;A</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>151744</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987004</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-987005-G-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-80G&gt;T</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>152084</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987005</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5894</th>\n",
       "      <td>4-1004540-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*147C&gt;T</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>61</td>\n",
       "      <td>152232</td>\n",
       "      <td>0.000401</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004540</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5895</th>\n",
       "      <td>4-1004543-T-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*150T&gt;C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>152230</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004543</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5896</th>\n",
       "      <td>4-1004547-G-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*154G&gt;C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>152188</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004547</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5897</th>\n",
       "      <td>4-1004548-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*155C&gt;T</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>97</td>\n",
       "      <td>152206</td>\n",
       "      <td>0.000637</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004548</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5898</th>\n",
       "      <td>4-1004550-G-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*157G&gt;T</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>152180</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004550</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5899 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Variant ID  Gene Consequence           Annotation Flags  \\\n",
       "0      4-987000-A-G  IDUA    c.-85A>G  5_prime_UTR_variant    []   \n",
       "1      4-987002-A-G  IDUA    c.-83A>G  5_prime_UTR_variant    []   \n",
       "2      4-987003-T-G  IDUA    c.-82T>G  5_prime_UTR_variant    []   \n",
       "3      4-987004-G-A  IDUA    c.-81G>A  5_prime_UTR_variant    []   \n",
       "4      4-987005-G-T  IDUA    c.-80G>T  5_prime_UTR_variant    []   \n",
       "...             ...   ...         ...                  ...   ...   \n",
       "5894  4-1004540-C-T  IDUA   c.*147C>T  3_prime_UTR_variant    []   \n",
       "5895  4-1004543-T-C  IDUA   c.*150T>C  3_prime_UTR_variant    []   \n",
       "5896  4-1004547-G-C  IDUA   c.*154G>C  3_prime_UTR_variant    []   \n",
       "5897  4-1004548-C-T  IDUA   c.*155C>T  3_prime_UTR_variant    []   \n",
       "5898  4-1004550-G-T  IDUA   c.*157G>T  3_prime_UTR_variant    []   \n",
       "\n",
       "      Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0                1         146532          0.000007                      0   \n",
       "1                1         147068          0.000007                      0   \n",
       "2                1         144336          0.000007                      0   \n",
       "3                1         151744          0.000007                      0   \n",
       "4                1         152084          0.000007                      0   \n",
       "...            ...            ...               ...                    ...   \n",
       "5894            61         152232          0.000401                      0   \n",
       "5895             1         152230          0.000007                      0   \n",
       "5896             5         152188          0.000033                      0   \n",
       "5897            97         152206          0.000637                      0   \n",
       "5898             1         152180          0.000007                      0   \n",
       "\n",
       "      Source Chromosome Location Reference Alternative  \n",
       "0     Genome          4   987000         A           G  \n",
       "1     Genome          4   987002         A           G  \n",
       "2     Genome          4   987003         T           G  \n",
       "3     Genome          4   987004         G           A  \n",
       "4     Genome          4   987005         G           T  \n",
       "...      ...        ...      ...       ...         ...  \n",
       "5894  Genome          4  1004540         C           T  \n",
       "5895  Genome          4  1004543         T           C  \n",
       "5896  Genome          4  1004547         G           C  \n",
       "5897  Genome          4  1004548         C           T  \n",
       "5898  Genome          4  1004550         G           T  \n",
       "\n",
       "[5899 rows x 14 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynoma import GeneSearch\n",
    "gs = GeneSearch(3, \"idua\")\n",
    "df, _ = gs.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Transcript Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Source</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-987010-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-75C&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>149172</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987010</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-987013-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-72G&gt;A</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>151256</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987013</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-987014-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-71C&gt;T</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>37</td>\n",
       "      <td>152092</td>\n",
       "      <td>0.000243</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987014</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-987021-T-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-64T&gt;C</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>146758</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987021</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-987023-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-62C&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>152126</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987023</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1199</th>\n",
       "      <td>4-1004450-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*57G&gt;A</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>152196</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004450</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1200</th>\n",
       "      <td>4-1004450-G-GCA</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*59_*60dup</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>152196</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004450</td>\n",
       "      <td>G</td>\n",
       "      <td>GCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1201</th>\n",
       "      <td>4-1004452-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*59A&gt;C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>7</td>\n",
       "      <td>152152</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004452</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1202</th>\n",
       "      <td>4-1004458-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*65C&gt;G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>152180</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004458</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1203</th>\n",
       "      <td>4-1004459-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*66C&gt;G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>23</td>\n",
       "      <td>152202</td>\n",
       "      <td>0.000151</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004459</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1204 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Variant ID  Gene   Consequence           Annotation Flags  \\\n",
       "0        4-987010-C-G  IDUA      c.-75C>G  5_prime_UTR_variant    []   \n",
       "1        4-987013-G-A  IDUA      c.-72G>A  5_prime_UTR_variant    []   \n",
       "2        4-987014-C-T  IDUA      c.-71C>T  5_prime_UTR_variant    []   \n",
       "3        4-987021-T-C  IDUA      c.-64T>C  5_prime_UTR_variant    []   \n",
       "4        4-987023-C-G  IDUA      c.-62C>G  5_prime_UTR_variant    []   \n",
       "...               ...   ...           ...                  ...   ...   \n",
       "1199    4-1004450-G-A  IDUA      c.*57G>A  3_prime_UTR_variant    []   \n",
       "1200  4-1004450-G-GCA  IDUA  c.*59_*60dup  3_prime_UTR_variant    []   \n",
       "1201    4-1004452-A-C  IDUA      c.*59A>C  3_prime_UTR_variant    []   \n",
       "1202    4-1004458-C-G  IDUA      c.*65C>G  3_prime_UTR_variant    []   \n",
       "1203    4-1004459-C-G  IDUA      c.*66C>G  3_prime_UTR_variant    []   \n",
       "\n",
       "      Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0                1         149172          0.000007                      0   \n",
       "1                3         151256          0.000020                      0   \n",
       "2               37         152092          0.000243                      0   \n",
       "3                2         146758          0.000014                      0   \n",
       "4                1         152126          0.000007                      0   \n",
       "...            ...            ...               ...                    ...   \n",
       "1199             1         152196          0.000007                      0   \n",
       "1200             1         152196          0.000007                      0   \n",
       "1201             7         152152          0.000046                      0   \n",
       "1202             2         152180          0.000013                      0   \n",
       "1203            23         152202          0.000151                      0   \n",
       "\n",
       "      Source Chromosome Location Reference Alternative  \n",
       "0     Genome          4   987010         C           G  \n",
       "1     Genome          4   987013         G           A  \n",
       "2     Genome          4   987014         C           T  \n",
       "3     Genome          4   987021         T           C  \n",
       "4     Genome          4   987023         C           G  \n",
       "...      ...        ...      ...       ...         ...  \n",
       "1199  Genome          4  1004450         G           A  \n",
       "1200  Genome          4  1004450         G         GCA  \n",
       "1201  Genome          4  1004452         A           C  \n",
       "1202  Genome          4  1004458         C           G  \n",
       "1203  Genome          4  1004459         C           G  \n",
       "\n",
       "[1204 rows x 14 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynoma import TranscriptSearch\n",
    "ts = TranscriptSearch(3, \"ENST00000247933\")\n",
    "df, _ = ts.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Variant Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'hgdp:japanese'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/Documents/pynoma/pynoma/DataManager.py\u001b[0m in \u001b[0;36m__build_variant_search_standard_df\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    227\u001b[0m                 \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 228\u001b[0;31m                     \u001b[0mnew_row_name\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mPOPULATION_ID_MAP\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpiece\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\" \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    229\u001b[0m                 \u001b[0;32mexcept\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'hgdp:japanese'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-13-7fe5e5471c5f>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mpynoma\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mVariantSearch\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mvs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mVariantSearch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'4-1002747-G-A'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0mdf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmeta\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_data\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/pynoma/pynoma/Search.py\u001b[0m in \u001b[0;36mget_data\u001b[0;34m(self, raw)\u001b[0m\n\u001b[1;32m    197\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mjson_data\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    198\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 199\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdm\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mDataManager\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjson_data\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdataset_id\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvariant_search\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    200\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdm\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstandard_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdm\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvariant_metadata\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/pynoma/pynoma/DataManager.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, json_data, gnomad_version, variant_search)\u001b[0m\n\u001b[1;32m     45\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstandard_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     46\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvariant_metadata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m  \u001b[0;31m# Used in variant searches\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 47\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_process_raw_json\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     48\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     49\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/pynoma/pynoma/DataManager.py\u001b[0m in \u001b[0;36m_process_raw_json\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     50\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_process_raw_json\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     51\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvariant_search\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 52\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__process_variant_search_data\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     53\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/pynoma/pynoma/DataManager.py\u001b[0m in \u001b[0;36m__process_variant_search_data\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    209\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    210\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__process_variant_search_data\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 211\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__build_variant_search_standard_df\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    212\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__extract_variant_metadata\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    213\u001b[0m         \u001b[0;32mreturn\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/pynoma/pynoma/DataManager.py\u001b[0m in \u001b[0;36m__build_variant_search_standard_df\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    228\u001b[0m                     \u001b[0mnew_row_name\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mPOPULATION_ID_MAP\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpiece\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\" \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    229\u001b[0m                 \u001b[0;32mexcept\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 230\u001b[0;31m                     \u001b[0mnew_row_name\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mSUBPOPULATION_ID_MAP\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpiece\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\" \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    231\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    232\u001b[0m             \u001b[0mnew_row_name\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnew_row_name\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'hgdp:japanese'"
     ]
    }
   ],
   "source": [
    "from pynoma import VariantSearch\n",
    "vs = VariantSearch(3, '4-1002747-G-A')\n",
    "df, meta = vs.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'variantId': '4-1002747-G-A',\n",
       " 'reference_genome': 'GRCh38',\n",
       " 'chrom': '4',\n",
       " 'pos': 1002747,\n",
       " 'ref': 'G',\n",
       " 'alt': 'A',\n",
       " 'colocatedVariants': ['4-1002747-G-T'],\n",
       " 'multiNucleotideVariants': [],\n",
       " 'exome': None,\n",
       " 'genome': {'ac': 120,\n",
       "  'an': 142464,\n",
       "  'ac_hemi': 0,\n",
       "  'ac_hom': 0,\n",
       "  'faf95': {'popmax': 0.0013063199999999997, 'popmax_population': 'NFE'},\n",
       "  'filters': [],\n",
       "  'age_distribution': None,\n",
       "  'qualityMetrics': {'alleleBalance': {'alt': {'bin_edges': [0,\n",
       "      0.05,\n",
       "      0.1,\n",
       "      0.15,\n",
       "      0.2,\n",
       "      0.25,\n",
       "      0.3,\n",
       "      0.35,\n",
       "      0.4,\n",
       "      0.45,\n",
       "      0.5,\n",
       "      0.55,\n",
       "      0.6,\n",
       "      0.65,\n",
       "      0.7,\n",
       "      0.75,\n",
       "      0.8,\n",
       "      0.85,\n",
       "      0.9,\n",
       "      0.95,\n",
       "      1],\n",
       "     'bin_freq': [0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      3,\n",
       "      3,\n",
       "      8,\n",
       "      26,\n",
       "      16,\n",
       "      26,\n",
       "      17,\n",
       "      13,\n",
       "      5,\n",
       "      3,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'genotypeDepth': {'all': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [11,\n",
       "      67,\n",
       "      641,\n",
       "      3138,\n",
       "      6868,\n",
       "      12475,\n",
       "      21477,\n",
       "      17174,\n",
       "      5825,\n",
       "      1989,\n",
       "      945,\n",
       "      476,\n",
       "      212,\n",
       "      94,\n",
       "      59,\n",
       "      30,\n",
       "      19,\n",
       "      11,\n",
       "      2,\n",
       "      4],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 13},\n",
       "    'alt': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [0,\n",
       "      0,\n",
       "      0,\n",
       "      8,\n",
       "      13,\n",
       "      25,\n",
       "      29,\n",
       "      15,\n",
       "      10,\n",
       "      9,\n",
       "      8,\n",
       "      3,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'genotypeQuality': {'all': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [42,\n",
       "      34,\n",
       "      79,\n",
       "      100,\n",
       "      20247,\n",
       "      14345,\n",
       "      9795,\n",
       "      6614,\n",
       "      5441,\n",
       "      3845,\n",
       "      2497,\n",
       "      2219,\n",
       "      1635,\n",
       "      1101,\n",
       "      926,\n",
       "      674,\n",
       "      461,\n",
       "      380,\n",
       "      262,\n",
       "      833],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0},\n",
       "    'alt': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      1,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      0,\n",
       "      119],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'siteQualityMetrics': [{'metric': 'FS', 'value': 1.1223795673810724},\n",
       "    {'metric': 'InbreedingCoeff', 'value': -0.0008395130824121466},\n",
       "    {'metric': 'MQ', 'value': 59.99216274475224},\n",
       "    {'metric': 'MQRankSum', 'value': 0},\n",
       "    {'metric': 'QD', 'value': 13.505564},\n",
       "    {'metric': 'ReadPosRankSum', 'value': 0.345},\n",
       "    {'metric': 'SiteQuality', 'value': 100738},\n",
       "    {'metric': 'SOR', 'value': 0.804},\n",
       "    {'metric': 'VQSLOD', 'value': None}]}},\n",
       " 'flags': [],\n",
       " 'rsid': 'rs121965019',\n",
       " 'sortedTranscriptConsequences': [{'canonical': True,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'p.Trp402Ter',\n",
       "   'hgvsc': 'c.1205G>A',\n",
       "   'hgvsp': 'p.Trp402Ter',\n",
       "   'lof': 'HC',\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'stop_gained',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000247933',\n",
       "   'transcript_version': '8'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'p.Trp270Ter',\n",
       "   'hgvsc': 'c.809G>A',\n",
       "   'hgvsp': 'p.Trp270Ter',\n",
       "   'lof': 'HC',\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'stop_gained',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000514224',\n",
       "   'transcript_version': '1'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'n.7G>A',\n",
       "   'hgvsc': 'n.7G>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000502829',\n",
       "   'transcript_version': '1'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '12',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'n.1312G>A',\n",
       "   'hgvsc': 'n.1312G>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000514698',\n",
       "   'transcript_version': '5'}]}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'AFR'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/Documents/pynoma/pynoma/DataManager.py\u001b[0m in \u001b[0;36m__build_variant_search_standard_df\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    227\u001b[0m                 \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 228\u001b[0;31m                     \u001b[0mnew_row_name\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mPOPULATION_ID_MAP\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpiece\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\" \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    229\u001b[0m                 \u001b[0;32mexcept\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'AFR'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-14-2bf1a085e449>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mvs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mVariantSearch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'4-980801-G-A'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mdf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmeta\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_data\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/pynoma/pynoma/Search.py\u001b[0m in \u001b[0;36mget_data\u001b[0;34m(self, raw)\u001b[0m\n\u001b[1;32m    197\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mjson_data\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    198\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 199\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdm\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mDataManager\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjson_data\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdataset_id\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvariant_search\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    200\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdm\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstandard_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdm\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvariant_metadata\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/pynoma/pynoma/DataManager.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, json_data, gnomad_version, variant_search)\u001b[0m\n\u001b[1;32m     45\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstandard_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     46\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvariant_metadata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m  \u001b[0;31m# Used in variant searches\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 47\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_process_raw_json\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     48\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     49\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/pynoma/pynoma/DataManager.py\u001b[0m in \u001b[0;36m_process_raw_json\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     50\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_process_raw_json\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     51\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvariant_search\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 52\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__process_variant_search_data\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     53\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     54\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/pynoma/pynoma/DataManager.py\u001b[0m in \u001b[0;36m__process_variant_search_data\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    209\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    210\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__process_variant_search_data\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 211\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__build_variant_search_standard_df\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    212\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__extract_variant_metadata\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    213\u001b[0m         \u001b[0;32mreturn\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/Documents/pynoma/pynoma/DataManager.py\u001b[0m in \u001b[0;36m__build_variant_search_standard_df\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    228\u001b[0m                     \u001b[0mnew_row_name\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mPOPULATION_ID_MAP\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpiece\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\" \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    229\u001b[0m                 \u001b[0;32mexcept\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 230\u001b[0;31m                     \u001b[0mnew_row_name\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mSUBPOPULATION_ID_MAP\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpiece\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m\" \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    231\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    232\u001b[0m             \u001b[0mnew_row_name\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnew_row_name\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'AFR'"
     ]
    }
   ],
   "source": [
    "vs = VariantSearch(2, '4-980801-G-A')\n",
    "df, meta = vs.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
