{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Source</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-987010-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-75C&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>149172</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987010</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-987013-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-72G&gt;A</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>151256</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987013</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-987014-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-71C&gt;T</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>37</td>\n",
       "      <td>152092</td>\n",
       "      <td>0.000243</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987014</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-987021-T-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-64T&gt;C</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>146758</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987021</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-987023-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-62C&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>152126</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987023</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1199</th>\n",
       "      <td>4-1004450-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*57G&gt;A</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>152196</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004450</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1200</th>\n",
       "      <td>4-1004450-G-GCA</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*59_*60dup</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>152196</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004450</td>\n",
       "      <td>G</td>\n",
       "      <td>GCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1201</th>\n",
       "      <td>4-1004452-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*59A&gt;C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>7</td>\n",
       "      <td>152152</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004452</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1202</th>\n",
       "      <td>4-1004458-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*65C&gt;G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>152180</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004458</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1203</th>\n",
       "      <td>4-1004459-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*66C&gt;G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>23</td>\n",
       "      <td>152202</td>\n",
       "      <td>0.000151</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004459</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1204 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Variant ID  Gene   Consequence           Annotation Flags  \\\n",
       "0        4-987010-C-G  IDUA      c.-75C>G  5_prime_UTR_variant    []   \n",
       "1        4-987013-G-A  IDUA      c.-72G>A  5_prime_UTR_variant    []   \n",
       "2        4-987014-C-T  IDUA      c.-71C>T  5_prime_UTR_variant    []   \n",
       "3        4-987021-T-C  IDUA      c.-64T>C  5_prime_UTR_variant    []   \n",
       "4        4-987023-C-G  IDUA      c.-62C>G  5_prime_UTR_variant    []   \n",
       "...               ...   ...           ...                  ...   ...   \n",
       "1199    4-1004450-G-A  IDUA      c.*57G>A  3_prime_UTR_variant    []   \n",
       "1200  4-1004450-G-GCA  IDUA  c.*59_*60dup  3_prime_UTR_variant    []   \n",
       "1201    4-1004452-A-C  IDUA      c.*59A>C  3_prime_UTR_variant    []   \n",
       "1202    4-1004458-C-G  IDUA      c.*65C>G  3_prime_UTR_variant    []   \n",
       "1203    4-1004459-C-G  IDUA      c.*66C>G  3_prime_UTR_variant    []   \n",
       "\n",
       "      Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0                1         149172          0.000007                      0   \n",
       "1                3         151256          0.000020                      0   \n",
       "2               37         152092          0.000243                      0   \n",
       "3                2         146758          0.000014                      0   \n",
       "4                1         152126          0.000007                      0   \n",
       "...            ...            ...               ...                    ...   \n",
       "1199             1         152196          0.000007                      0   \n",
       "1200             1         152196          0.000007                      0   \n",
       "1201             7         152152          0.000046                      0   \n",
       "1202             2         152180          0.000013                      0   \n",
       "1203            23         152202          0.000151                      0   \n",
       "\n",
       "      Source Chromosome Location Reference Alternative  \n",
       "0     Genome          4   987010         C           G  \n",
       "1     Genome          4   987013         G           A  \n",
       "2     Genome          4   987014         C           T  \n",
       "3     Genome          4   987021         T           C  \n",
       "4     Genome          4   987023         C           G  \n",
       "...      ...        ...      ...       ...         ...  \n",
       "1199  Genome          4  1004450         G           A  \n",
       "1200  Genome          4  1004450         G         GCA  \n",
       "1201  Genome          4  1004452         A           C  \n",
       "1202  Genome          4  1004458         C           G  \n",
       "1203  Genome          4  1004459         C           G  \n",
       "\n",
       "[1204 rows x 14 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynoma import GeneSearch\n",
    "gs = GeneSearch(3, \"idua\")\n",
    "df, df_clin = gs.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>clinvar_variation_id</th>\n",
       "      <th>gold_stars</th>\n",
       "      <th>major_consequence</th>\n",
       "      <th>pos</th>\n",
       "      <th>variant_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>350186</td>\n",
       "      <td>1</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>987014</td>\n",
       "      <td>4-987014-C-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>990330</td>\n",
       "      <td>0</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>987083</td>\n",
       "      <td>4-987083-C-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>550458</td>\n",
       "      <td>1</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>987085</td>\n",
       "      <td>4-987085-A-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>639529</td>\n",
       "      <td>1</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>987086</td>\n",
       "      <td>4-987086-T-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>557150</td>\n",
       "      <td>1</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>987087</td>\n",
       "      <td>4-987087-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>404</th>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>558359</td>\n",
       "      <td>1</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1004349</td>\n",
       "      <td>4-1004349-G-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>Likely benign</td>\n",
       "      <td>743489</td>\n",
       "      <td>1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1004381</td>\n",
       "      <td>4-1004381-G-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>Likely benign</td>\n",
       "      <td>754394</td>\n",
       "      <td>1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1004390</td>\n",
       "      <td>4-1004390-A-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>960079</td>\n",
       "      <td>1</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>1004391</td>\n",
       "      <td>4-1004391-T-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>11920</td>\n",
       "      <td>0</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>1004391</td>\n",
       "      <td>4-1004391-T-G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>409 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      clinical_significance clinvar_variation_id  gold_stars  \\\n",
       "0    Uncertain significance               350186           1   \n",
       "1    Uncertain significance               990330           0   \n",
       "2                Pathogenic               550458           1   \n",
       "3                Pathogenic               639529           1   \n",
       "4         Likely pathogenic               557150           1   \n",
       "..                      ...                  ...         ...   \n",
       "404  Uncertain significance               558359           1   \n",
       "405           Likely benign               743489           1   \n",
       "406           Likely benign               754394           1   \n",
       "407              Pathogenic               960079           1   \n",
       "408              Pathogenic                11920           0   \n",
       "\n",
       "       major_consequence      pos     variant_id  \n",
       "0    5_prime_UTR_variant   987014   4-987014-C-T  \n",
       "1    5_prime_UTR_variant   987083   4-987083-C-T  \n",
       "2             start_lost   987085   4-987085-A-C  \n",
       "3             start_lost   987086   4-987086-T-C  \n",
       "4             start_lost   987087   4-987087-G-A  \n",
       "..                   ...      ...            ...  \n",
       "404          stop_gained  1004349  4-1004349-G-T  \n",
       "405   synonymous_variant  1004381  4-1004381-G-T  \n",
       "406   synonymous_variant  1004390  4-1004390-A-T  \n",
       "407            stop_lost  1004391  4-1004391-T-C  \n",
       "408            stop_lost  1004391  4-1004391-T-G  \n",
       "\n",
       "[409 rows x 6 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_clin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>flags</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>hgvs</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>lof</th>\n",
       "      <th>lof_filter</th>\n",
       "      <th>lof_flags</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>exome</th>\n",
       "      <th>genome</th>\n",
       "      <th>lof_curation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-75C&gt;G</td>\n",
       "      <td>c.-75C&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987010</td>\n",
       "      <td>None</td>\n",
       "      <td>ENST00000247933</td>\n",
       "      <td>4-987010-C-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 149...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-72G&gt;A</td>\n",
       "      <td>c.-72G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987013</td>\n",
       "      <td>None</td>\n",
       "      <td>ENST00000247933</td>\n",
       "      <td>4-987013-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 3, 'ac_hemi': 0, 'ac_hom': 0, 'an': 151...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-71C&gt;T</td>\n",
       "      <td>c.-71C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987014</td>\n",
       "      <td>rs886059748</td>\n",
       "      <td>ENST00000247933</td>\n",
       "      <td>4-987014-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 37, 'ac_hemi': 0, 'ac_hom': 0, 'an': 15...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-64T&gt;C</td>\n",
       "      <td>c.-64T&gt;C</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987021</td>\n",
       "      <td>rs1473070711</td>\n",
       "      <td>ENST00000247933</td>\n",
       "      <td>4-987021-T-C</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 146...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-62C&gt;G</td>\n",
       "      <td>c.-62C&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>987023</td>\n",
       "      <td>rs1447064463</td>\n",
       "      <td>ENST00000247933</td>\n",
       "      <td>4-987023-C-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 152...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1199</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*57G&gt;A</td>\n",
       "      <td>c.*57G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1004450</td>\n",
       "      <td>None</td>\n",
       "      <td>ENST00000514224</td>\n",
       "      <td>4-1004450-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 152...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1200</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*59_*60dup</td>\n",
       "      <td>c.*59_*60dup</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1004450</td>\n",
       "      <td>rs1393615274</td>\n",
       "      <td>ENST00000514224</td>\n",
       "      <td>4-1004450-G-GCA</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 152...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1201</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*59A&gt;C</td>\n",
       "      <td>c.*59A&gt;C</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1004452</td>\n",
       "      <td>rs733349</td>\n",
       "      <td>ENST00000514224</td>\n",
       "      <td>4-1004452-A-C</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 7, 'ac_hemi': 0, 'ac_hom': 0, 'an': 152...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1202</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*65C&gt;G</td>\n",
       "      <td>c.*65C&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1004458</td>\n",
       "      <td>rs942204021</td>\n",
       "      <td>ENST00000514224</td>\n",
       "      <td>4-1004458-C-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 152...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1203</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*66C&gt;G</td>\n",
       "      <td>c.*66C&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1004459</td>\n",
       "      <td>rs529309396</td>\n",
       "      <td>ENST00000514224</td>\n",
       "      <td>4-1004459-C-G</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 23, 'ac_hemi': 0, 'ac_hom': 0, 'an': 15...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1204 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              consequence flags          gene_id gene_symbol          hgvs  \\\n",
       "0     5_prime_UTR_variant    []  ENSG00000127415        IDUA      c.-75C>G   \n",
       "1     5_prime_UTR_variant    []  ENSG00000127415        IDUA      c.-72G>A   \n",
       "2     5_prime_UTR_variant    []  ENSG00000127415        IDUA      c.-71C>T   \n",
       "3     5_prime_UTR_variant    []  ENSG00000127415        IDUA      c.-64T>C   \n",
       "4     5_prime_UTR_variant    []  ENSG00000127415        IDUA      c.-62C>G   \n",
       "...                   ...   ...              ...         ...           ...   \n",
       "1199  3_prime_UTR_variant    []  ENSG00000127415        IDUA      c.*57G>A   \n",
       "1200  3_prime_UTR_variant    []  ENSG00000127415        IDUA  c.*59_*60dup   \n",
       "1201  3_prime_UTR_variant    []  ENSG00000127415        IDUA      c.*59A>C   \n",
       "1202  3_prime_UTR_variant    []  ENSG00000127415        IDUA      c.*65C>G   \n",
       "1203  3_prime_UTR_variant    []  ENSG00000127415        IDUA      c.*66C>G   \n",
       "\n",
       "             hgvsc hgvsp   lof lof_filter lof_flags      pos          rsid  \\\n",
       "0         c.-75C>G  None  None       None      None   987010          None   \n",
       "1         c.-72G>A  None  None       None      None   987013          None   \n",
       "2         c.-71C>T  None  None       None      None   987014   rs886059748   \n",
       "3         c.-64T>C  None  None       None      None   987021  rs1473070711   \n",
       "4         c.-62C>G  None  None       None      None   987023  rs1447064463   \n",
       "...            ...   ...   ...        ...       ...      ...           ...   \n",
       "1199      c.*57G>A  None  None       None      None  1004450          None   \n",
       "1200  c.*59_*60dup  None  None       None      None  1004450  rs1393615274   \n",
       "1201      c.*59A>C  None  None       None      None  1004452      rs733349   \n",
       "1202      c.*65C>G  None  None       None      None  1004458   rs942204021   \n",
       "1203      c.*66C>G  None  None       None      None  1004459   rs529309396   \n",
       "\n",
       "        transcript_id       variant_id exome  \\\n",
       "0     ENST00000247933     4-987010-C-G  None   \n",
       "1     ENST00000247933     4-987013-G-A  None   \n",
       "2     ENST00000247933     4-987014-C-T  None   \n",
       "3     ENST00000247933     4-987021-T-C  None   \n",
       "4     ENST00000247933     4-987023-C-G  None   \n",
       "...               ...              ...   ...   \n",
       "1199  ENST00000514224    4-1004450-G-A  None   \n",
       "1200  ENST00000514224  4-1004450-G-GCA  None   \n",
       "1201  ENST00000514224    4-1004452-A-C  None   \n",
       "1202  ENST00000514224    4-1004458-C-G  None   \n",
       "1203  ENST00000514224    4-1004459-C-G  None   \n",
       "\n",
       "                                                 genome lof_curation  \n",
       "0     {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 149...         None  \n",
       "1     {'ac': 3, 'ac_hemi': 0, 'ac_hom': 0, 'an': 151...         None  \n",
       "2     {'ac': 37, 'ac_hemi': 0, 'ac_hom': 0, 'an': 15...         None  \n",
       "3     {'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 146...         None  \n",
       "4     {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 152...         None  \n",
       "...                                                 ...          ...  \n",
       "1199  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 152...         None  \n",
       "1200  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 152...         None  \n",
       "1201  {'ac': 7, 'ac_hemi': 0, 'ac_hom': 0, 'an': 152...         None  \n",
       "1202  {'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 152...         None  \n",
       "1203  {'ac': 23, 'ac_hemi': 0, 'ac_hom': 0, 'an': 15...         None  \n",
       "\n",
       "[1204 rows x 17 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df, df_clin = gs.get_data(standard=False)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>clinvar_variation_id</th>\n",
       "      <th>gold_stars</th>\n",
       "      <th>major_consequence</th>\n",
       "      <th>pos</th>\n",
       "      <th>variant_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>350186</td>\n",
       "      <td>1</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>987014</td>\n",
       "      <td>4-987014-C-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>990330</td>\n",
       "      <td>0</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>987083</td>\n",
       "      <td>4-987083-C-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>550458</td>\n",
       "      <td>1</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>987085</td>\n",
       "      <td>4-987085-A-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>639529</td>\n",
       "      <td>1</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>987086</td>\n",
       "      <td>4-987086-T-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>557150</td>\n",
       "      <td>1</td>\n",
       "      <td>start_lost</td>\n",
       "      <td>987087</td>\n",
       "      <td>4-987087-G-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>404</th>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>558359</td>\n",
       "      <td>1</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1004349</td>\n",
       "      <td>4-1004349-G-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>Likely benign</td>\n",
       "      <td>743489</td>\n",
       "      <td>1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1004381</td>\n",
       "      <td>4-1004381-G-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>Likely benign</td>\n",
       "      <td>754394</td>\n",
       "      <td>1</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1004390</td>\n",
       "      <td>4-1004390-A-T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>960079</td>\n",
       "      <td>1</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>1004391</td>\n",
       "      <td>4-1004391-T-C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>408</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>11920</td>\n",
       "      <td>0</td>\n",
       "      <td>stop_lost</td>\n",
       "      <td>1004391</td>\n",
       "      <td>4-1004391-T-G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>409 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      clinical_significance clinvar_variation_id  gold_stars  \\\n",
       "0    Uncertain significance               350186           1   \n",
       "1    Uncertain significance               990330           0   \n",
       "2                Pathogenic               550458           1   \n",
       "3                Pathogenic               639529           1   \n",
       "4         Likely pathogenic               557150           1   \n",
       "..                      ...                  ...         ...   \n",
       "404  Uncertain significance               558359           1   \n",
       "405           Likely benign               743489           1   \n",
       "406           Likely benign               754394           1   \n",
       "407              Pathogenic               960079           1   \n",
       "408              Pathogenic                11920           0   \n",
       "\n",
       "       major_consequence      pos     variant_id  \n",
       "0    5_prime_UTR_variant   987014   4-987014-C-T  \n",
       "1    5_prime_UTR_variant   987083   4-987083-C-T  \n",
       "2             start_lost   987085   4-987085-A-C  \n",
       "3             start_lost   987086   4-987086-T-C  \n",
       "4             start_lost   987087   4-987087-G-A  \n",
       "..                   ...      ...            ...  \n",
       "404          stop_gained  1004349  4-1004349-G-T  \n",
       "405   synonymous_variant  1004381  4-1004381-G-T  \n",
       "406   synonymous_variant  1004390  4-1004390-A-T  \n",
       "407            stop_lost  1004391  4-1004391-T-C  \n",
       "408            stop_lost  1004391  4-1004391-T-G  \n",
       "\n",
       "[409 rows x 6 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_clin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>flags</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>hgvs</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>lof</th>\n",
       "      <th>lof_filter</th>\n",
       "      <th>lof_flags</th>\n",
       "      <th>...</th>\n",
       "      <th>genome</th>\n",
       "      <th>lof_curation</th>\n",
       "      <th>African</th>\n",
       "      <th>Latino</th>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <th>East Asian</th>\n",
       "      <th>European (Finnish)</th>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <th>Other</th>\n",
       "      <th>South Asian</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-72G&gt;A</td>\n",
       "      <td>c.-72G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 311...</td>\n",
       "      <td>None</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.304291e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-71C&gt;T</td>\n",
       "      <td>c.-71C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>{'ac': 10, 'ac_hemi': 0, 'ac_hom': 0, 'an': 31...</td>\n",
       "      <td>None</td>\n",
       "      <td>1.154468e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>5.207655e-04</td>\n",
       "      <td>9.191176e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-62C&gt;G</td>\n",
       "      <td>c.-62C&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 312...</td>\n",
       "      <td>None</td>\n",
       "      <td>1.155001e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-55G&gt;A</td>\n",
       "      <td>c.-55G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 312...</td>\n",
       "      <td>None</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.497726e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-51C&gt;A</td>\n",
       "      <td>c.-51C&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1764</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*53G&gt;A</td>\n",
       "      <td>c.*53G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "      <td>None</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.179245e-03</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1765</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*55C&gt;T</td>\n",
       "      <td>c.*55C&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "      <td>None</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.482562e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1766</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*59A&gt;C</td>\n",
       "      <td>c.*59A&gt;C</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "      <td>None</td>\n",
       "      <td>1.148897e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1767</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*65C&gt;G</td>\n",
       "      <td>c.*65C&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "      <td>None</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.418485e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1768</th>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>ENSG00000127415</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*66C&gt;G</td>\n",
       "      <td>c.*66C&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>{'ac': 3, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "      <td>None</td>\n",
       "      <td>3.447483e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1769 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              consequence flags          gene_id gene_symbol      hgvs  \\\n",
       "0     5_prime_UTR_variant    []  ENSG00000127415        IDUA  c.-72G>A   \n",
       "1     5_prime_UTR_variant    []  ENSG00000127415        IDUA  c.-71C>T   \n",
       "2     5_prime_UTR_variant    []  ENSG00000127415        IDUA  c.-62C>G   \n",
       "3     5_prime_UTR_variant    []  ENSG00000127415        IDUA  c.-55G>A   \n",
       "4     5_prime_UTR_variant    []  ENSG00000127415        IDUA  c.-51C>A   \n",
       "...                   ...   ...              ...         ...       ...   \n",
       "1764  3_prime_UTR_variant    []  ENSG00000127415        IDUA  c.*53G>A   \n",
       "1765  3_prime_UTR_variant    []  ENSG00000127415        IDUA  c.*55C>T   \n",
       "1766  3_prime_UTR_variant    []  ENSG00000127415        IDUA  c.*59A>C   \n",
       "1767  3_prime_UTR_variant    []  ENSG00000127415        IDUA  c.*65C>G   \n",
       "1768  3_prime_UTR_variant    []  ENSG00000127415        IDUA  c.*66C>G   \n",
       "\n",
       "         hgvsc hgvsp   lof lof_filter lof_flags  ...  \\\n",
       "0     c.-72G>A  None  None       None      None  ...   \n",
       "1     c.-71C>T  None  None       None      None  ...   \n",
       "2     c.-62C>G  None  None       None      None  ...   \n",
       "3     c.-55G>A  None  None       None      None  ...   \n",
       "4     c.-51C>A  None  None       None      None  ...   \n",
       "...        ...   ...   ...        ...       ...  ...   \n",
       "1764  c.*53G>A  None  None       None      None  ...   \n",
       "1765  c.*55C>T  None  None       None      None  ...   \n",
       "1766  c.*59A>C  None  None       None      None  ...   \n",
       "1767  c.*65C>G  None  None       None      None  ...   \n",
       "1768  c.*66C>G  None  None       None      None  ...   \n",
       "\n",
       "                                                 genome lof_curation  \\\n",
       "0     {'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 311...         None   \n",
       "1     {'ac': 10, 'ac_hemi': 0, 'ac_hom': 0, 'an': 31...         None   \n",
       "2     {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 312...         None   \n",
       "3     {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 312...         None   \n",
       "4                                                  None         None   \n",
       "...                                                 ...          ...   \n",
       "1764  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...         None   \n",
       "1765  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...         None   \n",
       "1766  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...         None   \n",
       "1767  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...         None   \n",
       "1768  {'ac': 3, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...         None   \n",
       "\n",
       "           African        Latino Ashkenazi Jewish    East Asian  \\\n",
       "0     0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "1     1.154468e-04  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "2     1.155001e-04  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "3     0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "4     0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "...            ...           ...              ...           ...   \n",
       "1764  0.000000e+00  1.179245e-03     0.000000e+00  0.000000e+00   \n",
       "1765  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "1766  1.148897e-04  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "1767  0.000000e+00  0.000000e+00     0.000000e+00  6.418485e-04   \n",
       "1768  3.447483e-04  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "\n",
       "     European (Finnish) European (non-Finnish)         Other   South Asian  \n",
       "0          0.000000e+00           1.304291e-04  0.000000e+00  0.000000e+00  \n",
       "1          0.000000e+00           5.207655e-04  9.191176e-04  0.000000e+00  \n",
       "2          0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "3          0.000000e+00           6.497726e-05  0.000000e+00  0.000000e+00  \n",
       "4          0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "...                 ...                    ...           ...           ...  \n",
       "1764       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "1765       0.000000e+00           6.482562e-05  0.000000e+00  0.000000e+00  \n",
       "1766       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "1767       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "1768       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "\n",
       "[1769 rows x 25 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gs = GeneSearch(2, \"idua\")\n",
    "df, _ = gs.get_data(standard=False, additional_population_info=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Region Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Source</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Trp402Ter</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>[]</td>\n",
       "      <td>130</td>\n",
       "      <td>151238</td>\n",
       "      <td>0.000860</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002747</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-1002750-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Val</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>150266</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002750</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-1002751-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>151840</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002751</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-1002756-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val405Gly</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>150082</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002756</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-1002759-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ser406Trp</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>151328</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002759</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4-1002766-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala408Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>151864</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002766</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4-1002767-G-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Arg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2757</td>\n",
       "      <td>150922</td>\n",
       "      <td>0.018268</td>\n",
       "      <td>68</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002767</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4-1002768-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Glu</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>151856</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4-1002770-AC-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val411SerfsTer29</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>148668</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4-1002770-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Thr410Pro</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>148666</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variant ID  Gene         Consequence          Annotation Flags  \\\n",
       "0   4-1002747-G-A  IDUA         p.Trp402Ter         stop_gained    []   \n",
       "1   4-1002750-C-T  IDUA         p.Ala403Val    missense_variant    []   \n",
       "2   4-1002751-C-G  IDUA         p.Ala403Ala  synonymous_variant    []   \n",
       "3   4-1002756-T-G  IDUA         p.Val405Gly    missense_variant    []   \n",
       "4   4-1002759-C-G  IDUA         p.Ser406Trp    missense_variant    []   \n",
       "5   4-1002766-C-T  IDUA         p.Ala408Ala  synonymous_variant    []   \n",
       "6   4-1002767-G-C  IDUA         p.Gly409Arg    missense_variant    []   \n",
       "7   4-1002768-G-A  IDUA         p.Gly409Glu    missense_variant    []   \n",
       "8  4-1002770-AC-A  IDUA  p.Val411SerfsTer29  frameshift_variant    []   \n",
       "9   4-1002770-A-C  IDUA         p.Thr410Pro    missense_variant    []   \n",
       "\n",
       "   Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0           130         151238          0.000860                      0   \n",
       "1             2         150266          0.000013                      0   \n",
       "2             2         151840          0.000013                      0   \n",
       "3             1         150082          0.000007                      0   \n",
       "4             3         151328          0.000020                      0   \n",
       "5             3         151864          0.000020                      0   \n",
       "6          2757         150922          0.018268                     68   \n",
       "7             2         151856          0.000013                      0   \n",
       "8             1         148668          0.000007                      0   \n",
       "9             5         148666          0.000034                      0   \n",
       "\n",
       "   Source Chromosome Location Reference Alternative  \n",
       "0  Genome          4  1002747         G           A  \n",
       "1  Genome          4  1002750         C           T  \n",
       "2  Genome          4  1002751         C           G  \n",
       "3  Genome          4  1002756         T           G  \n",
       "4  Genome          4  1002759         C           G  \n",
       "5  Genome          4  1002766         C           T  \n",
       "6  Genome          4  1002767         G           C  \n",
       "7  Genome          4  1002768         G           A  \n",
       "8  Genome          4  1002770        AC           A  \n",
       "9  Genome          4  1002770         A           C  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynoma import RegionSearch\n",
    "rs = RegionSearch(3, 4, 1002741, 1002771)\n",
    "df, _ = rs.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Source</th>\n",
       "      <th>...</th>\n",
       "      <th>African</th>\n",
       "      <th>Amish</th>\n",
       "      <th>Latino</th>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <th>East Asian</th>\n",
       "      <th>European (Finnish)</th>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <th>Other</th>\n",
       "      <th>South Asian</th>\n",
       "      <th>Middle Eastern</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Trp402Ter</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>[]</td>\n",
       "      <td>130</td>\n",
       "      <td>151238</td>\n",
       "      <td>0.000860</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>3.168877e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.971868e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.865877e-04</td>\n",
       "      <td>1.621128e-03</td>\n",
       "      <td>4.803074e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-1002750-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Val</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>150266</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>4.895721e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-1002751-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala403Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>151840</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.944988e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-1002756-T-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val405Gly</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>150082</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.482756e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-1002759-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ser406Trp</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>151328</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.425693e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4-1002766-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Ala408Ala</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>151864</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.418783e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4-1002767-G-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Arg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2757</td>\n",
       "      <td>150922</td>\n",
       "      <td>0.018268</td>\n",
       "      <td>68</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>5.894567e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.057680e-02</td>\n",
       "      <td>9.243212e-03</td>\n",
       "      <td>1.628089e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.872431e-04</td>\n",
       "      <td>1.294343e-02</td>\n",
       "      <td>1.048218e-03</td>\n",
       "      <td>9.740260e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4-1002768-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Gly409Glu</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>151856</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.945768e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4-1002770-AC-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Val411SerfsTer29</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>148668</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>2.484596e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4-1002770-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>p.Thr410Pro</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>148666</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>...</td>\n",
       "      <td>1.242298e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Variant ID  Gene         Consequence          Annotation Flags  \\\n",
       "0   4-1002747-G-A  IDUA         p.Trp402Ter         stop_gained    []   \n",
       "1   4-1002750-C-T  IDUA         p.Ala403Val    missense_variant    []   \n",
       "2   4-1002751-C-G  IDUA         p.Ala403Ala  synonymous_variant    []   \n",
       "3   4-1002756-T-G  IDUA         p.Val405Gly    missense_variant    []   \n",
       "4   4-1002759-C-G  IDUA         p.Ser406Trp    missense_variant    []   \n",
       "5   4-1002766-C-T  IDUA         p.Ala408Ala  synonymous_variant    []   \n",
       "6   4-1002767-G-C  IDUA         p.Gly409Arg    missense_variant    []   \n",
       "7   4-1002768-G-A  IDUA         p.Gly409Glu    missense_variant    []   \n",
       "8  4-1002770-AC-A  IDUA  p.Val411SerfsTer29  frameshift_variant    []   \n",
       "9   4-1002770-A-C  IDUA         p.Thr410Pro    missense_variant    []   \n",
       "\n",
       "   Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0           130         151238          0.000860                      0   \n",
       "1             2         150266          0.000013                      0   \n",
       "2             2         151840          0.000013                      0   \n",
       "3             1         150082          0.000007                      0   \n",
       "4             3         151328          0.000020                      0   \n",
       "5             3         151864          0.000020                      0   \n",
       "6          2757         150922          0.018268                     68   \n",
       "7             2         151856          0.000013                      0   \n",
       "8             1         148668          0.000007                      0   \n",
       "9             5         148666          0.000034                      0   \n",
       "\n",
       "   Source  ...       African         Amish        Latino Ashkenazi Jewish  \\\n",
       "0  Genome  ...  3.168877e-04  0.000000e+00  1.971868e-04     0.000000e+00   \n",
       "1  Genome  ...  4.895721e-05  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "2  Genome  ...  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "3  Genome  ...  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "4  Genome  ...  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "5  Genome  ...  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "6  Genome  ...  5.894567e-02  0.000000e+00  1.057680e-02     9.243212e-03   \n",
       "7  Genome  ...  0.000000e+00  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "8  Genome  ...  2.484596e-05  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "9  Genome  ...  1.242298e-04  0.000000e+00  0.000000e+00     0.000000e+00   \n",
       "\n",
       "     East Asian European (Finnish) European (non-Finnish)         Other  \\\n",
       "0  0.000000e+00       2.865877e-04           1.621128e-03  4.803074e-04   \n",
       "1  0.000000e+00       0.000000e+00           0.000000e+00  0.000000e+00   \n",
       "2  0.000000e+00       0.000000e+00           2.944988e-05  0.000000e+00   \n",
       "3  0.000000e+00       0.000000e+00           1.482756e-05  0.000000e+00   \n",
       "4  0.000000e+00       0.000000e+00           4.425693e-05  0.000000e+00   \n",
       "5  0.000000e+00       0.000000e+00           4.418783e-05  0.000000e+00   \n",
       "6  1.628089e-02       0.000000e+00           4.872431e-04  1.294343e-02   \n",
       "7  0.000000e+00       0.000000e+00           2.945768e-05  0.000000e+00   \n",
       "8  0.000000e+00       0.000000e+00           0.000000e+00  0.000000e+00   \n",
       "9  0.000000e+00       0.000000e+00           0.000000e+00  0.000000e+00   \n",
       "\n",
       "    South Asian Middle Eastern  \n",
       "0  0.000000e+00   0.000000e+00  \n",
       "1  0.000000e+00   0.000000e+00  \n",
       "2  0.000000e+00   0.000000e+00  \n",
       "3  0.000000e+00   0.000000e+00  \n",
       "4  0.000000e+00   0.000000e+00  \n",
       "5  0.000000e+00   0.000000e+00  \n",
       "6  1.048218e-03   9.740260e-03  \n",
       "7  0.000000e+00   0.000000e+00  \n",
       "8  0.000000e+00   0.000000e+00  \n",
       "9  0.000000e+00   0.000000e+00  \n",
       "\n",
       "[10 rows x 24 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df, _ = rs.get_data(additional_population_info=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Source</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "      <th>African</th>\n",
       "      <th>Amish</th>\n",
       "      <th>Latino</th>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <th>East Asian</th>\n",
       "      <th>European (Finnish)</th>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <th>Other</th>\n",
       "      <th>South Asian</th>\n",
       "      <th>Middle Eastern</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>130</td>\n",
       "      <td>151238</td>\n",
       "      <td>0.000860</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002747</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3.168877e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.971868e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.865877e-04</td>\n",
       "      <td>1.621128e-03</td>\n",
       "      <td>4.803074e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>150266</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002750</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4.895721e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>151840</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002751</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.944988e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>150082</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002756</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.482756e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>151328</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002759</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.425693e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3</td>\n",
       "      <td>151864</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002766</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.418783e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2757</td>\n",
       "      <td>150922</td>\n",
       "      <td>0.018268</td>\n",
       "      <td>68</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002767</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>5.894567e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.057680e-02</td>\n",
       "      <td>9.243212e-03</td>\n",
       "      <td>1.628089e-02</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.872431e-04</td>\n",
       "      <td>1.294343e-02</td>\n",
       "      <td>1.048218e-03</td>\n",
       "      <td>9.740260e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>151856</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002768</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.945768e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>148668</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>2.484596e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5</td>\n",
       "      <td>148666</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1002770</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1.242298e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0           130         151238          0.000860                      0   \n",
       "1             2         150266          0.000013                      0   \n",
       "2             2         151840          0.000013                      0   \n",
       "3             1         150082          0.000007                      0   \n",
       "4             3         151328          0.000020                      0   \n",
       "5             3         151864          0.000020                      0   \n",
       "6          2757         150922          0.018268                     68   \n",
       "7             2         151856          0.000013                      0   \n",
       "8             1         148668          0.000007                      0   \n",
       "9             5         148666          0.000034                      0   \n",
       "\n",
       "   Source Chromosome Location Reference Alternative       African  \\\n",
       "0  Genome          4  1002747         G           A  3.168877e-04   \n",
       "1  Genome          4  1002750         C           T  4.895721e-05   \n",
       "2  Genome          4  1002751         C           G  0.000000e+00   \n",
       "3  Genome          4  1002756         T           G  0.000000e+00   \n",
       "4  Genome          4  1002759         C           G  0.000000e+00   \n",
       "5  Genome          4  1002766         C           T  0.000000e+00   \n",
       "6  Genome          4  1002767         G           C  5.894567e-02   \n",
       "7  Genome          4  1002768         G           A  0.000000e+00   \n",
       "8  Genome          4  1002770        AC           A  2.484596e-05   \n",
       "9  Genome          4  1002770         A           C  1.242298e-04   \n",
       "\n",
       "          Amish        Latino Ashkenazi Jewish    East Asian  \\\n",
       "0  0.000000e+00  1.971868e-04     0.000000e+00  0.000000e+00   \n",
       "1  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "2  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "3  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "4  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "5  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "6  0.000000e+00  1.057680e-02     9.243212e-03  1.628089e-02   \n",
       "7  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "8  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "9  0.000000e+00  0.000000e+00     0.000000e+00  0.000000e+00   \n",
       "\n",
       "  European (Finnish) European (non-Finnish)         Other   South Asian  \\\n",
       "0       2.865877e-04           1.621128e-03  4.803074e-04  0.000000e+00   \n",
       "1       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "2       0.000000e+00           2.944988e-05  0.000000e+00  0.000000e+00   \n",
       "3       0.000000e+00           1.482756e-05  0.000000e+00  0.000000e+00   \n",
       "4       0.000000e+00           4.425693e-05  0.000000e+00  0.000000e+00   \n",
       "5       0.000000e+00           4.418783e-05  0.000000e+00  0.000000e+00   \n",
       "6       0.000000e+00           4.872431e-04  1.294343e-02  1.048218e-03   \n",
       "7       0.000000e+00           2.945768e-05  0.000000e+00  0.000000e+00   \n",
       "8       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "9       0.000000e+00           0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "\n",
       "  Middle Eastern  \n",
       "0   0.000000e+00  \n",
       "1   0.000000e+00  \n",
       "2   0.000000e+00  \n",
       "3   0.000000e+00  \n",
       "4   0.000000e+00  \n",
       "5   0.000000e+00  \n",
       "6   9.740260e-03  \n",
       "7   0.000000e+00  \n",
       "8   0.000000e+00  \n",
       "9   0.000000e+00  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.iloc[:, 5:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Number of Hemizygotes</th>\n",
       "      <th>...</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "      <th>African</th>\n",
       "      <th>Latino</th>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <th>East Asian</th>\n",
       "      <th>European (Finnish)</th>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <th>Other</th>\n",
       "      <th>South Asian</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>X-1002741-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>1</td>\n",
       "      <td>31378</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.488451e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>X-1002742-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>2</td>\n",
       "      <td>31316</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>2.303086e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>X-1002746-TATCA-T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>1</td>\n",
       "      <td>31376</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>TATCA</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.489293e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>X-1002765-TTATC-T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>9</td>\n",
       "      <td>31374</td>\n",
       "      <td>0.000287</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>TTATC</td>\n",
       "      <td>T</td>\n",
       "      <td>9.189065e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.486767e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>X-1002767-ATCTATCTT-A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>1</td>\n",
       "      <td>31348</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>ATCTATCTT</td>\n",
       "      <td>A</td>\n",
       "      <td>1.150483e-04</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>X-1002771-ATCTT-A</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>2</td>\n",
       "      <td>31338</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>ATCTT</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.179245e-03</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.496037e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Variant ID  Gene Consequence Annotation  Flags  Allele Count  \\\n",
       "0          X-1002741-C-T  None        None       None  [par]             1   \n",
       "1          X-1002742-G-A  None        None       None  [par]             2   \n",
       "2      X-1002746-TATCA-T  None        None       None  [par]             1   \n",
       "3      X-1002765-TTATC-T  None        None       None  [par]             9   \n",
       "4  X-1002767-ATCTATCTT-A  None        None       None  [par]             1   \n",
       "5      X-1002771-ATCTT-A  None        None       None  [par]             2   \n",
       "\n",
       "   Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0          31378          0.000032                      0   \n",
       "1          31316          0.000064                      0   \n",
       "2          31376          0.000032                      0   \n",
       "3          31374          0.000287                      0   \n",
       "4          31348          0.000032                      0   \n",
       "5          31338          0.000064                      0   \n",
       "\n",
       "   Number of Hemizygotes  ...  Reference Alternative       African  \\\n",
       "0                      0  ...          C           T  0.000000e+00   \n",
       "1                      0  ...          G           A  2.303086e-04   \n",
       "2                      0  ...      TATCA           T  0.000000e+00   \n",
       "3                      0  ...      TTATC           T  9.189065e-04   \n",
       "4                      0  ...  ATCTATCTT           A  1.150483e-04   \n",
       "5                      0  ...      ATCTT           A  0.000000e+00   \n",
       "\n",
       "         Latino Ashkenazi Jewish    East Asian European (Finnish)  \\\n",
       "0  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "1  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "2  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "3  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "4  0.000000e+00     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "5  1.179245e-03     0.000000e+00  0.000000e+00       0.000000e+00   \n",
       "\n",
       "  European (non-Finnish)         Other   South Asian  \n",
       "0           6.488451e-05  0.000000e+00  0.000000e+00  \n",
       "1           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "2           6.489293e-05  0.000000e+00  0.000000e+00  \n",
       "3           6.486767e-05  0.000000e+00  0.000000e+00  \n",
       "4           0.000000e+00  0.000000e+00  0.000000e+00  \n",
       "5           6.496037e-05  0.000000e+00  0.000000e+00  \n",
       "\n",
       "[6 rows x 23 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynoma import RegionSearch\n",
    "rs = RegionSearch(2, 'X', 1002741, 1002771)\n",
    "df, _ = rs.get_data(additional_population_info=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>consequence</th>\n",
       "      <th>flags</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>hgvs</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>lof</th>\n",
       "      <th>lof_filter</th>\n",
       "      <th>lof_flags</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>exome</th>\n",
       "      <th>genome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002741</td>\n",
       "      <td>rs782385877</td>\n",
       "      <td>X-1002741-C-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002742</td>\n",
       "      <td>rs181914603</td>\n",
       "      <td>X-1002742-G-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002746</td>\n",
       "      <td>None</td>\n",
       "      <td>X-1002746-TATCA-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002765</td>\n",
       "      <td>None</td>\n",
       "      <td>X-1002765-TTATC-T</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 9, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002767</td>\n",
       "      <td>None</td>\n",
       "      <td>X-1002767-ATCTATCTT-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>None</td>\n",
       "      <td>[par]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1002771</td>\n",
       "      <td>None</td>\n",
       "      <td>X-1002771-ATCTT-A</td>\n",
       "      <td>None</td>\n",
       "      <td>{'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  consequence  flags gene_id gene_symbol transcript_id  hgvs hgvsc hgvsp  \\\n",
       "0        None  [par]    None        None          None  None  None  None   \n",
       "1        None  [par]    None        None          None  None  None  None   \n",
       "2        None  [par]    None        None          None  None  None  None   \n",
       "3        None  [par]    None        None          None  None  None  None   \n",
       "4        None  [par]    None        None          None  None  None  None   \n",
       "5        None  [par]    None        None          None  None  None  None   \n",
       "\n",
       "    lof lof_filter lof_flags      pos         rsid             variant_id  \\\n",
       "0  None       None      None  1002741  rs782385877          X-1002741-C-T   \n",
       "1  None       None      None  1002742  rs181914603          X-1002742-G-A   \n",
       "2  None       None      None  1002746         None      X-1002746-TATCA-T   \n",
       "3  None       None      None  1002765         None      X-1002765-TTATC-T   \n",
       "4  None       None      None  1002767         None  X-1002767-ATCTATCTT-A   \n",
       "5  None       None      None  1002771         None      X-1002771-ATCTT-A   \n",
       "\n",
       "  exome                                             genome  \n",
       "0  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "1  None  {'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "2  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "3  None  {'ac': 9, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "4  None  {'ac': 1, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  \n",
       "5  None  {'ac': 2, 'ac_hemi': 0, 'ac_hom': 0, 'an': 313...  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df, _ = rs.get_data(standard=False)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Transcript Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Flags</th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Allele Frequency</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Source</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Location</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-987010-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-75C&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>149172</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987010</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4-987013-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-72G&gt;A</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>151256</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987013</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4-987014-C-T</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-71C&gt;T</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>37</td>\n",
       "      <td>152092</td>\n",
       "      <td>0.000243</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987014</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4-987021-T-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-64T&gt;C</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>146758</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987021</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4-987023-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.-62C&gt;G</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>152126</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>987023</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1199</th>\n",
       "      <td>4-1004450-G-A</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*57G&gt;A</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>152196</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004450</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1200</th>\n",
       "      <td>4-1004450-G-GCA</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*59_*60dup</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>152196</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004450</td>\n",
       "      <td>G</td>\n",
       "      <td>GCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1201</th>\n",
       "      <td>4-1004452-A-C</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*59A&gt;C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>7</td>\n",
       "      <td>152152</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004452</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1202</th>\n",
       "      <td>4-1004458-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*65C&gt;G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>152180</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004458</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1203</th>\n",
       "      <td>4-1004459-C-G</td>\n",
       "      <td>IDUA</td>\n",
       "      <td>c.*66C&gt;G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>[]</td>\n",
       "      <td>23</td>\n",
       "      <td>152202</td>\n",
       "      <td>0.000151</td>\n",
       "      <td>0</td>\n",
       "      <td>Genome</td>\n",
       "      <td>4</td>\n",
       "      <td>1004459</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1204 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Variant ID  Gene   Consequence           Annotation Flags  \\\n",
       "0        4-987010-C-G  IDUA      c.-75C>G  5_prime_UTR_variant    []   \n",
       "1        4-987013-G-A  IDUA      c.-72G>A  5_prime_UTR_variant    []   \n",
       "2        4-987014-C-T  IDUA      c.-71C>T  5_prime_UTR_variant    []   \n",
       "3        4-987021-T-C  IDUA      c.-64T>C  5_prime_UTR_variant    []   \n",
       "4        4-987023-C-G  IDUA      c.-62C>G  5_prime_UTR_variant    []   \n",
       "...               ...   ...           ...                  ...   ...   \n",
       "1199    4-1004450-G-A  IDUA      c.*57G>A  3_prime_UTR_variant    []   \n",
       "1200  4-1004450-G-GCA  IDUA  c.*59_*60dup  3_prime_UTR_variant    []   \n",
       "1201    4-1004452-A-C  IDUA      c.*59A>C  3_prime_UTR_variant    []   \n",
       "1202    4-1004458-C-G  IDUA      c.*65C>G  3_prime_UTR_variant    []   \n",
       "1203    4-1004459-C-G  IDUA      c.*66C>G  3_prime_UTR_variant    []   \n",
       "\n",
       "      Allele Count  Allele Number  Allele Frequency  Number of Homozygotes  \\\n",
       "0                1         149172          0.000007                      0   \n",
       "1                3         151256          0.000020                      0   \n",
       "2               37         152092          0.000243                      0   \n",
       "3                2         146758          0.000014                      0   \n",
       "4                1         152126          0.000007                      0   \n",
       "...            ...            ...               ...                    ...   \n",
       "1199             1         152196          0.000007                      0   \n",
       "1200             1         152196          0.000007                      0   \n",
       "1201             7         152152          0.000046                      0   \n",
       "1202             2         152180          0.000013                      0   \n",
       "1203            23         152202          0.000151                      0   \n",
       "\n",
       "      Source Chromosome Location Reference Alternative  \n",
       "0     Genome          4   987010         C           G  \n",
       "1     Genome          4   987013         G           A  \n",
       "2     Genome          4   987014         C           T  \n",
       "3     Genome          4   987021         T           C  \n",
       "4     Genome          4   987023         C           G  \n",
       "...      ...        ...      ...       ...         ...  \n",
       "1199  Genome          4  1004450         G           A  \n",
       "1200  Genome          4  1004450         G         GCA  \n",
       "1201  Genome          4  1004452         A           C  \n",
       "1202  Genome          4  1004458         C           G  \n",
       "1203  Genome          4  1004459         C           G  \n",
       "\n",
       "[1204 rows x 14 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynoma import TranscriptSearch\n",
    "ts = TranscriptSearch(3, \"ENST00000247933\")\n",
    "df, _ = ts.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Variant Search"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Allele Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>1</td>\n",
       "      <td>2082</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other XX</th>\n",
       "      <td>1</td>\n",
       "      <td>1030</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other XY</th>\n",
       "      <td>0</td>\n",
       "      <td>1052</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino</th>\n",
       "      <td>3</td>\n",
       "      <td>15214</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino XX</th>\n",
       "      <td>2</td>\n",
       "      <td>6754</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1KG:CHSXY</th>\n",
       "      <td>0</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1KG:GW</th>\n",
       "      <td>0</td>\n",
       "      <td>232</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1KG:GWDXX</th>\n",
       "      <td>0</td>\n",
       "      <td>116</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1KG:GWDXY</th>\n",
       "      <td>0</td>\n",
       "      <td>116</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>130</td>\n",
       "      <td>151238</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000860</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>252 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Allele Count  Allele Number  Number of Homozygotes  \\\n",
       "Other                 1           2082                      0   \n",
       "Other XX              1           1030                      0   \n",
       "Other XY              0           1052                      0   \n",
       "Latino                3          15214                      0   \n",
       "Latino XX             2           6754                      0   \n",
       "...                 ...            ...                    ...   \n",
       "1KG:CHSXY             0            104                      0   \n",
       "1KG:GW                0            232                      0   \n",
       "1KG:GWDXX             0            116                      0   \n",
       "1KG:GWDXY             0            116                      0   \n",
       "Total               130         151238                      0   \n",
       "\n",
       "           Allele Frequency  \n",
       "Other              0.000480  \n",
       "Other XX           0.000971  \n",
       "Other XY           0.000000  \n",
       "Latino             0.000197  \n",
       "Latino XX          0.000296  \n",
       "...                     ...  \n",
       "1KG:CHSXY          0.000000  \n",
       "1KG:GW             0.000000  \n",
       "1KG:GWDXX          0.000000  \n",
       "1KG:GWDXY          0.000000  \n",
       "Total              0.000860  \n",
       "\n",
       "[252 rows x 4 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from pynoma import VariantSearch\n",
    "vs = VariantSearch(3, '4-1002747-G-A')\n",
    "df, meta = vs.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Allele Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>African</th>\n",
       "      <td>1.0</td>\n",
       "      <td>12584.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>African XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>5882.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>African XY</th>\n",
       "      <td>1.0</td>\n",
       "      <td>6702.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino</th>\n",
       "      <td>0.0</td>\n",
       "      <td>20714.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>12188.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>8526.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <td>0.0</td>\n",
       "      <td>7168.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>3198.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>3970.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian</th>\n",
       "      <td>1.0</td>\n",
       "      <td>8262.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Japanese</th>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Korean</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Other</th>\n",
       "      <td>1.0</td>\n",
       "      <td>4730.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>3970.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian XY</th>\n",
       "      <td>1.0</td>\n",
       "      <td>4292.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>9570.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish) XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>4822.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish) XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>4748.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>69370.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Bulgarian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>844.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Estonian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>4652.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) North-western</th>\n",
       "      <td>0.0</td>\n",
       "      <td>26032.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Other</th>\n",
       "      <td>0.0</td>\n",
       "      <td>17886.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Southern</th>\n",
       "      <td>0.0</td>\n",
       "      <td>5236.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Swedish</th>\n",
       "      <td>0.0</td>\n",
       "      <td>14720.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>28986.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>40384.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>0.0</td>\n",
       "      <td>3980.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1916.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>2064.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>19158.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>4038.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>15120.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>65000.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total XY</th>\n",
       "      <td>2.0</td>\n",
       "      <td>85806.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>2.0</td>\n",
       "      <td>150806.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000013</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Allele Count  Allele Number  \\\n",
       "African                                        1.0        12584.0   \n",
       "African XX                                     0.0         5882.0   \n",
       "African XY                                     1.0         6702.0   \n",
       "Latino                                         0.0        20714.0   \n",
       "Latino XX                                      0.0        12188.0   \n",
       "Latino XY                                      0.0         8526.0   \n",
       "Ashkenazi Jewish                               0.0         7168.0   \n",
       "Ashkenazi Jewish XX                            0.0         3198.0   \n",
       "Ashkenazi Jewish XY                            0.0         3970.0   \n",
       "East Asian                                     1.0         8262.0   \n",
       "East Asian Japanese                            0.0            8.0   \n",
       "East Asian Korean                              0.0         1964.0   \n",
       "East Asian Other                               1.0         4730.0   \n",
       "East Asian XX                                  0.0         3970.0   \n",
       "East Asian XY                                  1.0         4292.0   \n",
       "European (Finnish)                             0.0         9570.0   \n",
       "European (Finnish) XX                          0.0         4822.0   \n",
       "European (Finnish) XY                          0.0         4748.0   \n",
       "European (non-Finnish)                         0.0        69370.0   \n",
       "European (non-Finnish) Bulgarian               0.0          844.0   \n",
       "European (non-Finnish) Estonian                0.0         4652.0   \n",
       "European (non-Finnish) North-western           0.0        26032.0   \n",
       "European (non-Finnish) Other                   0.0        17886.0   \n",
       "European (non-Finnish) Southern                0.0         5236.0   \n",
       "European (non-Finnish) Swedish                 0.0        14720.0   \n",
       "European (non-Finnish) XX                      0.0        28986.0   \n",
       "European (non-Finnish) XY                      0.0        40384.0   \n",
       "Other                                          0.0         3980.0   \n",
       "Other XX                                       0.0         1916.0   \n",
       "Other XY                                       0.0         2064.0   \n",
       "South Asian                                    0.0        19158.0   \n",
       "South Asian XX                                 0.0         4038.0   \n",
       "South Asian XY                                 0.0        15120.0   \n",
       "Total XX                                       0.0        65000.0   \n",
       "Total XY                                       2.0        85806.0   \n",
       "Total                                          2.0       150806.0   \n",
       "\n",
       "                                      Number of Homozygotes  Allele Frequency  \n",
       "African                                                 0.0          0.000079  \n",
       "African XX                                              0.0          0.000000  \n",
       "African XY                                              0.0          0.000149  \n",
       "Latino                                                  0.0          0.000000  \n",
       "Latino XX                                               0.0          0.000000  \n",
       "Latino XY                                               0.0          0.000000  \n",
       "Ashkenazi Jewish                                        0.0          0.000000  \n",
       "Ashkenazi Jewish XX                                     0.0          0.000000  \n",
       "Ashkenazi Jewish XY                                     0.0          0.000000  \n",
       "East Asian                                              0.0          0.000121  \n",
       "East Asian Japanese                                     0.0          0.000000  \n",
       "East Asian Korean                                       0.0          0.000000  \n",
       "East Asian Other                                        0.0          0.000211  \n",
       "East Asian XX                                           0.0          0.000000  \n",
       "East Asian XY                                           0.0          0.000233  \n",
       "European (Finnish)                                      0.0          0.000000  \n",
       "European (Finnish) XX                                   0.0          0.000000  \n",
       "European (Finnish) XY                                   0.0          0.000000  \n",
       "European (non-Finnish)                                  0.0          0.000000  \n",
       "European (non-Finnish) Bulgarian                        0.0          0.000000  \n",
       "European (non-Finnish) Estonian                         0.0          0.000000  \n",
       "European (non-Finnish) North-western                    0.0          0.000000  \n",
       "European (non-Finnish) Other                            0.0          0.000000  \n",
       "European (non-Finnish) Southern                         0.0          0.000000  \n",
       "European (non-Finnish) Swedish                          0.0          0.000000  \n",
       "European (non-Finnish) XX                               0.0          0.000000  \n",
       "European (non-Finnish) XY                               0.0          0.000000  \n",
       "Other                                                   0.0          0.000000  \n",
       "Other XX                                                0.0          0.000000  \n",
       "Other XY                                                0.0          0.000000  \n",
       "South Asian                                             0.0          0.000000  \n",
       "South Asian XX                                          0.0          0.000000  \n",
       "South Asian XY                                          0.0          0.000000  \n",
       "Total XX                                                0.0          0.000000  \n",
       "Total XY                                                0.0          0.000023  \n",
       "Total                                                   0.0          0.000013  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vs = VariantSearch(2, '4-980866-C-A')\n",
    "df, meta = vs.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "African 1.0\n",
      "African XX 0.0\n",
      "African XY 1.0\n",
      "Latino 0.0\n",
      "Latino XX 0.0\n",
      "Latino XY 0.0\n",
      "Ashkenazi Jewish 0.0\n",
      "Ashkenazi Jewish XX 0.0\n",
      "Ashkenazi Jewish XY 0.0\n",
      "East Asian 1.0\n",
      "East Asian Japanese 0.0\n",
      "East Asian Korean 0.0\n",
      "East Asian Other 1.0\n",
      "East Asian XX 0.0\n",
      "East Asian XY 1.0\n",
      "European (Finnish) 0.0\n",
      "European (Finnish) XX 0.0\n",
      "European (Finnish) XY 0.0\n",
      "European (non-Finnish) 0.0\n",
      "European (non-Finnish) Bulgarian 0.0\n",
      "European (non-Finnish) Estonian 0.0\n",
      "European (non-Finnish) North-western 0.0\n",
      "European (non-Finnish) Other 0.0\n",
      "European (non-Finnish) Southern 0.0\n",
      "European (non-Finnish) Swedish 0.0\n",
      "European (non-Finnish) XX 0.0\n",
      "European (non-Finnish) XY 0.0\n",
      "Other 0.0\n",
      "Other XX 0.0\n",
      "Other XY 0.0\n",
      "South Asian 0.0\n",
      "South Asian XX 0.0\n",
      "South Asian XY 0.0\n",
      "Total XX 0.0\n",
      "Total XY 2.0\n",
      "Total 2.0\n"
     ]
    }
   ],
   "source": [
    "for row in df.iterrows():\n",
    "    print(row[0], row[1]['Allele Count'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'variantId': '4-980866-C-A',\n",
       " 'reference_genome': 'GRCh37',\n",
       " 'chrom': '4',\n",
       " 'pos': 980866,\n",
       " 'ref': 'C',\n",
       " 'alt': 'A',\n",
       " 'colocatedVariants': ['4-980866-C-T'],\n",
       " 'multiNucleotideVariants': None,\n",
       " 'exome': {'ac': 1,\n",
       "  'an': 119640,\n",
       "  'ac_hemi': 0,\n",
       "  'ac_hom': 0,\n",
       "  'faf95': {'popmax': None, 'popmax_population': None},\n",
       "  'filters': [],\n",
       "  'populations': [{'id': 'AFR',\n",
       "    'ac': 0,\n",
       "    'an': 3928,\n",
       "    'ac_hemi': 0,\n",
       "    'ac_hom': 0},\n",
       "   {'id': 'AFR_FEMALE', 'ac': 0, 'an': 2222, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'AFR_MALE', 'ac': 0, 'an': 1706, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'AMR', 'ac': 0, 'an': 19888, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'AMR_FEMALE', 'ac': 0, 'an': 11790, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'AMR_MALE', 'ac': 0, 'an': 8098, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'ASJ', 'ac': 0, 'an': 6878, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'ASJ_FEMALE', 'ac': 0, 'an': 3118, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'ASJ_MALE', 'ac': 0, 'an': 3760, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'EAS', 'ac': 1, 'an': 6702, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'EAS_JPN', 'ac': 0, 'an': 8, 'ac_hemi': None, 'ac_hom': 0},\n",
       "   {'id': 'EAS_KOR', 'ac': 0, 'an': 1964, 'ac_hemi': None, 'ac_hom': 0},\n",
       "   {'id': 'EAS_OEA', 'ac': 1, 'an': 4730, 'ac_hemi': None, 'ac_hom': 0},\n",
       "   {'id': 'EAS_FEMALE', 'ac': 0, 'an': 3432, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'EAS_MALE', 'ac': 1, 'an': 3270, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'FIN', 'ac': 0, 'an': 6184, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'FIN_FEMALE', 'ac': 0, 'an': 3018, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'FIN_MALE', 'ac': 0, 'an': 3166, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'NFE', 'ac': 0, 'an': 54006, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'NFE_BGR', 'ac': 0, 'an': 844, 'ac_hemi': None, 'ac_hom': 0},\n",
       "   {'id': 'NFE_EST', 'ac': 0, 'an': 82, 'ac_hemi': None, 'ac_hom': 0},\n",
       "   {'id': 'NFE_NWE', 'ac': 0, 'an': 17468, 'ac_hemi': None, 'ac_hom': 0},\n",
       "   {'id': 'NFE_ONF', 'ac': 0, 'an': 15760, 'ac_hemi': None, 'ac_hom': 0},\n",
       "   {'id': 'NFE_SEU', 'ac': 0, 'an': 5132, 'ac_hemi': None, 'ac_hom': 0},\n",
       "   {'id': 'NFE_SWE', 'ac': 0, 'an': 14720, 'ac_hemi': None, 'ac_hom': 0},\n",
       "   {'id': 'NFE_FEMALE', 'ac': 0, 'an': 22216, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'NFE_MALE', 'ac': 0, 'an': 31790, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'OTH', 'ac': 0, 'an': 2896, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'OTH_FEMALE', 'ac': 0, 'an': 1356, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'OTH_MALE', 'ac': 0, 'an': 1540, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'SAS', 'ac': 0, 'an': 19158, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'SAS_FEMALE', 'ac': 0, 'an': 4038, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'SAS_MALE', 'ac': 0, 'an': 15120, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'FEMALE', 'ac': 0, 'an': 51190, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "   {'id': 'MALE', 'ac': 1, 'an': 68450, 'ac_hemi': 0, 'ac_hom': 0}],\n",
       "  'age_distribution': {'het': {'bin_edges': [30,\n",
       "     35,\n",
       "     40,\n",
       "     45,\n",
       "     50,\n",
       "     55,\n",
       "     60,\n",
       "     65,\n",
       "     70,\n",
       "     75,\n",
       "     80],\n",
       "    'bin_freq': [0, 0, 0, 0, 0, 1, 0, 0, 0, 0],\n",
       "    'n_smaller': 0,\n",
       "    'n_larger': 0},\n",
       "   'hom': {'bin_edges': [30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80],\n",
       "    'bin_freq': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],\n",
       "    'n_smaller': 0,\n",
       "    'n_larger': 0}},\n",
       "  'qualityMetrics': {'alleleBalance': {'alt': {'bin_edges': [0,\n",
       "      0.05,\n",
       "      0.1,\n",
       "      0.15,\n",
       "      0.2,\n",
       "      0.25,\n",
       "      0.3,\n",
       "      0.35,\n",
       "      0.4,\n",
       "      0.45,\n",
       "      0.5,\n",
       "      0.55,\n",
       "      0.6,\n",
       "      0.65,\n",
       "      0.7,\n",
       "      0.75,\n",
       "      0.8,\n",
       "      0.85,\n",
       "      0.9,\n",
       "      0.95,\n",
       "      1],\n",
       "     'bin_freq': [0, 0, 10, 8, 2, 4, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'genotypeDepth': {'all': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [38361,\n",
       "      26585,\n",
       "      19442,\n",
       "      13011,\n",
       "      10226,\n",
       "      6187,\n",
       "      5130,\n",
       "      5323,\n",
       "      1212,\n",
       "      181,\n",
       "      56,\n",
       "      15,\n",
       "      8,\n",
       "      3,\n",
       "      2,\n",
       "      2,\n",
       "      0,\n",
       "      1,\n",
       "      0,\n",
       "      0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 3},\n",
       "    'alt': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [0, 7, 13, 6, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'genotypeQuality': {'all': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [7056,\n",
       "      14233,\n",
       "      7006,\n",
       "      12882,\n",
       "      11150,\n",
       "      5108,\n",
       "      8633,\n",
       "      7071,\n",
       "      3051,\n",
       "      5261,\n",
       "      4507,\n",
       "      2024,\n",
       "      5289,\n",
       "      1493,\n",
       "      2402,\n",
       "      1479,\n",
       "      2560,\n",
       "      472,\n",
       "      1949,\n",
       "      6097],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0},\n",
       "    'alt': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [6, 4, 4, 3, 1, 4, 1, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'siteQualityMetrics': [{'metric': 'BaseQRankSum', 'value': -1.067},\n",
       "    {'metric': 'ClippingRankSum', 'value': -0.067},\n",
       "    {'metric': 'DP', 'value': 2224721},\n",
       "    {'metric': 'FS', 'value': 4.571},\n",
       "    {'metric': 'InbreedingCoeff', 'value': 0.0261},\n",
       "    {'metric': 'MQ', 'value': 60},\n",
       "    {'metric': 'MQRankSum', 'value': 0.11},\n",
       "    {'metric': 'pab_max', 'value': 1},\n",
       "    {'metric': 'QD', 'value': 0.78},\n",
       "    {'metric': 'ReadPosRankSum', 'value': 0.572},\n",
       "    {'metric': 'RF', 'value': 0.20822502302944865},\n",
       "    {'metric': 'SiteQuality', 'value': 703.21},\n",
       "    {'metric': 'SOR', 'value': 0.374},\n",
       "    {'metric': 'VQSLOD', 'value': -2.103},\n",
       "    {'metric': 'VQSR_NEGATIVE_TRAIN_SITE', 'value': 1},\n",
       "    {'metric': 'VQSR_POSITIVE_TRAIN_SITE', 'value': None}]}},\n",
       " 'genome': {'ac': 1,\n",
       "  'an': 31166,\n",
       "  'ac_hemi': 0,\n",
       "  'ac_hom': 0,\n",
       "  'faf95': {'popmax': None, 'popmax_population': None},\n",
       "  'filters': [],\n",
       "  'age_distribution': {'het': {'bin_edges': [30,\n",
       "     35,\n",
       "     40,\n",
       "     45,\n",
       "     50,\n",
       "     55,\n",
       "     60,\n",
       "     65,\n",
       "     70,\n",
       "     75,\n",
       "     80],\n",
       "    'bin_freq': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],\n",
       "    'n_smaller': 0,\n",
       "    'n_larger': 0},\n",
       "   'hom': {'bin_edges': [30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80],\n",
       "    'bin_freq': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],\n",
       "    'n_smaller': 0,\n",
       "    'n_larger': 0}},\n",
       "  'qualityMetrics': {'alleleBalance': {'alt': {'bin_edges': [0,\n",
       "      0.05,\n",
       "      0.1,\n",
       "      0.15,\n",
       "      0.2,\n",
       "      0.25,\n",
       "      0.3,\n",
       "      0.35,\n",
       "      0.4,\n",
       "      0.45,\n",
       "      0.5,\n",
       "      0.55,\n",
       "      0.6,\n",
       "      0.65,\n",
       "      0.7,\n",
       "      0.75,\n",
       "      0.8,\n",
       "      0.85,\n",
       "      0.9,\n",
       "      0.95,\n",
       "      1],\n",
       "     'bin_freq': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'genotypeDepth': {'all': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [15,\n",
       "      96,\n",
       "      366,\n",
       "      1222,\n",
       "      2927,\n",
       "      3175,\n",
       "      4014,\n",
       "      3023,\n",
       "      675,\n",
       "      129,\n",
       "      35,\n",
       "      15,\n",
       "      6,\n",
       "      3,\n",
       "      2,\n",
       "      1,\n",
       "      2,\n",
       "      1,\n",
       "      1,\n",
       "      0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0},\n",
       "    'alt': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'genotypeQuality': {'all': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [5,\n",
       "      15,\n",
       "      14,\n",
       "      26,\n",
       "      54,\n",
       "      51,\n",
       "      129,\n",
       "      208,\n",
       "      182,\n",
       "      426,\n",
       "      645,\n",
       "      415,\n",
       "      1611,\n",
       "      547,\n",
       "      1292,\n",
       "      724,\n",
       "      1856,\n",
       "      367,\n",
       "      1720,\n",
       "      5421],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0},\n",
       "    'alt': {'bin_edges': [0,\n",
       "      5,\n",
       "      10,\n",
       "      15,\n",
       "      20,\n",
       "      25,\n",
       "      30,\n",
       "      35,\n",
       "      40,\n",
       "      45,\n",
       "      50,\n",
       "      55,\n",
       "      60,\n",
       "      65,\n",
       "      70,\n",
       "      75,\n",
       "      80,\n",
       "      85,\n",
       "      90,\n",
       "      95,\n",
       "      100],\n",
       "     'bin_freq': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],\n",
       "     'n_smaller': 0,\n",
       "     'n_larger': 0}},\n",
       "   'siteQualityMetrics': [{'metric': 'BaseQRankSum', 'value': 1.43},\n",
       "    {'metric': 'ClippingRankSum', 'value': -0.835},\n",
       "    {'metric': 'DP', 'value': 608180},\n",
       "    {'metric': 'FS', 'value': 4.1},\n",
       "    {'metric': 'InbreedingCoeff', 'value': 0.0017},\n",
       "    {'metric': 'MQ', 'value': 60},\n",
       "    {'metric': 'MQRankSum', 'value': 1.55},\n",
       "    {'metric': 'pab_max', 'value': 0.3074562549591067},\n",
       "    {'metric': 'QD', 'value': 20.94},\n",
       "    {'metric': 'ReadPosRankSum', 'value': -1.61},\n",
       "    {'metric': 'RF', 'value': 0.6972491779403581},\n",
       "    {'metric': 'SiteQuality', 'value': 502.48},\n",
       "    {'metric': 'SOR', 'value': 0.164},\n",
       "    {'metric': 'VQSLOD', 'value': 11.29},\n",
       "    {'metric': 'VQSR_NEGATIVE_TRAIN_SITE', 'value': None},\n",
       "    {'metric': 'VQSR_POSITIVE_TRAIN_SITE', 'value': None}]}},\n",
       " 'flags': [],\n",
       " 'rsid': 'rs990934606',\n",
       " 'sortedTranscriptConsequences': [{'canonical': True,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '8',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'c.-7C>A',\n",
       "   'hgvsc': 'c.-7C>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': '5_prime_UTR_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000247933',\n",
       "   'transcript_version': '4'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '8',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'c.-7C>A',\n",
       "   'hgvsc': 'c.-7C>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': '5_prime_UTR_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000453894',\n",
       "   'transcript_version': '1'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '8',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'c.-7C>A',\n",
       "   'hgvsc': 'c.-7C>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': '5_prime_UTR_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000502910',\n",
       "   'transcript_version': '1'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000145217',\n",
       "   'gene_version': '9',\n",
       "   'gene_symbol': 'SLC26A1',\n",
       "   'hgvs': 'c.576+4050G>T',\n",
       "   'hgvsc': 'c.576+4050G>T',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'intron_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000398520',\n",
       "   'transcript_version': '2'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '8',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'n.3C>A',\n",
       "   'hgvsc': 'n.3C>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000506561',\n",
       "   'transcript_version': '1'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '8',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'n.13C>A',\n",
       "   'hgvsc': 'n.13C>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000508168',\n",
       "   'transcript_version': '1'},\n",
       "  {'canonical': None,\n",
       "   'gene_id': 'ENSG00000127415',\n",
       "   'gene_version': '8',\n",
       "   'gene_symbol': 'IDUA',\n",
       "   'hgvs': 'n.35C>A',\n",
       "   'hgvsc': 'n.35C>A',\n",
       "   'hgvsp': None,\n",
       "   'lof': None,\n",
       "   'lof_flags': None,\n",
       "   'lof_filter': None,\n",
       "   'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "   'polyphen_prediction': None,\n",
       "   'sift_prediction': None,\n",
       "   'transcript_id': 'ENST00000514698',\n",
       "   'transcript_version': '1'}]}"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Allele Count</th>\n",
       "      <th>Allele Number</th>\n",
       "      <th>Number of Homozygotes</th>\n",
       "      <th>Allele Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>African</th>\n",
       "      <td>0.0</td>\n",
       "      <td>8642.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>African XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>3654.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>African XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>4988.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino</th>\n",
       "      <td>0.0</td>\n",
       "      <td>842.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>410.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Latino XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>432.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish</th>\n",
       "      <td>0.0</td>\n",
       "      <td>286.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>78.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ashkenazi Jewish XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1558.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Japanese</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Korean</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian Other</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>538.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>East Asian XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1020.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish)</th>\n",
       "      <td>0.0</td>\n",
       "      <td>3446.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish) XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1838.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (Finnish) XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1608.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish)</th>\n",
       "      <td>2.0</td>\n",
       "      <td>15334.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Bulgarian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Estonian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>4524.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) North-western</th>\n",
       "      <td>2.0</td>\n",
       "      <td>8572.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Other</th>\n",
       "      <td>0.0</td>\n",
       "      <td>2134.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Southern</th>\n",
       "      <td>0.0</td>\n",
       "      <td>104.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) Swedish</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) XX</th>\n",
       "      <td>2.0</td>\n",
       "      <td>6762.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>European (non-Finnish) XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>8572.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1086.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>562.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>524.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian XX</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>South Asian XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total XX</th>\n",
       "      <td>2.0</td>\n",
       "      <td>13842.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total XY</th>\n",
       "      <td>0.0</td>\n",
       "      <td>17352.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>2.0</td>\n",
       "      <td>31194.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000064</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Allele Count  Allele Number  \\\n",
       "African                                        0.0         8642.0   \n",
       "African XX                                     0.0         3654.0   \n",
       "African XY                                     0.0         4988.0   \n",
       "Latino                                         0.0          842.0   \n",
       "Latino XX                                      0.0          410.0   \n",
       "Latino XY                                      0.0          432.0   \n",
       "Ashkenazi Jewish                               0.0          286.0   \n",
       "Ashkenazi Jewish XX                            0.0           78.0   \n",
       "Ashkenazi Jewish XY                            0.0          208.0   \n",
       "East Asian                                     0.0         1558.0   \n",
       "East Asian Japanese                            0.0            0.0   \n",
       "East Asian Korean                              0.0            0.0   \n",
       "East Asian Other                               0.0            0.0   \n",
       "East Asian XX                                  0.0          538.0   \n",
       "East Asian XY                                  0.0         1020.0   \n",
       "European (Finnish)                             0.0         3446.0   \n",
       "European (Finnish) XX                          0.0         1838.0   \n",
       "European (Finnish) XY                          0.0         1608.0   \n",
       "European (non-Finnish)                         2.0        15334.0   \n",
       "European (non-Finnish) Bulgarian               0.0            0.0   \n",
       "European (non-Finnish) Estonian                0.0         4524.0   \n",
       "European (non-Finnish) North-western           2.0         8572.0   \n",
       "European (non-Finnish) Other                   0.0         2134.0   \n",
       "European (non-Finnish) Southern                0.0          104.0   \n",
       "European (non-Finnish) Swedish                 0.0            0.0   \n",
       "European (non-Finnish) XX                      2.0         6762.0   \n",
       "European (non-Finnish) XY                      0.0         8572.0   \n",
       "Other                                          0.0         1086.0   \n",
       "Other XX                                       0.0          562.0   \n",
       "Other XY                                       0.0          524.0   \n",
       "South Asian                                    0.0            0.0   \n",
       "South Asian XX                                 0.0            0.0   \n",
       "South Asian XY                                 0.0            0.0   \n",
       "Total XX                                       2.0        13842.0   \n",
       "Total XY                                       0.0        17352.0   \n",
       "Total                                          2.0        31194.0   \n",
       "\n",
       "                                      Number of Homozygotes  Allele Frequency  \n",
       "African                                                 0.0          0.000000  \n",
       "African XX                                              0.0          0.000000  \n",
       "African XY                                              0.0          0.000000  \n",
       "Latino                                                  0.0          0.000000  \n",
       "Latino XX                                               0.0          0.000000  \n",
       "Latino XY                                               0.0          0.000000  \n",
       "Ashkenazi Jewish                                        0.0          0.000000  \n",
       "Ashkenazi Jewish XX                                     0.0          0.000000  \n",
       "Ashkenazi Jewish XY                                     0.0          0.000000  \n",
       "East Asian                                              0.0          0.000000  \n",
       "East Asian Japanese                                     0.0          0.000000  \n",
       "East Asian Korean                                       0.0          0.000000  \n",
       "East Asian Other                                        0.0          0.000000  \n",
       "East Asian XX                                           0.0          0.000000  \n",
       "East Asian XY                                           0.0          0.000000  \n",
       "European (Finnish)                                      0.0          0.000000  \n",
       "European (Finnish) XX                                   0.0          0.000000  \n",
       "European (Finnish) XY                                   0.0          0.000000  \n",
       "European (non-Finnish)                                  0.0          0.000130  \n",
       "European (non-Finnish) Bulgarian                        0.0          0.000000  \n",
       "European (non-Finnish) Estonian                         0.0          0.000000  \n",
       "European (non-Finnish) North-western                    0.0          0.000233  \n",
       "European (non-Finnish) Other                            0.0          0.000000  \n",
       "European (non-Finnish) Southern                         0.0          0.000000  \n",
       "European (non-Finnish) Swedish                          0.0          0.000000  \n",
       "European (non-Finnish) XX                               0.0          0.000296  \n",
       "European (non-Finnish) XY                               0.0          0.000000  \n",
       "Other                                                   0.0          0.000000  \n",
       "Other XX                                                0.0          0.000000  \n",
       "Other XY                                                0.0          0.000000  \n",
       "South Asian                                             0.0          0.000000  \n",
       "South Asian XX                                          0.0          0.000000  \n",
       "South Asian XY                                          0.0          0.000000  \n",
       "Total XX                                                0.0          0.000144  \n",
       "Total XY                                                0.0          0.000000  \n",
       "Total                                                   0.0          0.000064  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vs = VariantSearch(2, '4-980801-G-A')\n",
    "df, meta = vs.get_data()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
