{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pynomad import VariantSearch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "vs = VariantSearch(3, '4-1002747-G-A')\n",
    "js=vs.get_json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'data': {'variant': {'variantId': '4-1002747-G-A',\n",
       "   'reference_genome': 'GRCh38',\n",
       "   'chrom': '4',\n",
       "   'pos': 1002747,\n",
       "   'ref': 'G',\n",
       "   'alt': 'A',\n",
       "   'colocatedVariants': ['4-1002747-G-T'],\n",
       "   'multiNucleotideVariants': [],\n",
       "   'exome': None,\n",
       "   'genome': {'ac': 120,\n",
       "    'an': 142464,\n",
       "    'ac_hemi': 0,\n",
       "    'ac_hom': 0,\n",
       "    'faf95': {'popmax': 0.0013063199999999997, 'popmax_population': 'NFE'},\n",
       "    'filters': [],\n",
       "    'populations': [{'id': 'AFR',\n",
       "      'ac': 13,\n",
       "      'an': 41628,\n",
       "      'ac_hemi': 0,\n",
       "      'ac_hom': 0},\n",
       "     {'id': 'AFR_FEMALE', 'ac': 6, 'an': 22480, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AFR_MALE', 'ac': 7, 'an': 19148, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AMI', 'ac': 0, 'an': 898, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AMI_FEMALE', 'ac': 0, 'an': 470, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AMI_MALE', 'ac': 0, 'an': 428, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AMR', 'ac': 3, 'an': 13600, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AMR_FEMALE', 'ac': 2, 'an': 5882, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'AMR_MALE', 'ac': 1, 'an': 7718, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'ASJ', 'ac': 0, 'an': 3316, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'ASJ_FEMALE', 'ac': 0, 'an': 1756, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'ASJ_MALE', 'ac': 0, 'an': 1560, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'EAS', 'ac': 0, 'an': 3124, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'EAS_FEMALE', 'ac': 0, 'an': 1440, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'EAS_MALE', 'ac': 0, 'an': 1684, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'FIN', 'ac': 3, 'an': 10322, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'FIN_FEMALE', 'ac': 1, 'an': 2476, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'FIN_MALE', 'ac': 2, 'an': 7846, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'NFE', 'ac': 100, 'an': 64398, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'NFE_FEMALE', 'ac': 59, 'an': 37292, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'NFE_MALE', 'ac': 41, 'an': 27106, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'OTH', 'ac': 1, 'an': 2142, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'OTH_FEMALE', 'ac': 1, 'an': 1094, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'OTH_MALE', 'ac': 0, 'an': 1048, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'SAS', 'ac': 0, 'an': 3036, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'SAS_FEMALE', 'ac': 0, 'an': 560, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'SAS_MALE', 'ac': 0, 'an': 2476, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'FEMALE', 'ac': 69, 'an': 73450, 'ac_hemi': 0, 'ac_hom': 0},\n",
       "     {'id': 'MALE', 'ac': 51, 'an': 69014, 'ac_hemi': 0, 'ac_hom': 0}],\n",
       "    'age_distribution': None,\n",
       "    'qualityMetrics': {'alleleBalance': {'alt': {'bin_edges': [0,\n",
       "        0.05,\n",
       "        0.1,\n",
       "        0.15,\n",
       "        0.2,\n",
       "        0.25,\n",
       "        0.3,\n",
       "        0.35,\n",
       "        0.4,\n",
       "        0.45,\n",
       "        0.5,\n",
       "        0.55,\n",
       "        0.6,\n",
       "        0.65,\n",
       "        0.7,\n",
       "        0.75,\n",
       "        0.8,\n",
       "        0.85,\n",
       "        0.9,\n",
       "        0.95,\n",
       "        1],\n",
       "       'bin_freq': [0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        3,\n",
       "        3,\n",
       "        8,\n",
       "        26,\n",
       "        16,\n",
       "        26,\n",
       "        17,\n",
       "        13,\n",
       "        5,\n",
       "        3,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0],\n",
       "       'n_smaller': 0,\n",
       "       'n_larger': 0}},\n",
       "     'genotypeDepth': {'all': {'bin_edges': [0,\n",
       "        5,\n",
       "        10,\n",
       "        15,\n",
       "        20,\n",
       "        25,\n",
       "        30,\n",
       "        35,\n",
       "        40,\n",
       "        45,\n",
       "        50,\n",
       "        55,\n",
       "        60,\n",
       "        65,\n",
       "        70,\n",
       "        75,\n",
       "        80,\n",
       "        85,\n",
       "        90,\n",
       "        95,\n",
       "        100],\n",
       "       'bin_freq': [11,\n",
       "        67,\n",
       "        641,\n",
       "        3138,\n",
       "        6868,\n",
       "        12475,\n",
       "        21477,\n",
       "        17174,\n",
       "        5825,\n",
       "        1989,\n",
       "        945,\n",
       "        476,\n",
       "        212,\n",
       "        94,\n",
       "        59,\n",
       "        30,\n",
       "        19,\n",
       "        11,\n",
       "        2,\n",
       "        4],\n",
       "       'n_smaller': 0,\n",
       "       'n_larger': 13},\n",
       "      'alt': {'bin_edges': [0,\n",
       "        5,\n",
       "        10,\n",
       "        15,\n",
       "        20,\n",
       "        25,\n",
       "        30,\n",
       "        35,\n",
       "        40,\n",
       "        45,\n",
       "        50,\n",
       "        55,\n",
       "        60,\n",
       "        65,\n",
       "        70,\n",
       "        75,\n",
       "        80,\n",
       "        85,\n",
       "        90,\n",
       "        95,\n",
       "        100],\n",
       "       'bin_freq': [0,\n",
       "        0,\n",
       "        0,\n",
       "        8,\n",
       "        13,\n",
       "        25,\n",
       "        29,\n",
       "        15,\n",
       "        10,\n",
       "        9,\n",
       "        8,\n",
       "        3,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0],\n",
       "       'n_smaller': 0,\n",
       "       'n_larger': 0}},\n",
       "     'genotypeQuality': {'all': {'bin_edges': [0,\n",
       "        5,\n",
       "        10,\n",
       "        15,\n",
       "        20,\n",
       "        25,\n",
       "        30,\n",
       "        35,\n",
       "        40,\n",
       "        45,\n",
       "        50,\n",
       "        55,\n",
       "        60,\n",
       "        65,\n",
       "        70,\n",
       "        75,\n",
       "        80,\n",
       "        85,\n",
       "        90,\n",
       "        95,\n",
       "        100],\n",
       "       'bin_freq': [42,\n",
       "        34,\n",
       "        79,\n",
       "        100,\n",
       "        20247,\n",
       "        14345,\n",
       "        9795,\n",
       "        6614,\n",
       "        5441,\n",
       "        3845,\n",
       "        2497,\n",
       "        2219,\n",
       "        1635,\n",
       "        1101,\n",
       "        926,\n",
       "        674,\n",
       "        461,\n",
       "        380,\n",
       "        262,\n",
       "        833],\n",
       "       'n_smaller': 0,\n",
       "       'n_larger': 0},\n",
       "      'alt': {'bin_edges': [0,\n",
       "        5,\n",
       "        10,\n",
       "        15,\n",
       "        20,\n",
       "        25,\n",
       "        30,\n",
       "        35,\n",
       "        40,\n",
       "        45,\n",
       "        50,\n",
       "        55,\n",
       "        60,\n",
       "        65,\n",
       "        70,\n",
       "        75,\n",
       "        80,\n",
       "        85,\n",
       "        90,\n",
       "        95,\n",
       "        100],\n",
       "       'bin_freq': [0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        1,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        0,\n",
       "        119],\n",
       "       'n_smaller': 0,\n",
       "       'n_larger': 0}},\n",
       "     'siteQualityMetrics': [{'metric': 'FS', 'value': 1.1223795673810724},\n",
       "      {'metric': 'InbreedingCoeff', 'value': -0.0008395130824121466},\n",
       "      {'metric': 'MQ', 'value': 59.99216274475224},\n",
       "      {'metric': 'MQRankSum', 'value': 0},\n",
       "      {'metric': 'QD', 'value': 13.505564},\n",
       "      {'metric': 'ReadPosRankSum', 'value': 0.345},\n",
       "      {'metric': 'SiteQuality', 'value': 100738},\n",
       "      {'metric': 'SOR', 'value': 0.804},\n",
       "      {'metric': 'VQSLOD', 'value': None}]}},\n",
       "   'flags': [],\n",
       "   'rsid': 'rs121965019',\n",
       "   'sortedTranscriptConsequences': [{'canonical': True,\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_version': '12',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Trp402Ter',\n",
       "     'hgvsc': 'c.1205G>A',\n",
       "     'hgvsp': 'p.Trp402Ter',\n",
       "     'lof': 'HC',\n",
       "     'lof_flags': None,\n",
       "     'lof_filter': None,\n",
       "     'major_consequence': 'stop_gained',\n",
       "     'polyphen_prediction': None,\n",
       "     'sift_prediction': None,\n",
       "     'transcript_id': 'ENST00000247933',\n",
       "     'transcript_version': '8'},\n",
       "    {'canonical': None,\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_version': '12',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'p.Trp270Ter',\n",
       "     'hgvsc': 'c.809G>A',\n",
       "     'hgvsp': 'p.Trp270Ter',\n",
       "     'lof': 'HC',\n",
       "     'lof_flags': None,\n",
       "     'lof_filter': None,\n",
       "     'major_consequence': 'stop_gained',\n",
       "     'polyphen_prediction': None,\n",
       "     'sift_prediction': None,\n",
       "     'transcript_id': 'ENST00000514224',\n",
       "     'transcript_version': '1'},\n",
       "    {'canonical': None,\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_version': '12',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'n.7G>A',\n",
       "     'hgvsc': 'n.7G>A',\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_flags': None,\n",
       "     'lof_filter': None,\n",
       "     'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "     'polyphen_prediction': None,\n",
       "     'sift_prediction': None,\n",
       "     'transcript_id': 'ENST00000502829',\n",
       "     'transcript_version': '1'},\n",
       "    {'canonical': None,\n",
       "     'gene_id': 'ENSG00000127415',\n",
       "     'gene_version': '12',\n",
       "     'gene_symbol': 'IDUA',\n",
       "     'hgvs': 'n.1312G>A',\n",
       "     'hgvsc': 'n.1312G>A',\n",
       "     'hgvsp': None,\n",
       "     'lof': None,\n",
       "     'lof_flags': None,\n",
       "     'lof_filter': None,\n",
       "     'major_consequence': 'non_coding_transcript_exon_variant',\n",
       "     'polyphen_prediction': None,\n",
       "     'sift_prediction': None,\n",
       "     'transcript_id': 'ENST00000514698',\n",
       "     'transcript_version': '5'}]}}}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "js"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>data.variant.variantId</th>\n",
       "      <th>data.variant.reference_genome</th>\n",
       "      <th>data.variant.chrom</th>\n",
       "      <th>data.variant.pos</th>\n",
       "      <th>data.variant.ref</th>\n",
       "      <th>data.variant.alt</th>\n",
       "      <th>data.variant.colocatedVariants</th>\n",
       "      <th>data.variant.multiNucleotideVariants</th>\n",
       "      <th>data.variant.exome</th>\n",
       "      <th>data.variant.genome.ac</th>\n",
       "      <th>...</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.pab_max</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.QD</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.ReadPosRankSum</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.RF</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.SiteQuality</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.SOR</th>\n",
       "      <th>data.variant.genome.qualityMetrics.siteQualityMetrics.VQSLOD</th>\n",
       "      <th>data.variant.flags</th>\n",
       "      <th>data.variant.rsid</th>\n",
       "      <th>data.variant.sortedTranscriptConsequences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4-1002747-G-A</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>4</td>\n",
       "      <td>1002747</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>[4-1002747-G-T]</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>120</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>13.505564</td>\n",
       "      <td>0.345</td>\n",
       "      <td>None</td>\n",
       "      <td>100738</td>\n",
       "      <td>0.804</td>\n",
       "      <td>None</td>\n",
       "      <td>[]</td>\n",
       "      <td>rs121965019</td>\n",
       "      <td>[{'canonical': True, 'gene_id': 'ENSG000001274...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 55 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  data.variant.variantId data.variant.reference_genome data.variant.chrom  \\\n",
       "0          4-1002747-G-A                        GRCh38                  4   \n",
       "\n",
       "   data.variant.pos data.variant.ref data.variant.alt  \\\n",
       "0           1002747                G                A   \n",
       "\n",
       "  data.variant.colocatedVariants data.variant.multiNucleotideVariants  \\\n",
       "0                [4-1002747-G-T]                                   []   \n",
       "\n",
       "  data.variant.exome  data.variant.genome.ac  ...  \\\n",
       "0               None                     120  ...   \n",
       "\n",
       "   data.variant.genome.qualityMetrics.siteQualityMetrics.pab_max  \\\n",
       "0                                               None               \n",
       "\n",
       "   data.variant.genome.qualityMetrics.siteQualityMetrics.QD  \\\n",
       "0                                          13.505564          \n",
       "\n",
       "   data.variant.genome.qualityMetrics.siteQualityMetrics.ReadPosRankSum  \\\n",
       "0                                              0.345                      \n",
       "\n",
       "   data.variant.genome.qualityMetrics.siteQualityMetrics.RF  \\\n",
       "0                                               None          \n",
       "\n",
       "  data.variant.genome.qualityMetrics.siteQualityMetrics.SiteQuality  \\\n",
       "0                                             100738                  \n",
       "\n",
       "  data.variant.genome.qualityMetrics.siteQualityMetrics.SOR  \\\n",
       "0                                              0.804          \n",
       "\n",
       "  data.variant.genome.qualityMetrics.siteQualityMetrics.VQSLOD  \\\n",
       "0                                               None             \n",
       "\n",
       "  data.variant.flags data.variant.rsid  \\\n",
       "0                 []       rs121965019   \n",
       "\n",
       "           data.variant.sortedTranscriptConsequences  \n",
       "0  [{'canonical': True, 'gene_id': 'ENSG000001274...  \n",
       "\n",
       "[1 rows x 55 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.json_normalize(js)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
